diff --git a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
index 70d411ed..01544e95 100644
--- a/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
+++ b/ACHNBrowserUI/ACHNBrowserUI.xcodeproj/project.pbxproj
@@ -218,12 +218,15 @@
 		024363D72465EA15006A37A0 /* TodayCollectionProgressSection.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = TodayCollectionProgressSection.swift; sourceTree = "<group>"; };
 		028620742454C32800795719 /* AppIconPickerView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AppIconPickerView.swift; sourceTree = "<group>"; };
 		157C03E6247AE56600568917 /* TodaySectionView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; name = TodaySectionView.swift; path = ACHNBrowserUI/views/todayDashboard/TodaySectionView.swift; sourceTree = SOURCE_ROOT; };
-		22EB1DC0246D4D1F00C62F76 /* ru */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ru; path = ru.lproj/Localizable.strings; sourceTree = "<group>"; };
-		3246F93724725C37001A1E06 /* zh-Hans */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = "zh-Hans"; path = "zh-Hans.lproj/Localizable.strings"; sourceTree = "<group>"; };
+		1EEB847F2486660F0084CB25 /* en */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = en; path = en.lproj/Localizable.strings; sourceTree = "<group>"; };
+		1EEB8480248666840084CB25 /* es */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = es; path = es.lproj/Localizable.strings; sourceTree = "<group>"; };
+		1EEB84812486671D0084CB25 /* zh-Hant-TW */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = "zh-Hant-TW"; path = "zh-Hant-TW.lproj/Localizable.strings"; sourceTree = "<group>"; };
+		1EEB8483248668350084CB25 /* it */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = it; path = it.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		22EB1DC0246D4D1F00C62F76 /* ru */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = ru; path = ru.lproj/Localizable.strings; sourceTree = "<group>"; };
+		3246F93724725C37001A1E06 /* zh-Hans */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = "zh-Hans"; path = "zh-Hans.lproj/Localizable.strings"; sourceTree = "<group>"; };
 		325C0943246053E10043B6A7 /* SwiftUI.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = SwiftUI.framework; path = Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/SwiftUI.framework; sourceTree = DEVELOPER_DIR; };
-		3DB1260324686F0B00E74AA5 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/InfoPlist.strings; sourceTree = "<group>"; };
-		3DF14B2C24664B2600438646 /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/Localizable.strings; sourceTree = "<group>"; };
-		3DF14B2E24664B4A00438646 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/Localizable.strings; sourceTree = "<group>"; };
+		3DB1260324686F0B00E74AA5 /* de */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = de; path = de.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		3DF14B2E24664B4A00438646 /* de */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = de; path = de.lproj/Localizable.strings; sourceTree = "<group>"; };
 		4C16FB40247AC9B0009F24E3 /* GridStack.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = GridStack.swift; sourceTree = "<group>"; };
 		4C2D676A245DA41E005831C4 /* TurnipsChartView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TurnipsChartView.swift; sourceTree = "<group>"; };
 		4C2D676D245F0666005831C4 /* TurnipsChartBottomLegendView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = TurnipsChartBottomLegendView.swift; sourceTree = "<group>"; };
@@ -298,9 +301,7 @@
 		69875F5624654F7100B1D46C /* UserListFormViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = UserListFormViewModel.swift; sourceTree = "<group>"; };
 		69875F58246557D000B1D46C /* UserListDetailView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = UserListDetailView.swift; sourceTree = "<group>"; };
 		69875F5A24655AF000B1D46C /* UserListDetailViewModel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = UserListDetailViewModel.swift; sourceTree = "<group>"; };
-		6992892A24800A0C00F5CBC0 /* ja */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ja; path = ja.lproj/Localizable.strings; sourceTree = "<group>"; };
-		6992892B24800A1900F5CBC0 /* ja */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = ja; path = ja.lproj/InfoPlist.strings; sourceTree = "<group>"; };
-		69933E0B246D1CDD00528B51 /* ZH_TW */ = {isa = PBXFileReference; fileEncoding = 10; lastKnownFileType = text.plist.strings; name = ZH_TW; path = ZH_TW.lproj/Localizable.strings; sourceTree = "<group>"; };
+		6992892A24800A0C00F5CBC0 /* ja */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; lineEnding = 0; name = ja; path = ja.lproj/Localizable.strings; sourceTree = "<group>"; };
 		69933E0C246D1D6100528B51 /* items_zh.json */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.json; path = items_zh.json; sourceTree = "<group>"; };
 		69933E0E246D476E00528B51 /* DocumentPickerView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DocumentPickerView.swift; sourceTree = "<group>"; };
 		69933E16246D85A600528B51 /* items_ja.json */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.json; path = items_ja.json; sourceTree = "<group>"; };
@@ -361,7 +362,6 @@
 		69CFCAD2245ADC7A0059C067 /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/LaunchScreen.storyboard; sourceTree = "<group>"; };
 		69CFCAD4245ADC7A0059C067 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
 		69CFCADA245ADC9C0059C067 /* HomeView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = HomeView.swift; sourceTree = "<group>"; };
-		69D99D4524829184001ABEF0 /* es */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = es; path = es.lproj/Localizable.strings; sourceTree = "<group>"; };
 		69E4BBAE245ADF26001035E2 /* UI */ = {isa = PBXFileReference; lastKnownFileType = folder; name = UI; path = ACHNBrowserUI/packages/UI; sourceTree = "<group>"; };
 		69E4BBB3245AE472001035E2 /* ItemRow.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ItemRow.swift; sourceTree = "<group>"; };
 		69E4BBB5245AE5C9001035E2 /* ItemDetailView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ItemDetailView.swift; sourceTree = "<group>"; };
@@ -992,13 +992,12 @@
 				Base,
 				de,
 				fr,
-				ZH_TW,
-				zh_TW,
 				ru,
 				"zh-Hans",
 				it,
 				ja,
 				es,
+				"zh-Hant-TW",
 			);
 			mainGroup = 69AF8C44243E283D001DD9CB;
 			packageReferences = (
@@ -1229,7 +1228,7 @@
 			isa = PBXVariantGroup;
 			children = (
 				3DB1260324686F0B00E74AA5 /* de */,
-				6992892B24800A1900F5CBC0 /* ja */,
+				1EEB8483248668350084CB25 /* it */,
 			);
 			name = InfoPlist.strings;
 			sourceTree = "<group>";
@@ -1237,15 +1236,16 @@
 		3DF14B2D24664B2600438646 /* Localizable.strings */ = {
 			isa = PBXVariantGroup;
 			children = (
-				3DF14B2C24664B2600438646 /* en */,
 				3DF14B2E24664B4A00438646 /* de */,
 				69AE062824683429006856E4 /* fr */,
-				69933E0B246D1CDD00528B51 /* ZH_TW */,
 				22EB1DC0246D4D1F00C62F76 /* ru */,
 				3246F93724725C37001A1E06 /* zh-Hans */,
 				69E98C6E247439AD00A200FE /* it */,
 				6992892A24800A0C00F5CBC0 /* ja */,
-				69D99D4524829184001ABEF0 /* es */,
+				1EEB847F2486660F0084CB25 /* en */,
+				1EEB8480248666840084CB25 /* es */,
+				1EEB84812486671D0084CB25 /* zh-Hant-TW */,
+				1E15379524867B00007B9DF0 /* Base */,
 			);
 			name = Localizable.strings;
 			sourceTree = "<group>";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/de.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/de.lproj/Localizable.strings
index 530b13a7..8cf0683d 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/de.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/de.lproj/Localizable.strings
@@ -299,7 +299,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 // IslandDetailView
 "Join Island" = "Insel beitreten";
 "Island" = "Insel";
-"Description" = "Beschreibung";
 "Visitors" = "Besucher";
 "Loading Visitors...." = "Lade Besucher....";
 
@@ -400,8 +399,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 // CategoriesView
 "Nature" = "Natur";
 "Wardrobe" = "Kleidung";
-"No results for %@" = "Kein Ergebnis für %@";
-
 
 // Shared
 // SearchField
@@ -454,7 +451,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "Catching with a net (Fall)" = "Mit Kescher fangen (Herbst)";
 "Catching with a net (Winter)" = "Mit Kescher fangen (Winter)";
 "Catching with a net while on a quest for Wisp" = "Mit Kescher fangen während der Aufgabe von Buhu";
-"Celeste" = "Eufemia";
 "Cherry blossom seasonal DIY using cherry-blossom petals (caught with net)" = "Bastelanleitung aus der Kirschblüten-Saison; benötigt Kirschblüten (mit Kescher fangen)";
 "Cherry tree" = "Kirschbaum";
 "Chopping a bamboo tree (Spring)" = "Bambusbaum fällen (Frühling)";
@@ -526,8 +522,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "Jolly Redd's Treasure Trawler" = "Reiners Schatzkutter";
 "K.K. concert" = "K.K. Konzert";
 "K.K. concert, Nook Shopping Catalog" = "K.K. Konzert, Nook Shopping Katalog";
-"Kicks" = "Schubert";
-"Label" = "Minna";
 "Labelle" = "Minna";
 "Large rug" = "Großer Teppich";
 "Large shadow" = "Großer Schatten";
@@ -547,7 +541,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "Nook Shopping Promotion" = "Nook Shopping Werbung";
 "Nook Shopping Seasonal" = "Nook Shopping Saisonal";
 "Nook Shopping" = "Nook Shopping";
-"Nook's Cranny" = "Nooks Laden";
 "Nook's Cranny\nCrafting" = "Nooks Laden\nBasteln";
 "Normal villagers" = "\"Ausgeglichene\" Bewohner";
 "Obtained from Tom Nook after completing one Nature Day Nook Miles+ task" = "Erhalten von Tom Nook nach Abschluss einer Tag der Natur Nook Meilen+ Aufgabe";
@@ -642,7 +635,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "River (mouth)" = "Flussmündung";
 "River" = "Fluss";
 "Rover" = "Olli";
-"Saharah" = "Aziza";
 "Season sale item" = "Saisonverkaufsartikel";
 "Season sale item\n*Not available in Europe" = "Saisonverkaufsartikel\n*Nicht in Europa verfügbar";
 "Seed bag" = "Saatgutbeutel";
@@ -729,7 +721,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "Take your chance and go visit some mystery island!" = "Nutz deine Chance und besuche mysteriöse Inseln!";
 "Native" = "Nativ"; // TODO needs a better one
 "Normal" = "Normal";
-"Uncommon" = "Ungewöhnlich";
 "Gold" = "Gold";
 "Trash" = "Müll";
 "Bamboos" = "Bambus";
@@ -807,7 +798,7 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 
 // Added 2020-05-23
 "Museum Day" = "Internationaler Museumstag";
-"Wedding Season" = "Hochzeitssaison";
+"Wedding season" = "Hochzeitssaison";
 "Bug-Off contest" = "Insektikus-Turnier";
 "Meteor shower" =  "Sternschnuppen";
 
@@ -829,7 +820,6 @@ Du kannst das Abonnement jederzeit in deinen iTunes Kontoeinstellungen beenden.
 "Add Chore" = "Aufgabe hinzufügen";
 "Chores" = "Aufgaben";
 "Have a friend to visit? A gift to a villager? Track your chores and to-dos here." = "Du willst einen Freund besuchen? Einem Bewohner ein bestimmtes Geschenk überreichen? Verwalte hier deine Augaben und To-Dos.";
-"Reset" = "Zurücksetzen";
 "Manage and keep track of your chores and to-dos." = "Verwalte und beobachte deine Aufgaben und To-Dos.";
 
 // Design
diff --git a/ACHNBrowserUI/ACHNBrowserUI/en.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/en.lproj/Localizable.strings
index 02fae42f..3afddd47 100644
Binary files a/ACHNBrowserUI/ACHNBrowserUI/en.lproj/Localizable.strings and b/ACHNBrowserUI/ACHNBrowserUI/en.lproj/Localizable.strings differ
diff --git a/ACHNBrowserUI/ACHNBrowserUI/es.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/es.lproj/Localizable.strings
index 6b215365..608ee1ef 100644
Binary files a/ACHNBrowserUI/ACHNBrowserUI/es.lproj/Localizable.strings and b/ACHNBrowserUI/ACHNBrowserUI/es.lproj/Localizable.strings differ
diff --git a/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
index c12b3b4d..57d04d8f 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/fr.lproj/Localizable.strings
@@ -152,7 +152,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Nook's Cranny upgraded" = "Boutique Nook améliorée";
 "Able sisters" = "Soeur Doigts de Fée";
 "Visiting" = "En tant que visiteur";
-"Shop" = "Magasin";
 "Residents service" = "Mairie";
 "Building" = "Batîment";
 "App Settings" = "Réglages de l'application";
@@ -171,7 +170,7 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Edit" = "Éditer";
 "Search items" = "Chercher des objets";
 "Items added to your list from the search will be displayed there." = "Les objets ajoutés à votre liste vont apparaitre ici";
-"No results for %@" = "Aucun résultat pour %@";
+"No results for %@" = "Pas de résultats pour %@";
 
 // UserListSubscribeCallView
 "In order to create more than one list, you need to subscribe to AC Helper+" = "Pour pouvoir créer plus d'une liste, vous devez souscrire à AC Helper+";
@@ -310,7 +309,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Jock" = "Sportif";
 "Cranky" = "Grincheux";
 "Smug" = "Suffisant";
-"Normal" = "Normal";
 "Peppy" = "Énergique";
 "Snooty" = "Prétentieux";
 "Uchi" = "Big sister";
@@ -401,7 +399,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 // CategoriesView
 "Nature" = "Nature";
 "Wardrobe" = "Garde-robe";
-"No results for %@" = "Pas de résultats pour %@";
 
 // Shared
 // SearchField
@@ -583,7 +580,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Art Deco Walls" = "Murs Art Deco";
 "Outdoors" = "pleine air";
 "Rose Rug" = "Tapis Rose";
-"Fish" = "Poisson";
 "shop" = "magasin";
 "Wood Plank Walls" = "Murs en planches de bois";
 "Party" = "fête";
@@ -593,7 +589,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Home Appliances" = "Appareils électroménagers";
 "Shop" = "Magazin";
 "Decadence" = "Décadence";
-"Orange" = "Orange";
 "Kitchen Things" = "Choses de cuisine";
 "Herringbone" = "Chevrons";
 "Brown" = "Marron";
@@ -659,13 +654,11 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Beauty" = "Jolie";
 "Tatami" = "Tamami";
 "Camouflage" = "Camouflage";
-"School" = "école";
 "Throwback" = "nostalique";
 "Colorful" = "coloré";
 "Flower Walls" = "Murs Fleurs";
 "Dot" = "Poid";
 "Nature - Green" = "Nature - Vert";
-"Cute" = "mignon";
 "Hall Walls" = "Murs de Hall";
 "Theatrical" = "théâtrale";
 "Air Conditioning" = "Air conditionnée";
@@ -701,7 +694,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "NA" = "NA";
 "Fairy tale" = "compte de fée";
 "Special Warm Nature Walls" = "Murs spéciale chaud";
-"Iron" = "fer";
 "Elegant" = "Élégant";
 "Dining" = "Manger";
 "Wood Tile Walls" = "Murs en bois";
@@ -752,12 +744,10 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Cool" = "Cool";
 "Hospital" = "Hopital";
 "Morocco" = "Maroc";
-"Flowers" = "fleurs";
 "Everyday" = "tous les jours";
 "Study" = "Étude";
 "Work Bench" = "Table de travail";
 "House Door Decor" = "Décor de porte";
-"Space" = "espace";
 "Comfy" = "confortable";
 "Arch" = "Arche";
 "Space" = "Éspace";
@@ -799,7 +789,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 // Chore
 "Title" = "Titre";
 "Chore title" = "Nom de la tâche";
-"Description" = "Description";
 "Short description about the chore" = "Une petite description de la tâche";
 "Add Chore" = "Ajouter une tâche";
 "Chores" = "Tâches";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/it.lproj/InfoPlist.strings b/ACHNBrowserUI/ACHNBrowserUI/it.lproj/InfoPlist.strings
index 8ade8f06..e516d3b2 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/it.lproj/InfoPlist.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/it.lproj/InfoPlist.strings
@@ -1,4 +1,4 @@
-/* 
+/*
   InfoPlist.strings
   ACHNBrowserUI
 
diff --git a/ACHNBrowserUI/ACHNBrowserUI/it.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/it.lproj/Localizable.strings
index 5b071b7d..3969580e 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/it.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/it.lproj/Localizable.strings
@@ -1,3 +1,562 @@
+/*
+ Localizable.strings
+ ACHNBrowserUI
+ 
+ Created by Jan on 09.05.20.
+ Copyright © 2020 Thomas Ricouard. All rights reserved.
+ */
+
+"Loading..." = "Cargando...";
+"Tap here to preview a notification" = "Pulsa aquí para previsualizar una notificación";
+"Upgrade to +" = "Mejorar a +";
+"You're subscribed to AC Helper+" = "Te has suscrito a AC Helper+";
+
+"Close" = "Cerrar";
+"Dismiss" = "Cancelar";
+"Save" = "Guardar";
+
+"EEEE, dd MMMM" = "EEEE, dd MMMM";
+"dM" = "dM";
+"dd" = "dd";
+"MMM" = "MMM";
+"MMM dd" = "MMM dd";
+"dd MMMM" = "dd MMMM";
+
+"Dashboard" = "Pantalla Principal";
+"Catalog" = "Catálogo";
+"Turnips" = "Nabos";
+"Villagers" = "Aldeanos";
+"Collection" = "Colección";
+
+"Change Section Order" = "Cambiar el orden de las secciones";
+
+"Drag & Drop to Rearrange" = "Arrastra y suelta para reorganizar";
+"Hidden" = "Oculto";
+"Today Sections" = "Secciones principales";
+"Done" = "Hecho";
+" to re-order sections." = " para reorganizar las secciones.";
+"Check or un-check" = "verifica o no verifica";
+" rows to hide sections from the dashboard." = " para esconder las secciones principales";
+
+"Birthdays" = "Cumpleaños";
+"Today's Birthdays" = "Cumpleaños de hoy";
+"Today's Birthday" = "Cumpleaños de hoy";
+
+"Collection Progress" = "Progreso de la colección";
+"Share" = "Compartir";
+
+"Currently Available" = "Disponible ahora";
+"%lld/%lld Bugs" = "%lld/%lld Insectos";
+"%lld/%lld Fish" = "%lld/%lld Peces";
+"%lld NEW" = "%lld NUEVO";
+
+"Events" = "Eventos";
+"Currently" = "En este momento";
+"No Events Today" = "No hay eventos hoy";
+"Upcoming" = "Próximamente";
+
+"Today's Tasks" = "Tareas de hoy";
+"This feature requires " = "Esta función requiere ";
+"Learn more >" = "Saber más >";
+
+"Today is sunday, don't forget to buy more turnips and fill your buy price." = "¡Hoy es domingo! No olvides comprar más nabos y poner tu precio de compra.";
+"Today's average price should be around " = "El precio medio de hoy debería ser sobre";
+" in the morning, and " = " por la mañana, y ";
+" this afternoon." = " por la tarde.";
+
+"New on Nookazon" = "Nuevo en Nookazon";
+"Nookazon is loading..." = "Nookazon está cargando...";
+
+"What's New" = "Novedades";
+"See what's new in update %@" = "Mira qué hay de nuevo en esta actualización %@";
+
+"You're subscribed to AC Helper+. Thank you so much for you support!" = "Te has suscrito a AC Helper*. ¡Muchas gracias por tu apoyo!";
+"If you enjoy the application, consider subscribing to AC Helper+, to get access to some awesome features and support us!" = "¡Si disfrutas de la aplicación, considera suscribirte a AC Helper+, para obtener acceso a funciones asombrosas y apoyarnos!";
+
+"AC Helper+" = "AC Helper+";
+"ACHelperPlusDescription" = "Suscribirte a AC Helper+ es una excelente forma de mostrar tu apoyo a nuestro proyecto gratuito de código abierto.♥️
+
+¡También tendrás acceso a otras funciones, como predicciones de los precios de los nabos y la capacidad de crear más de una lista de objetos!  📈";
+
+"ACHelperPlusDetails" = "Sobre la función de notificación:
+
+Diariamente a las 8 am. y a las 12 pm., recibirás una notificación con el precio promedio de compra en tu tienda.
+
+¡Cuantos más precios diarios agregues después del lunes por la mañana en la aplicación, más precisas serán las próximas predicciones!
+
+Sobre la funcionalidad de la lista de objetos:
+
+En la aplicación gratuita, solo puedes crear una sola lista de elementos en la pestaña de su colección.
+
+Una vez que te hayas suscrito a AC Helper +, puedes crear tantas listas como desees.";
+
+"ACHelperPlusPriceAndAboDetail %@ %@" = "Las compras de %@ por mes o %@ por año serán aplicadas a tu cuenta de iTunes.
+
+Las suscripciones se renovarán automáticamente si no son canceladas 24h antes de finalizar la suscripción.
+
+Puedes cancelar en cualquier momento desde los ajustes de iTunes. Cualquier periodo de prueba gratuita restante si se compra una suscripción será perdido";
+"Subscribe for %@ / Month" = "Suscríbete por %@ / Mes";
+"%@ Monthly" = "%@ / Mes";
+"%@ Yearly" = "%@ / Año";
+"Buy lifetime AC Helper+ for %@" = "Comprar AC Helper+ de por vida por %@";
+"Thank you for your support!" = "¡Gracias por tu apoyo!";
+"Thanks!" = "¡Gracias!";
+
+"About" = "Información";
+"The app" = "La app";
+"Souce code / report an issue" = "Código fuente / señalar un problema";
+"Contact / follow us on Twitter" = "Contacto / síguenos en Twitter";
+"Rate the app on the App Store" = "Valorar la App en la App Store";
+"Privacy Policy" = "Política de Privacidad";
+"Term of Use" = "Términos de Uso";
+"App version" = "Version de la app";
+"Game patch data" = "Versión del parche del juego";
+"Acknowledgements" = "Reconocimientos";
+"Our amazing contributors" = "Nuestros increíbles colaboradores";
+"The NookPlaza API by Azarro" = "La API de NookPlaza de Azarro";
+"Turnip.exchange" = "Turnip.exchange";
+"Nookazon for the marketplace" = "Nookazon para el mercado";
+"Shihab / JPEGuin for the icon" = "Shihab / JPEGuin por el icono";
+"Christian & Ninji for the turnip predictions algorithm" = "Christian & Ninji por el algoritmo de predicción de los nabos";
+"Error" = "Error";
+
+"Active Critters" = "Bichos atrapables";
+"Fishes" = "Peces";
+"Bugs" = "Insectos";
+"You caught them all!" = "¡Los has atrapado todos!";
+"New this month" = "Nuevo este mes";
+"To catch" = "Por atrapar";
+"Leaving this month" = "Abandonan este mes";
+"Caught" = "Atrapados";
+
+"Preferences" = "Preferencias";
+"Island name" = "Nombre de la isla";
+"Your island name" = "El nombre de tu isla";
+"Hemisphere" = "Hemisferio";
+"North" = "Norte";
+"South" = "Sur";
+"Starting fruit" = "Fruta inicial";
+"Apple" = "Manzana";
+"Cherry" = "Cereza";
+"Orange" = "Naranja";
+"Peach" = "Melocotón";
+"Pear" = "Pera";
+"Coconut" = "Coco";
+"Nook shop" = "Tienda Nook";
+"Tent" = "Tienda de campaña";
+"Nook's Cranny" = "MiniNook";
+"Nook's Cranny upgraded" = "MiniNook mejorado";
+"Able sisters" = "Hermanas Manitas";
+"Visiting" = "Visitando";
+"Shop" = "Tienda física";
+"Residents service" = "Oficina de gestión vecinal";
+"Building" = "Edificio";
+"App Settings" = "Ajustes de la App";
+"App Icon" = "Icono de la App";
+"Restore purchase" = "Restaurar compra";
+
+"Edit your list" = "Edita tú lista";
+"Name of your list" = "Nombre de tu lista";
+"List name" = "Nombre de la lista";
+"Can be nothing" = "Puede ser nada";
+"Icon" = "Icono";
+
+"Add %lld items" = "Añadie %lld objetos";
+"Edit" = "Editar";
+"Search items" = "Buscar objetos";
+"Items added to your list from the search will be displayed there." = "Los objetos añadidos a tu lista aparecerán aquí";
+"No results for %@" = "No hay resultados para %@";
+
+"In order to create more than one list, you need to subscribe to AC Helper+" = "Para crear más de una lista, necesitas suscribirte a AC Helper+";
+"Learn more..." = "Saber más...";
+
+"Create a new list" = "Crear una nueva lista";
+"Please select or go stars some items!" = "¡Selecciona o marca algunos objetos!";
+"When you stars some \%@, they'll be displayed here." = "Cuando marques algunos \%@, aparecerán aquí.";
+"Items" = "Objetos";
+"Critters" = "Bichos";
+"Lists" = "Listas";
+"More" = "Más";
+
+"Clear Selection" = "Limpiar la selección";
+"Sort items" = "Ordenar objetos";
+"Current Sort: %@" = "Orden por: %@";
+"name" = "nombre";
+"buy" = "precio de compra";
+"sell" = "precio de venta";
+"from" = "obtención";
+"set" = "set";
+"similar" = "similar";
+"Name" = "Nombre";
+"Buy" = "Precio de compra";
+"Sell" = "Precio de venta";
+"From" = "Obtención";
+"Set" = "Set";
+"Similar" = "Similar";
+
+"unknown source" = "Origen desconocido";
+
+"Rarity:" = "Rareza:";
+"Shadow size:" = "Tamaño de sombra:";
+"X-Small" = "Muy pequeña";
+"Small" = "Pequeña";
+"Medium" = "Mediana";
+"Large" = "Grande";
+"X-Large" = "Muy grande";
+"Customizable: %@" = "Personalizable: %@";
+"Yes" = "Sí";
+"No" = "No";
+"Common" = "Común";
+"Same color" = "Del mismo color";
+
+"Preview" = "Previsualizar";
+
+"Seasonality" = "De temporada";
+
+"Set items" = "Objetos del set";
+"Simillar items" = "Objetos similares";
+"Thematics" = "Temáticas";
+"Variants" = "Variantes";
+"Materials" = "Materiales";
+"Nookazon listings" = "Nookazon";
+"Loading Listings..." = "Cargando...";
+"No listings found on Nookazon" = "No se ha encontrado en Nookazon";
+"Your items lists" = "Tu lista de objetos";
+"Keywords" = "Palabras clave";
+
+"Minmax" = "Min-Max";
+"Profits" = "Beneficio";
+"Average daily buy prices" = "Precio diario de compra";
+"Daily min-max prices" = "Precio min-max diario";
+"Average profits" = "Precio medio";
+"Chart" = "Gráfico";
+"Mon" = "Lun";
+"Tue" = "Mar";
+"Wed" = "Mie";
+"Thu" = "Jue";
+"Fri" = "Vie";
+"Sat" = "Sab";
+"Your prices" = "Tus precios";
+"Edit your in game prices" = "Editar tus precios del juego";
+"Add your in game prices" = "Añadir tus precios de juego";
+"Turnips prediction" = "Predicción de nabos";
+"To help us support the application and get turnip predictions notification, you can try out AC Helper+" = "Para apoyar la aplicación y conseguir la notificación de predicción de nabos, puedes probar AC Helper+";
+"To help us support the application and get background playing of all the musics, you can try out AC Helper+" = "Para apoyar la aplicación y conseguir la música de fondo de todas las canciones, puedes probar AC Helper++";
+"%lld upcomingDailyNotifications" = "Has recibido %lld notificaciones.";
+"Profits estimates are computed using the average of the current period" = "Los beneficios estimados se basan en el precio medio del período actual";
+"Day" = "Día";
+"AM" = "AM";
+"PM" = "PM";
+"Please add the amount of turnips you bought and for how much" = "Por favor, añade la cantidad de nabos que has comprado y por cuánto";
+"Add your in game turnip prices to see the predictions chart" = "Añade tu precio de compra de nabos en el juego para ver el gráfico";
+"Add your in game turnip prices to see predictions" = "Añade tu precio de compra de nabos en el juego para ver predicciones";
+"Exchange" = "Cambio";
+
+"Monday" = "Lunes";
+"Tuesday" = "Martes";
+"Wednesday" = "Miércoles";
+"Thursday" = "Jueves";
+"Friday" = "Viernes";
+"Saturday" = "Sábado";
+"Add your turnip prices" = "Ajustar Los precios de los nabos";
+"Configuration" = "Configuración";
+"Clear all fields" = "Limpiar todo";
+"Receive prices predictions notification" = "Recibir las notificaciones de predicciones";
+"You can get daily notifications for your average turnip price by subscribing to AC Helper+" = "Puedes obtener notificaciones diarias de tu precio medio de nabos suscribiéndote a AC Helper+";
+"Preview a notification" = "Ver un ejemplo de notificación";
+"Your in game prices" = "Tus precios de juego";
+"Buy price" = "Precio de compra";
+"Amount bought" = "Cantidad comprada";
+"The more in game buy prices you'll add the better the predictions will be. Your buy price only won't give your correct averages. Add prices from the game as you get them daily." = "Cuantos mas precios de compra añadas mejor mes serán las predicciones. Solo tu precio de compra no te dará predicciones correctas. Añade precios del juego diariamente";
+"View on TurnipProphet" = "Ver en TurnipProphet";
+"View Turnip.Exchange islands" = "Ver las Islas de Turnip.Exchange";
+
+"Average" = "Medio";
+"Minimum" = "Mínimo";
+"MinMax" = "MinMax";
+
+"Join Island" = "Unirte a la isla";
+"Island" = "Isla";
+"Description" = "Descripción";
+"Visitors" = "Visitantes";
+"Loading Visitors...." = "Cargando visitantes....";
+
+"Search a villager" = "Buscar un vecino";
+"Personality" = "Personalidad";
+"Birthday" = "Cumpleaños";
+"Species" = "Especie";
+"Gender" = "Género";
+"Catch phrase" = "Frase típica";
+"Unknown" = "Desconocido";
+"Villager items" = "Objetos de vecinos";
+"Sort villagers" = "Ordenar vecinos";
+
+"Unkown" = "Desconocido";
+"Lazy" = "Perezoso";
+"Jock" = "Atlético";
+"Cranky" = "Gruñón";
+"Smug" = "Snob";
+"Normal" = "Normal";
+"Peppy" = "Alegre";
+"Snooty" = "Presumida";
+"Uchi" = "Dulce";
+"Male" = "Masculino";
+"Female" = "Femenino";
+"Alligator" = "Cocodrilo";
+"Anteater" = "Oso hormiguero";
+"Bear" = "Oso";
+"Bird" = "Pájaro";
+"Bull" = "Toro";
+"Cat" = "Gato";
+"Chicken" = "Pollo";
+"Cow" = "Vaca";
+"Cub" = "Lobo";
+"Deer" = "Ciervo";
+"Dog" = "Perro";
+"Duck" = "Pato";
+"Eagle" = "Águila";
+"Elephant" = "Elefante";
+"Frog" = "Rana";
+"Goat" = "Cabra";
+"Gorilla" = "Gorila";
+"Hamster" = "Hamster";
+"Hippo" = "Hipopótamo";
+"Horse" = "Casallo";
+"Kangaroo" = "Canguro";
+"Koala" = "Koala";
+"Lion" = "León";
+"Monkey" = "Mono";
+"Mouse" = "Ratón";
+"Octopus" = "Pulpo";
+"Ostrich" = "Avestruz";
+"Penguin" = "Pingüino";
+"Pig" = "Cerdo";
+"Rabbit" = "Coneji";
+"Rhino" = "Rinoceronte";
+"Sheep" = "Oveja";
+"Squirrel" = "Ardilla";
+"Tiger" = "Tigre";
+"Wolf" = "Lobo";
+"Like" = "Le gusta";
+"Gifts ideas" = "Ideas de regalo";
+
+"Make an Offer" = "Hacer una oferta";
+"Need Materials" = "Se necesitan materiales";
+
+"Turnip prices" = "Precio de nabos";
+"this morning" = "esta mañana";
+"this afternoon" = "esta tarde";
+"Your prices predictions for %@ should be around %lld bells. With a minimum of %lld and a maximum of %lld." = "Las predicciones para %@ deberían ser de %lld bayas. Con un mínimo de %lld y un máximo de %lld.";
+"Turnip prices (test)" = "recio de nabos (test)";
+"Your prices predictions for this morning should be around 100 bells. With a minimum of 50 and a maximum of 650." = "Las predicciones para esta mañana deberían ser de 100 bayas. Con un mínimo de 50 y un máximo de 650.";
+"Friendly reminder" = "Recordatorio amistoso";
+"It's time to buy turnips and add the buy price in the app." = "Es hora de comprar nabos y añadir el precio en la app.";
+"It's the start of the week, don't forget to add your store buy prices in the app as you play the game. You'll get more accurate predictions." = "Empieza la semana, no olvides añadir los precios de tu tienda en la app cuando juegues. Obtendrás unas predicciones más seguras. ";
+
+"housewares" = "Artículos del hogar";
+"miscellaneous" = "Varios";
+"wall-mounted" = "De pared";
+"wallpapers" = "Paredes";
+"floors" = "Suelos";
+"rugs" = "Alfombras";
+"photos" = "Fotos";
+"posters" = "Posters";
+"fencing" = "Vallas";
+"tools" = "Herramientas";
+"tops" = "Partes de arriba";
+"bottoms" = "Pantalones";
+"dresses" = "Vestidos";
+"headwear" = "Gorros";
+"accessories" = "Accesorios";
+"socks" = "Calcetines";
+"shoes" = "Zapatos";
+"bags" = "Bolsas";
+"umbrellas" = "Paraguas";
+"music" = "Música";
+"recipes" = "Recetas";
+"construction" = "Construcción";
+"nookmiles" = "Millas Nook";
+"other" = "Otros";
+"art" = "Arte";
+"bugs" = "Insectos";
+"fish" = "Peces";
+"fossils" = "Fósiles";
+
+"Nature" = "Natural";
+"Wardrobe" = "Armario";
+"No results for %@" = "Sin resultados para %@";
+
+"Search..." = "Buscar…";
+
+"See more" = "Ver más";
+"See less" = "Ver menos";
+
+"My data" = "Mis datos";
+"Collection size" = "Tamaño de mi colección";
+"Export my collection" = "Exportar mi colección";
+"Import a collection" = "Importar una colección";
+"Imported with success" = "Importado con éxito";
+"Reset my collection" = "Eliminar my colección";
+"Your AC Helper collection file was imported with success." = "Tu colección de AC Helper ha sido importada con éxito";
+"Are you sure?" = "¿Estás seguro?";
+"This will delete all your collected critters, items, villagers and items lists" = "Eso borrará todas tus colecciones de bichos, vecinos, objetos y listas.";
+
+"Mystery Islands" = "Islas misteriosas";
+"Trees: %@" = "Árboles: %@";
+"Trees" = "Árboles";
+"Flowers" = "Flores";
+"Flowers: %@" = "Flores: %@";
+"Fish" = "Peces";
+"Insects" = "Insectos";
+"Rocks Type" = "Tipo de rocas";
+"Rocks" = "Rocas";
+"Take your chance and go visit some mystery island!" = "¡Toma tu oportunidad y visita alguna Isla misteriosa!";
+"Native" = "Nativo";
+"Normal" = "Normal";
+"Uncommon" = "Poco común";
+"Gold" = "Pepitas de oro";
+"Trash" = "Basura";
+"Bamboos" = "Bambús";
+"Hardwood" = "Caducifolio";
+"Exotic fruits" = "Frutas exóticas";
+"Normal Island 1" = "Isla Normal 1";
+"Normal Island 2" = "Isla Normal 2";
+"Spiral Island" = "Isla Espiral";
+"Fidget Spinner Island" = "Isla del Fidget Spinner";
+"Mountain Island" = "Isla de la Montaña";
+"Bells Island" = "Isla de las Bayas";
+"Bamboo Island" = "Isla de Bambú";
+"Fruit Island" = "Isla de la Fruta";
+"Flower Island" = "Isla de las Flores";
+"Bell Island 2" = "Isla de las Bayas 2";
+"Tarantula Island" = "Isla de las Tarántulas";
+"Tree Island" = "Isla de los Árboles";
+"Big fish Island" = "Isla de Peces Grandes";
+"Tree Island 2" = "Isla de los Árboles 2";
+"Curvy River Island" = "Isla del Río Curvo";
+"Big Fish Island 2" = "Isla de los Peces Grandes 2";
+"Trash Island" = "Isla de la Basura";
+"Fins Island" = "Isla de Aletas";
+"Falls Island" = "Isla de las Cascadas";
+"Gold Island" = "Isla de las Pepitas de Oro";
+" - max %lld visit" = " - máx %lld visita";
+
+"River" = "Río";
+"Sea" = "Océano";
+"Ponds" = "Estanques";
+"Pond" = "Estanque";
+"Pier" = "Muelle";
+"Raining" = "Lluvia";
+"Clifftop" = "Sobre acantilado";
+"Mouth" = "Desembocadura";
+"Flying near flowers" = "Volando cerca de flores";
+"On the ground" = "En el suelo";
+"Flying" = "Volando por el aire";
+"Shaking trees (hardwood or cedar only)" = "Agitando árboles (Solo caducifolios o cedros)";
+"On trees (any kind)" = "En árboles (de todo tipo)";
+"Disguised on shoreline" = "Distinguido en la costa";
+"Pushing snowballs" = "Poniendo bolas de nieve";
+"On flowers" = "En las flores";
+"On tree stumps" = "En tocones de árbol";
+"On palm trees" = "Dn las palmeras";
+"On hardwoodcedar trees" = "En árboles caduciformes";
+"From hitting rocks" = "Golpeando rocas";
+"On riverponds" = "En estanques";
+"On rocks" = "En las rocas";
+"On rotten turnips" = "En nabos podridos";
+"Shaking trees" = "Agitando árboles";
+"On white flowers" = "En flores blancas";
+"Flying near trash (boots, tires, cans)" = "Volando cerca de basura (botas, neumáticos, latas...)";
+"Flying near bluepurpleblack flowers" = "Volando cerca de flores azules/moradas/negras";
+"Deguised under trees" = "Camuflado bajo árboles";
+"On villager" = "En un vecino";
+"Underground (dig where noise is loudest)" = "Bajo tierra (cavar donde el sonido es más fuerte)";
+"Jolly Redd's Treasure Trawler" = "Navío Ambulante de Ladino";
+"Mom" = "Mamá";
+"Dad" = "Papá";
+"Fishing Tourney" = "Torneo de pesca";
+"Picking flowers" = "Recoger flores";
+"Hitting a rock" = "Golpear una roca";
+"Beach" = "Playa";
+
+"A fairly normal island which you can get on your first tour, with one cliff, native fruit, coconuts, a river and a pond." = "Una isla normal donde puedes ir en tu primer tour, con un acantilado, fruta nativa, cocos, un río y un estanque";
+"Practically the same as Normal Island 1, but without the pond, and with a slightly bigger cliff." = "Prácticamente lo mismo que la Isla Normal 1, pero sin el estanque y con un acantilado un poco más grande.";
+"Somewhat more interesting in layout but it has nothing special about the resources you can get." = "Un tipo con una forma diferente más interesante pero con ningún recurso especial para obtener.";
+"The last of the four starter islands. No river, just a pond and the sea." = "La última de las cuatro islas principales. Sin río, solo con un estanque y el mar.";
+"Multiple levels, no river. The base level has regular trees, the middle level has your native fruit, and the top level has five rocks with materials." = "Distintas alturas, sin río. El nivel base tiene árboles normales, el nivel medio tu fruta nativa, y el nivel más alto 5 rocas con materiales";
+"Breaking the rock at the far north will let you vault over into the island with five money rocks on it." = "Romper la roca del norte te permitirá saltar a la isla con 5 rocas de dinero en ella";
+"Every tree on this island is bamboo and you can get to it at any time of year. The most common island you can get out of the 20, statistically." = "Cada árbol de esta isla es bambú y puedes ir durante todo el año. La isla más común de conseguir de las 20, estadísticamente";
+"There are 19 fruit trees on this island and all will be your 'sister fruit'. This will be different from your native fruit - but every time you come here, it will be the same one." = "Hay 19 árboles frutales en esta isla y todos serán de tu 'fruta hermana'. Esta es distinta de tu fruta nativa - pero cada vez que vengas aquí, será la misma.";
+"This island has a pond surrounded by rare flowers and the only insects that will spawn are those that hang around flowers." = "Esta isla tiene un estanque rodeado de flores raras y los únicos insectos que aparecerán son aquellos que vuelan alrededor de flores";
+"There are four small triangular cliffs, one in each corner, and the ground level is full of flowers and money rocks." = "Hay 4 pequeños acantilados triangulares, uno en cada esquina, y el nivel del suelo está lleno de flores y rocas de dinero";
+"Yes, the infamous one. The only insects that spawn here are tarantulas (as long as it is night time in a Tarantula month), and the rocks have crafting materials if you can get to them. The river is thin enough that you can jump over it without a vaulting pole." = "Si, la infame. Los únicos insectos que aparecen aquí son tarántulas (mientras sea de noche y en un mes que haya tarántulas), y las rocas tienen materiales si puedes llegar a ellas. El río es lo suficientemente fino como para saltar sin pértiga.";
+"This island has no river or pond and has tons of hardwood and coconut trees, and the only insects that spawn are those that are associated with trees." = "Esta isla no tiene río ni estanque y tiene un montón de árboles caducifolios y palmeras de cocos, los únicos insectos que aparecen son aquellos asociados con los árboles.";
+"Big Fish Island. This island has rare hybrid flowers and the only fish available are larger ones, including some of the more rare ones." = "La isla de los peces grandes, tiene flores híbridas bastante raras y los únicos peces disponibles son los más grandes, incluyendo los más raros.";
+"This island is exactly the same as Bamboo Island but the trees are all just regular hardwood trees." = "Esta isla es exactamente igual que la Isla de Bambú, pero los árboles son únicamente caducifolios.";
+"There is one square cliff at the north east, a few flowers and rocks and a small amount of fruit trees. Only insects associated with water spawn here." = "Hay un acantilado cuadrado en el noreste, unas cuantas flores y rocas y un puñado de árboles frutales. Solo los insectos asociados con agua aparecen aquí.";
+"This rare island only spawns big fish. Otherwise it is quite normal, with less flowers and a lower chance of appearing than the other big fish island." = "Esta rara isla solo tiene peces grandes. Por lo demás es bastante normal, con menos flores y menos probabilidad de aparecer que la otra isla de peces grandes";
+"Everything you can fish here is trash. Only water-related insects spawn." = "Todo lo que pesques aquí será basura. Solo aparecen insectos relacionados con el agua.";
+"A rectangular pond with rectangular cliffs inside, the tallest one being too tall to climb up to. The only fish that spawn here are the largest finned fish." = "Un estanque rectangular con acantilados cuadrados dentro, el más alto es demasiado alto como para escalarlo. Los únicos peces que  aparecen aquí son los largos y finos.";
+"Lots of cliffs, waterfalls and river forks. Normal resource spawns." = "Muchos acantilados, cascadas y cruces de ríos. Aparecen recursos normales.";
+"A very rare island with flowers, scorpions and a rectangular pond. If you climb up onto the cliff and climb down from the back, you can vault over to a tiny island in the middle and get 8 gold nuggets from a single rock." = "Una isla muy rara con flores, escorpiones y un estanque rectangular. Si escalas el acantilado y bajas por la parte trasera, podrás llegar a una pequeña isla en el centro y obtener 8 pepitas de oro de una sola roca.";
+
+"Using iCloud sync" = "Usando sincronización con iCloud";
+"Synchronized with iCloud" = "Sincronizado con iCloud";
+
+"Possible visitors" = "Posibles visitantes";
+"On" = "en";
+" your island" = " tu isla";
+"All day" = "Todo el día";
+"From 6pm to 5am" = "De 18H a 5H";
+"From 8pm to 5am" = "De 20H a 5H";
+"From 5am to 12pm" = "De 5H a 12H";
+
+"K.K Slider" = "Totakeke";
+"Daisy Mae" = "Juliana";
+"C.J" = "CJ";
+"Flick" = "Kamilo";
+"Kicks" = "Betunio";
+"Saharah" = "Alcatifa";
+"Label" = "Trini";
+"Leif" = "Gandulio";
+"Redd" = "Ladino";
+"Wisp" = "Buh";
+"Celeste" = "Estela";
+
+"Museum Day" = "Día Internacional de los Museos";
+"Wedding Season" = "Temporada de bodas";
+"Bug-Off contest" = "Caza de Bichos";
+"Meteor shower" =  "Lluvia de meteoros";
+
+"Music player" = "Reproductor de música";
+"Tracks" = "Canciones";
+
+"Vacation" = "Vacaciones";
+"Stripe Walls" = "Pared de rayas";
+"Office" = "Oficina";
+"Tile Walls" = "Pared de azulejos";
+"Tea Room Walls" = "Paredes salón de té";
+"Zen" = "Zen";
+"Deco Wood" = "Decoración de madera";
+"Rubber" = "Caucho";
+"Picture" = "Foto";
+"Cloth Floors" = "Suelos de tela";
+"Audio" = "Música";
+"Fruits" = "Frutas";
+"Antique" = "Antigüedades";
+"Honeycomb" = "Panal";
+"Japanese Style" = "Estilo Japonés";
+"Simple Walls" = "Paredes Simples";
+"Golden" = "Dorado";
+"Simple" = "Simple";
+"Country" = "Campo";
+"Sports Floor" = "Suelo deportivo";
+"Bamboo" = "Bambú";
+"Art Deco Walls" = "Paredes decorativas del museo";
+"Outdoors" = "Exterior";
+"Rose Rug" = "Alfombra Rosa";
+"Fish" = "Pez";
+
 "Tap here to preview a notification" = "Tocca per un anteprima della notifica";
 "Upgrade to +" = "Aggiorna a +";
 "You're subscribed to AC Helper+" = "Ti sei abbonato a AC Helper+";
@@ -507,8 +1066,8 @@ Puoi annullare il tuo abbonamento in ogni momento dalle impostazioni del tuo acc
 "Flying near light sources" = "Vola vicino fonti luminose";
 "Flying near trash (boots, tires, cans)" = "Vola vicino rifiuti (Scarponi, copertoni, scatolette)";
 "Flying" = "Vola";
-"Assessing fossils" = "Dissotterrare fossili"
-"Flying near water" = "Vola vicino all'acqua"
+"Assessing fossils" = "Dissotterrare fossili";
+"Flying near water" = "Vola vicino all'acqua";
 "From hitting rocks" = "Colpire le rocce";
 "From Zipper after trading one of every egg (all 6 types) during Bunny Day." = "Da Ovidio scambiando un uovo per tipo (6 tipi) durante la Caccia all'uovo";
 "Glowing dig spot" = "Punto luccicante nel terreno";
@@ -703,7 +1262,6 @@ Puoi annullare il tuo abbonamento in ogni momento dalle impostazioni del tuo acc
 "Sea (raining)" = "Mare (Pioggia)";
 "Sea (rainy days)" = "Mare (Quando piove)";
 
-
 // Added 2020-05-13
 "See more" = "Più info";
 "See less" = "Meno info";
@@ -972,7 +1530,7 @@ Puoi annullare il tuo abbonamento in ogni momento dalle impostazioni del tuo acc
 "Elegant" = "Élegante";
 "Dining" = "Sala da pranzo";
 "Wood Tile Walls" = "Muri Piastrelle di Legno";
-"Tile Floors" = Pavimenti a Piastrelle";
+"Tile Floors" = "Pavimenti a Piastrelle";
 "School" = "Scuola";
 "Cardboard" = "Cartone";
 "Tin Walls" = "Muri di Latta";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/InfoPlist.strings b/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/InfoPlist.strings
deleted file mode 100644
index 962c69b0..00000000
--- a/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/InfoPlist.strings
+++ /dev/null
@@ -1,9 +0,0 @@
-/*
-  InfoPlist.strings
-  ACHNBrowserUI
-
-  Created by mimikun on 14.05.20.
-  Copy template from de.lproj/InfoPlist.strings
-*/
-
-NSPhotoLibraryAddUsageDescription = "Speichern von Bildern wie beispielsweise Möbel oder generierte Bilder aus der \"Teilen\"-Funktion";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/Localizable.strings
index bfcd4016..6cc006b3 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/ja.lproj/Localizable.strings
@@ -213,7 +213,7 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Small" = "S";
 "Medium" = "M";
 "Large" = "L";
-"X-Large" = "XL";
+"X-Large" = "XL";
 "Customizable: %@" = "カスタマイズ可能: %@";
 "Flick:" = "Flick:";  // TODO
 "Yes" = "Yes";
@@ -437,7 +437,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Catching with a net (Fall)" = "あみで捕まえる(秋)";
 "Catching with a net (Winter)" = "あみで捕まえる(冬)";
 "Catching with a net while on a quest for Wisp" = "ゆうたろうのお願い中にあみで捕まえる";
-"Celeste" = "フーコ";
 "Cherry blossom seasonal DIY using cherry-blossom petals (caught with net)" = "桜の花びら(あみで捕まえる)を使って桜の季節にDIY";
 "Cherry tree" = "桜の木";
 "Chopping a bamboo tree (Spring)" = "竹を切る(春)";
@@ -484,7 +483,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Fishing up three old tires" = "タイヤを3つ釣る";
 "Fishing up trash" = "ゴミを釣る";
 "Fishing" = "釣り";
-"Flick" = "Flick";  // TODO
 "Flying near blue/purple/black flowers" = "青/紫/黒の花の近くを飛ぶ";
 "Flying near flowers" = "花の近くを飛ぶ";
 "Flying near light sources" = "明かりの近くを飛ぶ";
@@ -510,8 +508,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Jolly Redd's Treasure Trawler" = "Jolly Redd's Treasure Trawler";  // TODO
 "K.K. concert" = "とたけけコンサート";
 "K.K. concert, Nook Shopping Catalog" = "とたけけコンサート, たぬきショッピングカタログ";
-"Kicks" = "シャンク";
-"Label" = "ことの";
 "Labelle" = "ことの";
 "Large rug" = "大きなラグ";
 "Large shadow" = "影(Lサイズ)";  // TODO
@@ -531,7 +527,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Nook Shopping Promotion" = "たぬきショッピングプロモーション";
 "Nook Shopping Seasonal" = "たぬきショッピング(シーズン)";
 "Nook Shopping" = "たぬきショッピング";
-"Nook's Cranny" = "タヌキ商店";
 "Nook's Cranny\nCrafting" = "タヌキ商店\n手作り";
 "Normal villagers" = "Normal villagers";  // TODO kokomade
 "Obtained from Tom Nook after completing one Nature Day Nook Miles+ task" = "アースデーのたぬきマイル+タスクを1つ完了した後にたぬきちから入手";
@@ -713,7 +708,6 @@ You can cancel anytime with your iTunes account settings. Any unused portion of
 "Take your chance and go visit some mystery island!" = "この機会に謎の島に行ってみよう!";  // TODO
 "Native" = "Native"; // TODO
 "Normal" = "普通";
-"Uncommon" = "アンコモン";
 "Gold" = "きんこうせき";
 "Trash" = "ゴミ";
 "Bamboos" = "バンブー";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/ru.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/ru.lproj/Localizable.strings
index 90c2373e..bf5ab505 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/ru.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/ru.lproj/Localizable.strings
@@ -215,7 +215,7 @@
 "Small" = "Маленький";
 "Medium" = "Средний";
 "Large" = "Большой";
-"X-Large" = "Очень большой";
+"X-Large" = "Очень большой";
 "Customizable: %@" = "Возможны изменения: %@";
 "Flick:" = "Флик:";
 "Yes" = "Да";
@@ -386,8 +386,6 @@
 // CategoriesView
 "Nature" = "Природа";
 "Wardrobe" = "Гардероб";
-"No results for %@" = "Нет результатов для %@";
-
 
 // Shared
 // SearchField
diff --git a/ACHNBrowserUI/ACHNBrowserUI/zh-Hans.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/zh-Hans.lproj/Localizable.strings
index 8f506707..5fbdfa07 100644
--- a/ACHNBrowserUI/ACHNBrowserUI/zh-Hans.lproj/Localizable.strings
+++ b/ACHNBrowserUI/ACHNBrowserUI/zh-Hans.lproj/Localizable.strings
@@ -221,7 +221,7 @@
 "Small" = "小";
 "Medium" = "中";
 "Large" = "大";
-"X-Large" = "非常大";
+"X-Large" = "非常大";
 "Customizable: %@" = "可以改造: %@";
 "Flick:" = "Flick:";
 "C.J:" = "C.J:";
@@ -337,8 +337,6 @@
 // CategoriesView
 "Nature" = "图鉴";
 "Wardrobe" = "衣橱";
-"No results for %@" = "没有找到%@";
-
 
 // Shared
 // SearchField
diff --git a/ACHNBrowserUI/ACHNBrowserUI/zh-Hant-TW.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/zh-Hant-TW.lproj/Localizable.strings
new file mode 100644
index 00000000..fc3903db
--- /dev/null
+++ b/ACHNBrowserUI/ACHNBrowserUI/zh-Hant-TW.lproj/Localizable.strings
@@ -0,0 +1,59 @@
+/*
+  Localizable.strings
+  ACHNBrowserUI
+
+  zh.tw
+  Created by Thus on 14.05.20.
+  First created by klin0816
+  Copy template from en.lproj/Localizable.strings
+*/
+
+// SubscribeView
+"ACHelperPlusDescription" = " 訂閱 AC Helper+ 是對我們這個免費和開源計畫最好的支持。♥️
+ 
+您能獲取更多功能，大頭菜價格預測通知和創建無數個物品清單！ 📈
+
+以後有更多介紹";
+"ACHelperPlusDetails" = "關於價格預測：
+每天早上8點和12點您會收到您的商店的平均收購價格通知。
+輸入越多從星期一早上開始的價格, 價格會預測得越準確！
+
+關於清單功能：
+在免費的版本中, 您只能在 \"我的東西\" 中創建一個清單。
+只要您訂閱了 AC Helper+ 就能夠創建無數個您想要的清單。";
+
+"ACHelperPlusPriceAndAboDetail %@ %@" = "確認訂閱後 %@ 個月購買會出現在您的 iTunes 帳戶。
+訂閱將會自動更新除非你在該期間結束的 24 小時前取消訂閱。
+你能夠隨時在您的 iTunes 帳戶中取消訂閱，如果你在試用期間內取消，可能會立即無法享用內容。";
+
+// CategoryRow
+"housewares" = "傢俱";
+"miscellaneous" = "雜物";
+"wall-mounted" = "壁掛物";
+"wallpapers" = "壁紙";
+"floors" = "地板";
+"rugs" = "地毯";
+"photos" = "照片";
+"posters" = "海報";
+"fencing" = "柵欄";
+"tools" = "工具";
+"tops" = "上身";
+"bottoms" = "下身";
+"dresses" = "套裝";
+"headwear" = "頭戴物";
+"accessories" = "飾品";
+"socks" = "襪子";
+"shoes" = "鞋子";
+"bags" = "包包";
+"umbrellas" = "雨傘";
+"music" = "音樂";
+"recipes" = "方程式";
+"construction" = "建築物";
+"nookmiles" = "里程";
+"other" = "其他";
+"art" = "藝術品";
+"bugs" = "蟲";
+"fish" = "魚";
+"fossils" = "化石";
+
+"%lld upcomingDailyNotifications" = "你會在接下來的 %lld 個通知中收到大頭菜價錢預測";
diff --git a/ACHNBrowserUI/ACHNBrowserUI/zh_tw.lproj/Localizable.strings b/ACHNBrowserUI/ACHNBrowserUI/zh_tw.lproj/Localizable.strings
deleted file mode 100644
index ba1e08f5..00000000
Binary files a/ACHNBrowserUI/ACHNBrowserUI/zh_tw.lproj/Localizable.strings and /dev/null differ
