diff --git a/OpenGpxTracker.xcodeproj/project.pbxproj b/OpenGpxTracker.xcodeproj/project.pbxproj
index df35d829..9b297b1d 100644
--- a/OpenGpxTracker.xcodeproj/project.pbxproj
+++ b/OpenGpxTracker.xcodeproj/project.pbxproj
@@ -49,6 +49,7 @@
 		89F6E107227F7C5500CE3E3F /* Double+Measures.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F6E0F4227D091900CE3E3F /* Double+Measures.swift */; };
 		89F8749B1BBB7362004EF41A /* GPXTrack+length.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F8749A1BBB7362004EF41A /* GPXTrack+length.swift */; };
 		89F8749F1BBCB97F004EF41A /* GPXRoot+length.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F8749E1BBCB97F004EF41A /* GPXRoot+length.swift */; };
+		BF214B5E22E60B64000E96AA /* CLActivityType+Info.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */; };
 		BF36B6CF220C6F1A00077B46 /* GPXFileTableInterfaceController.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF36B6CE220C6F1A00077B46 /* GPXFileTableInterfaceController.swift */; };
 		BF5C0D2222D771D1001C2575 /* DistanceLabel.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF5C0D2122D771D0001C2575 /* DistanceLabel.swift */; };
 		BF637D132216C5FA0003EC04 /* WatchConnectivity.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = BF637D122216C5FA0003EC04 /* WatchConnectivity.framework */; settings = {ATTRIBUTES = (Weak, ); }; };
@@ -180,6 +181,7 @@
 		93BD71BED3CF101C2B01ABDA /* Pods_OpenGpxTracker_Watch_Extension.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_OpenGpxTracker_Watch_Extension.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		95D28CBDF2535138FA16F38F /* Pods_OpenGpxTracker.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_OpenGpxTracker.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		9A591D96EF2E41D00AD6BB3A /* Pods-OpenGpxTracker-Watch Extension.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-OpenGpxTracker-Watch Extension.debug.xcconfig"; path = "Pods/Target Support Files/Pods-OpenGpxTracker-Watch Extension/Pods-OpenGpxTracker-Watch Extension.debug.xcconfig"; sourceTree = "<group>"; };
+		BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "CLActivityType+Info.swift"; sourceTree = "<group>"; };
 		BF36B6CE220C6F1A00077B46 /* GPXFileTableInterfaceController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = GPXFileTableInterfaceController.swift; sourceTree = "<group>"; };
 		BF5C0D2122D771D0001C2575 /* DistanceLabel.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = DistanceLabel.swift; sourceTree = "<group>"; };
 		BF637D122216C5FA0003EC04 /* WatchConnectivity.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = WatchConnectivity.framework; path = System/Library/Frameworks/WatchConnectivity.framework; sourceTree = SDKROOT; };
@@ -260,6 +262,7 @@
 				89EA86642157C444002E03A7 /* Int+asFilesize.swift */,
 				89EA86622157B42C002E03A7 /* GPXFileInfo.swift */,
 				BF6408A2225C63E700BB5242 /* CoreDataHelper.swift */,
+				BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */,
 			);
 			name = Models;
 			sourceTree = "<group>";
@@ -720,6 +723,7 @@
 				899D677019CE024600C88A0A /* GPXFileManager.swift in Sources */,
 				898EECDE19C49B5800B4B207 /* OpenGpxTracker.xcdatamodeld in Sources */,
 				89EA86652157C444002E03A7 /* Int+asFilesize.swift in Sources */,
+				BF214B5E22E60B64000E96AA /* CLActivityType+Info.swift in Sources */,
 				BF5C0D2222D771D1001C2575 /* DistanceLabel.swift in Sources */,
 				89F6E0F5227D091900CE3E3F /* Double+Measures.swift in Sources */,
 				8913CD8E1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift in Sources */,
diff --git a/OpenGpxTracker/CLActivityType+Info.swift b/OpenGpxTracker/CLActivityType+Info.swift
new file mode 100644
index 00000000..aa6ad8db
--- /dev/null
+++ b/OpenGpxTracker/CLActivityType+Info.swift
@@ -0,0 +1,46 @@
+//
+//  CLActivityType+Info.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent on 22/7/19.
+//
+
+import CoreLocation
+
+/// Extends each CLActivityType to have a description and a name.
+extension CLActivityType {
+    
+    /// Returns the string name of this activty type
+    var name: String {
+        switch self {
+        case .other:                return "Automatic"
+        case .automotiveNavigation: return "Automotive navigation"
+        case .fitness:              return "Fitness"
+        case .otherNavigation:      return "Other navigation"
+        case .airborne:             return "Flight"
+        @unknown default:
+            return "Unknown Type: \(self.rawValue)"
+        }
+    }
+    
+    /// Returns a brief description of the purpose of the activty.
+    /// For example: for the activity type .other the description is
+    /// "System default. Automatically selects the mode"
+    var description: String {
+        switch self {
+        case .other:                return "System default. Automatically selects the mode"
+        case .automotiveNavigation: return "Car, motorbike, trucks..."
+        case .fitness:              return "Running, hiking, cycling..."
+        case .otherNavigation:      return "Other than automotive navigation"
+        case .airborne:             return "Airborne activities"
+        @unknown default:
+            return "Unknown Type: \(self.rawValue)"
+        }
+    }
+    
+    /// Number of activity types (5)
+    static var count: Int {
+        return 5
+    }
+    
+}
diff --git a/OpenGpxTracker/DistanceLabel.swift b/OpenGpxTracker/DistanceLabel.swift
index cff0de26..95867347 100644
--- a/OpenGpxTracker/DistanceLabel.swift
+++ b/OpenGpxTracker/DistanceLabel.swift
@@ -46,7 +46,7 @@ open class DistanceLabel: UILabel {
         }
         set {
             _distance = newValue
-            text = newValue.toDistance(useImperial: useImperial)
+            text = newValue.toDistance(useImperial: _useImperial)
         }
     }
 }
diff --git a/OpenGpxTracker/Preferences.swift b/OpenGpxTracker/Preferences.swift
index ec168790..ffeb75f3 100644
--- a/OpenGpxTracker/Preferences.swift
+++ b/OpenGpxTracker/Preferences.swift
@@ -9,6 +9,7 @@
 
 
 import Foundation
+import CoreLocation
 
 
 /// Key on Defaults for the Tile Server integer.
@@ -20,6 +21,9 @@ let kDefaultsKeyUseCache: String = "UseCache"
 /// Key on Defaults for the use of imperial units.
 let kDefaultsKeyUseImperial: String = "UseImperial"
 
+/// Key on Defaults for the current selected activity type.
+let kDefaultsKeyActivityType: String = "ActivityType"
+
 
 /// A class to handle app preferences in one single place.
 /// When the app starts for the first time the following preferences are set:
@@ -46,6 +50,9 @@ class Preferences: NSObject {
     /// In memory value of the preference.
     private var _tileServer: GPXTileServer = .apple
     
+    /// In memory value of the preference.
+    private var _activityType: CLActivityType = .other
+    
     /// UserDefaults.standard shortcut
     private let defaults = UserDefaults.standard
     
@@ -76,6 +83,12 @@ class Preferences: NSObject {
             _tileServer = GPXTileServer(rawValue: tileServerInt)!
             print("** Preferences:: loaded preference from defaults tileServerInt \(tileServerInt)")
         }
+        
+        // load previous activity type
+        if let activityTypeInt = defaults.object(forKey: kDefaultsKeyActivityType) as? Int {
+            _activityType = CLActivityType(rawValue: activityTypeInt)!
+            print("** Preferences:: loaded preference from defaults activityTypeInt \(activityTypeInt)")
+        }
     }
     
     /// If true, user prefers to display imperial units (miles, feets). Otherwise metric units
@@ -124,4 +137,25 @@ class Preferences: NSObject {
              defaults.set(newValue, forKey: kDefaultsKeyTileServerInt)
         }
     }
+    /// Gets and sets the type of activity preference
+    var locationActivityType: CLActivityType {
+        get {
+            return _activityType
+        }
+        set {
+            _activityType = newValue
+            defaults.set(newValue.rawValue, forKey: kDefaultsKeyActivityType)
+        }
+    }
+    
+    /// Gets and sets the activity type as its int value
+    var locationActivityTypeInt: Int {
+        get {
+            return _activityType.rawValue
+        }
+        set {
+            _activityType = CLActivityType(rawValue: newValue)!
+            defaults.set(newValue, forKey: kDefaultsKeyActivityType)
+        }
+    }
 }
diff --git a/OpenGpxTracker/PreferencesTableViewController.swift b/OpenGpxTracker/PreferencesTableViewController.swift
index 7b3ce9b7..9e78783a 100644
--- a/OpenGpxTracker/PreferencesTableViewController.swift
+++ b/OpenGpxTracker/PreferencesTableViewController.swift
@@ -7,6 +7,7 @@
 
 import Foundation
 import UIKit
+import CoreLocation
 
 import Cache
 
@@ -19,6 +20,9 @@ let kCacheSection = 1
 /// Map Source Section Id in PreferencesTableViewController
 let kMapSourceSection = 2
 
+/// Activity Type Section Id in PreferencesTableViewController
+let kActivityTypeSection = 3
+
 /// Cell Id of the Use Imperial units in UnitsSection
 let kUseImperialUnitsCell = 0
 
@@ -80,31 +84,34 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
     
     // MARK: - Table view data source
     
-    /// Returns 3 sections: Units, Cache, Map Source
+    /// Returns 4 sections: Units, Cache, Map Source, Activity Type
     override func numberOfSections(in tableView: UITableView?) -> Int {
         // Return the number of sections.
-        return 3
+        return 4
     }
     
     /// Returns the title of the existing sections.
-    /// Uses `kCacheSection`, `kUnitsSection`and `kMapSourceSection` for deciding which
-    /// is the section title
+    /// Uses `kCacheSection`, `kUnitsSection`, `kMapSourceSection` and `kActivityTypeSection`
+    /// for deciding which is the section title
     override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
         switch(section) {
         case kUnitsSection: return "Units"
         case kCacheSection: return "Cache"
         case kMapSourceSection: return "Map source"
+        case kActivityTypeSection: return "Activity Type"
         default: fatalError("Unknown section")
         }
     }
     
-    /// For section `kCacheSection` returns 2, `kUnitsSection` returns 1,
-    /// and for `kMapSourceSection` returns the number of tile servers defined in `GPXTileServer`
+    /// For section `kCacheSection` returns 2, for `kUnitsSection` returns 1,
+    /// for `kMapSourceSection` returns the number of tile servers defined in `GPXTileServer`,
+    /// and for kActivityTypeSection returns `CLActivityType.count`
     override func tableView(_ tableView: UITableView?, numberOfRowsInSection section: Int) -> Int {
         switch(section) {
         case kCacheSection: return 2
         case kUnitsSection: return 1
         case kMapSourceSection: return GPXTileServer.count
+        case kActivityTypeSection: return CLActivityType.count
         default: fatalError("Unknown section")
         }
     }
@@ -118,6 +125,9 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
     /// the tile server in the  `indexPath.row` index in `GPXTileServer`. The cell is marked
     /// if `selectedTileServerInt` is the same as `indexPath.row`.
     ///
+    /// If the section is kActivityTypeSection it returns a checkmark cell with the name
+    /// and description of the CLActivityType whose indexPath.row matches with the activity type.
+    ///
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         var cell = UITableViewCell(style: .value1, reuseIdentifier: "MapCell")
         
@@ -160,19 +170,41 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             if indexPath.row == preferences.tileServerInt {
                 cell.accessoryType = .checkmark
             }
-            
-            return cell
         }
+        
+        // Activity type section
+        if indexPath.section == kActivityTypeSection {
+            cell = UITableViewCell(style: .subtitle, reuseIdentifier: "ActivityCell")
+            let activity = CLActivityType(rawValue: indexPath.row + 1)!
+            cell.textLabel?.text = activity.name
+            cell.detailTextLabel?.text = activity.description
+            if indexPath.row + 1 == preferences.locationActivityTypeInt {
+                cell.accessoryType = .checkmark
+            }
+        }
+        
         return cell
     }
     
     /// Performs the following actions depending on the section and row selected:
-    /// 1. A cell in kCacheSection is selected:
-    ///     1. kUseOfflineCacheCell: Activates or desactivates the use of cache
-    ///        (`kDefaultUseCache` in defaults)
-    /// 2. A cell in kMapSourceSection is selected: Updates the default key (`kDefaultsKeyTileServerInt`)
+    /// If the cell `kUseImperialUnitCell` in `kUnitsSection`it sets or unsets the use of imperial
+    /// units (`useImperial` in `Preferences``and calls the delegate method `didUpdateUseImperial`.
+    ///
+    /// If a cell in kCacheSection is selected and the cell is
+    ///     1. kUseOfflineCacheCell: Activates or desactivates the `useCache` in `Preferences`,
+    ///        and calls the delegate method `didUpdateUseCache`
+    ///     2. KClearCacheCacheCell: Clears the current cache and calls
+    ///
+    /// If a cell in `kMapSourceSection` is selected: Updates `tileServerInt` in `Preferences` and
+    /// calls the delegate method `didUpdateTileServer`
+    ///
+    /// If a cell in `kActivitySection` is selected: Updates the `activityType` in `Preferences` and
+    /// calls the delegate method `didUpdateActivityType`.
+    ///
+    /// In each case checks or unchecks the corresponding cell in the UI.
     ///
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
         if indexPath.section == kUnitsSection {
             switch indexPath.row {
             case kUseImperialUnitsCell:
@@ -187,6 +219,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
                 fatalError("didSelectRowAt: Unknown cell")
             }
         }
+        
         if indexPath.section == kCacheSection {  // 0 -> sets and unsets cache
             switch indexPath.row {
             case kUseOfflineCacheCell:
@@ -224,6 +257,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
                 fatalError("didSelectRowAt: Unknown cell")
             }
         }
+        
         if indexPath.section == kMapSourceSection { // section 1 (sets tileServerInt in defaults
             print("PreferenccesTableView Map Tile Server section Row at index:  \(indexPath.row)")
             
@@ -239,6 +273,19 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             self.delegate?.didUpdateTileServer((indexPath as NSIndexPath).row)
         }
         
+        if indexPath.section == kActivityTypeSection {
+            print("PreferencesTableView Activity Type section Row at index:  \(indexPath.row + 1)")
+            let selected = IndexPath(row: preferences.locationActivityTypeInt - 1, section: indexPath.section)
+            
+            tableView.cellForRow(at: selected)?.accessoryType = .none
+            
+            //add checkmark to new tile server
+            tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
+            preferences.locationActivityTypeInt = indexPath.row + 1 // +1 as activityType raw value starts at index 1
+            
+            self.delegate?.didUpdateActivityType((indexPath as NSIndexPath).row + 1)
+        }
+        
         //unselect row
         tableView.deselectRow(at: indexPath, animated: true)
     }
diff --git a/OpenGpxTracker/PreferencesTableViewControllerDelegate.swift b/OpenGpxTracker/PreferencesTableViewControllerDelegate.swift
index 0121446a..830a5d13 100644
--- a/OpenGpxTracker/PreferencesTableViewControllerDelegate.swift
+++ b/OpenGpxTracker/PreferencesTableViewControllerDelegate.swift
@@ -19,7 +19,10 @@ protocol PreferencesTableViewControllerDelegate: class {
     /// User updated the usage of the caché
     func didUpdateUseCache(_ newUseCache: Bool)
     
-    /// User update the usage of imperial units
+    /// User updated the usage of imperial units
     func didUpdateUseImperial(_ newUseImperial: Bool)
     
+    /// User updated the activity type
+    func didUpdateActivityType(_ newActivityType: Int)
+    
 }
diff --git a/OpenGpxTracker/ViewController.swift b/OpenGpxTracker/ViewController.swift
index e8c9680d..63f77a74 100644
--- a/OpenGpxTracker/ViewController.swift
+++ b/OpenGpxTracker/ViewController.swift
@@ -102,7 +102,8 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
     let locationManager: CLLocationManager = {
         let manager = CLLocationManager()
         manager.requestAlwaysAuthorization()
-        
+        manager.activityType = CLActivityType(rawValue: Preferences.shared.locationActivityTypeInt)!
+        print("Chosen CLActivityType: \(manager.activityType.name)")
         manager.desiredAccuracy = kCLLocationAccuracyBest
         manager.distanceFilter = 2 //meters
         manager.headingFilter = 1 //degrees (1 is default)
@@ -411,7 +412,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         map.tileServer = Preferences.shared.tileServer
         map.useCache = Preferences.shared.useCache
         useImperial = Preferences.shared.useImperial
-        
+        //locationManager.activityType = Preferences.shared.locationActivityType
         
         //
         // Config user interface
@@ -471,13 +472,17 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         //tracked distance
         totalTrackedDistanceLabel.textAlignment = .right
         totalTrackedDistanceLabel.font = font36
-        totalTrackedDistanceLabel.text = 0.00.toDistance(useImperial: useImperial)
+        totalTrackedDistanceLabel.useImperial = useImperial
+        totalTrackedDistanceLabel.distance = 0.00
+        totalTrackedDistanceLabel.autoresizingMask = [.flexibleWidth, .flexibleLeftMargin, .flexibleRightMargin]
         //timeLabel.backgroundColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.5)
         map.addSubview(totalTrackedDistanceLabel)
         
         currentSegmentDistanceLabel.textAlignment = .right
         currentSegmentDistanceLabel.font = font18
-        currentSegmentDistanceLabel.text =  0.00.toDistance(useImperial: useImperial)
+        currentSegmentDistanceLabel.useImperial = useImperial
+        currentSegmentDistanceLabel.distance = 0.00
+        currentSegmentDistanceLabel.autoresizingMask = [.flexibleWidth, .flexibleLeftMargin, .flexibleRightMargin]
         //timeLabel.backgroundColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.5)
         map.addSubview(currentSegmentDistanceLabel)
         
@@ -1159,6 +1164,16 @@ extension ViewController: StopWatchDelegate {
 // MARK: PreferencesTableViewControllerDelegate
 
 extension ViewController: PreferencesTableViewControllerDelegate {
+    
+    /// Update the activity type that the location manager is using.
+    ///
+    /// When user changes the activity type in preferences, this function is invoked to update the activity type of the location manager.
+    ///
+    func didUpdateActivityType(_ newActivityType: Int) {
+        print("PreferencesTableViewControllerDelegate:: didUpdateActivityType: \(newActivityType)")
+        self.locationManager.activityType = CLActivityType(rawValue: newActivityType)!
+    }
+    
     ///
     /// Updates the `tileServer` the map is using.
     ///
