diff --git a/OpenGpxTracker-Watch Extension/GPXFileTableInterfaceController.swift b/OpenGpxTracker-Watch Extension/GPXFileTableInterfaceController.swift
index 32b65db3..986647da 100644
--- a/OpenGpxTracker-Watch Extension/GPXFileTableInterfaceController.swift	
+++ b/OpenGpxTracker-Watch Extension/GPXFileTableInterfaceController.swift	
@@ -10,7 +10,7 @@ import WatchKit
 import WatchConnectivity
 
 /// Text displayed when there are no GPX files in the folder.
-let kNoFiles = "No gpx files"
+let kNoFiles = NSLocalizedString("NO_FILES", comment: "no comment")
 
 ///
 /// WKInterfaceTable that displays the list of files that have been saved in previous sessions.
@@ -107,14 +107,14 @@ class GPXFileTableInterfaceController: WKInterfaceController {
     func updateProgressIndicators(status: sendingStatus, fileName: String?) {
         switch status {
         case .sending:
-            progressTitle.setText("Sending:")
+            progressTitle.setText(NSLocalizedString("SENDING", comment: "no comment"))
             guard let fileName = fileName else { return }
             
             /// count of pending files, does not seem to include the current one
             let fileTransfersCount = session?.outstandingFileTransfers.count ?? 0
             // if there are files pending for sending, filename will not be displayed with the name of file.
             if fileTransfersCount >= 1 {
-                progressFileName.setText("\(fileTransfersCount + 1) files")
+                progressFileName.setText(String(format: NSLocalizedString("X_FILES", comment: "no comment"), fileTransfersCount + 1))
             }
             else {
                 progressFileName.setText(fileName)
@@ -123,13 +123,13 @@ class GPXFileTableInterfaceController: WKInterfaceController {
         case .success:
             progressImageView.stopAnimating()
             progressImageView.setImage(UIImage(named: "Progress-success"))
-            progressTitle.setText("Sucessfully sent:")
+            progressTitle.setText(NSLocalizedString("SUCCESSFULLY_SENT", comment: "no comment"))
             hideProgressIndicatorsWithAnimation()
             
         case .failure:
             progressImageView.stopAnimating()
             progressImageView.setImage(UIImage(named: "Progress-failure"))
-            progressTitle.setText("Failed to send:")
+            progressTitle.setText(NSLocalizedString("FAILED_TO_SEND", comment: "no comment"))
             hideProgressIndicatorsWithAnimation()
         }
     }
@@ -139,7 +139,7 @@ class GPXFileTableInterfaceController: WKInterfaceController {
         super.willActivate()
         print("GPXFileTableInterfaceController:: willActivate willSendFile: \(willSendFile)")
         
-        self.setTitle("Your files")
+        self.setTitle(NSLocalizedString("YOUR_FILES", comment: "no comment"))
         session?.delegate = self
         
         if willSendFile == true {
@@ -199,17 +199,17 @@ class GPXFileTableInterfaceController: WKInterfaceController {
         if gpxFilesFound {
             
             /// Option lets user send selected file to iOS app
-            let shareOption = WKAlertAction(title: "Send to iOS app", style: .default) {
+            let shareOption = WKAlertAction(title: NSLocalizedString("SEND_TO_IOS", comment: "no comment"), style: .default) {
                 self.actionTransferFileAtIndex(rowIndex)
             }
             
             /// Option for users to cancel
-            let cancelOption = WKAlertAction(title: "Cancel", style: .cancel) {
+            let cancelOption = WKAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) {
                 self.actionSheetCancel()
             }
             
             /// Option to delete selected file
-            let deleteOption = WKAlertAction(title: "Delete", style: .destructive) {
+            let deleteOption = WKAlertAction(title: NSLocalizedString("DELETE", comment: "no comment"), style: .destructive) {
                 self.actionDeleteFileAtIndex(rowIndex)
                 self.loadTableData()
             }
@@ -217,7 +217,7 @@ class GPXFileTableInterfaceController: WKInterfaceController {
             /// Array of all available options
             let options = [shareOption, cancelOption, deleteOption]
             
-            presentAlert(withTitle: "GPX file selected", message: "What would you like to do?", preferredStyle: .actionSheet, actions: options)
+            presentAlert(withTitle: NSLocalizedString("FILE_SELECTED_TITLE", comment: "no comment"), message: NSLocalizedString("FILE_SELECTED_MESSAGE", comment: "no comment"), preferredStyle: .actionSheet, actions: options)
         }
     }
     
@@ -295,7 +295,7 @@ extension GPXFileTableInterfaceController: WCSessionDelegate {
     }
     
     func session(_ session: WCSession, didFinish fileTransfer: WCSessionFileTransfer, error: Error?) {
-        let doneAction = WKAlertAction(title: "Done", style: .default) { }
+        let doneAction = WKAlertAction(title: NSLocalizedString("DONE", comment: "no comment"), style: .default) { }
         guard let error = error else {
             print("WCSession: didFinish fileTransfer: \(fileTransfer.file.fileURL.absoluteString)")
             // presenting success indicator to user if file is successfully transferred
@@ -310,8 +310,9 @@ extension GPXFileTableInterfaceController: WCSessionDelegate {
         self.updateProgressIndicators(status: .failure, fileName: nil)
         
         // presents alert after 1.5s, with error message
+        // MARK: "as CVarArg" was suggested by XCode and my intruduce a bug...
         DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
-            self.presentAlert(withTitle: "Error Occurred", message: "GPX file was unsuccessfully sent to iOS app, due to \(error) ", preferredStyle: .alert, actions: [doneAction])
+            self.presentAlert(withTitle: NSLocalizedString("ERROR_OCCURED_TITLE", comment: "no comment"), message: String(format: NSLocalizedString("ERROR_OCCURED_MESSAGE", comment: "no comment"), error as CVarArg), preferredStyle: .alert, actions: [doneAction])
         }
     }
     
diff --git a/OpenGpxTracker-Watch Extension/Info.plist b/OpenGpxTracker-Watch Extension/Info.plist
index c361343a..dd6f136f 100644
--- a/OpenGpxTracker-Watch Extension/Info.plist	
+++ b/OpenGpxTracker-Watch Extension/Info.plist	
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>XPC!</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.6.1</string>
+	<string>1.6.2</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>CLKComplicationPrincipalClass</key>
diff --git a/OpenGpxTracker-Watch Extension/InterfaceController.swift b/OpenGpxTracker-Watch Extension/InterfaceController.swift
index 0645ea33..93498d23 100644
--- a/OpenGpxTracker-Watch Extension/InterfaceController.swift	
+++ b/OpenGpxTracker-Watch Extension/InterfaceController.swift	
@@ -24,7 +24,7 @@ let kWhiteBackgroundColor: UIColor = UIColor(red: 254.0/255.0, green: 254.0/255.
 let kDeleteWaypointAccesoryButtonTag = 666
 let kEditWaypointAccesoryButtonTag = 333
 
-let kNotGettingLocationText = "Not getting location"
+let kNotGettingLocationText = NSLocalizedString("NO_LOCATION", comment: "no comment")
 let kUnknownAccuracyText = "±···"
 let kUnknownSpeedText = "·.··"
 let kUnknownAltitudeText = "···"
@@ -130,7 +130,7 @@ class InterfaceController: WKInterfaceController {
             case .notStarted:
                 print("switched to non started")
                 // set Tracker button to allow Start
-                trackerButton.setTitle("Start Tracking")
+                trackerButton.setTitle(NSLocalizedString("START_TRACKING", comment: "no comment"))
                 trackerButton.setBackgroundColor(kGreenButtonBackgroundColor)
                 //save & reset button to transparent.
                 saveButton.setBackgroundColor(kDisabledBlueButtonBackgroundColor)
@@ -159,7 +159,7 @@ class InterfaceController: WKInterfaceController {
             case .tracking:
                 print("switched to tracking mode")
                 // set trackerButton to allow Pause
-                trackerButton.setTitle("Pause")
+                trackerButton.setTitle(NSLocalizedString("PAUSE", comment: "no comment"))
                 trackerButton.setBackgroundColor(kPurpleButtonBackgroundColor)
                 //activate save & reset buttons
                 saveButton.setBackgroundColor(kBlueButtonBackgroundColor)
@@ -170,7 +170,7 @@ class InterfaceController: WKInterfaceController {
             case .paused:
                 print("switched to paused mode")
                 // set trackerButton to allow Resume
-                self.trackerButton.setTitle("Resume")
+                self.trackerButton.setTitle(NSLocalizedString("RESUME", comment: "no comment"))
                 self.trackerButton.setBackgroundColor(kGreenButtonBackgroundColor)
                 // activate save & reset (just in case switched from .NotStarted)
                 saveButton.setBackgroundColor(kBlueButtonBackgroundColor)
@@ -213,7 +213,7 @@ class InterfaceController: WKInterfaceController {
         // This method is called when watch view controller is about to be visible to user
          print("InterfaceController:: willActivate")
         super.willActivate()
-        self.setTitle("GPX Tracker")
+        self.setTitle(NSLocalizedString("GPX_TRACKER", comment: "no comment"))
         
         stopWatch.delegate = self
         
@@ -291,7 +291,7 @@ class InterfaceController: WKInterfaceController {
         /// Just a 'done' button, without
         let action = WKAlertAction(title: "Done", style: .default) {}
         
-        presentAlert(withTitle: "GPX file saved", message: "Current session saved as \(filename).gpx ", preferredStyle: .alert, actions: [action])
+        presentAlert(withTitle: NSLocalizedString("FILE_SAVED_TITLE", comment: "no comment"), message: String(format: NSLocalizedString("NO_LOCATION", comment: "no comment"), filename), preferredStyle: .alert, actions: [action])
         
     }
     
@@ -345,7 +345,7 @@ class InterfaceController: WKInterfaceController {
             print("LocationServicesDisabledAlert: cancel pressed")
         }
         
-        presentAlert(withTitle: "Location services disabled", message: "Go to settings and enable location", preferredStyle: .alert, actions: [button])
+        presentAlert(withTitle: NSLocalizedString("LOCATION_SERVICES_DISABLED", comment: "no comment"), message: NSLocalizedString("ENABLE_LOCATION_SERVICES", comment: "no comment"), preferredStyle: .alert, actions: [button])
     }
     
     
@@ -361,7 +361,7 @@ class InterfaceController: WKInterfaceController {
             print("LocationServicesDeniedAlert: cancel pressed")
         }
         
-        presentAlert(withTitle: "Access to location denied", message: "On Location settings, allow always access to location for GPX Tracker", preferredStyle: .alert, actions: [button])
+        presentAlert(withTitle: NSLocalizedString("ACCESS_TO_LOCATION_DENIED", comment: "no comment"), message: NSLocalizedString("ALLOW_LOCATION", comment: "no comment"), preferredStyle: .alert, actions: [button])
     }
 
 }
diff --git a/OpenGpxTracker-Watch Extension/de.lproj/InfoPlist.strings b/OpenGpxTracker-Watch Extension/de.lproj/InfoPlist.strings
new file mode 100644
index 00000000..2334689e
--- /dev/null
+++ b/OpenGpxTracker-Watch Extension/de.lproj/InfoPlist.strings	
@@ -0,0 +1,12 @@
+/* 
+  InfoPlist.strings
+  OpenGpxTracker
+
+  Created by Kurt Höblinger on 21.08.19.
+  Copyright © 2019 TransitBox. All rights reserved.
+*/
+
+NSLocationAlwaysAndWhenInUseUsageDescription = "Der Standort wird nur verwendet um Deinen Standort auf der Karte darzustellen. Wähle \"immer\" um den Standort auch zur Verfügung zu stellen, wenn die App im Hintergrund arbeitet.";
+NSLocationAlwaysUsageDescription = "Dein Standort wird benötigt um den GPX Track auch dann zu erstellen, wenn die App im Hintergrund läuft.";
+NSLocationUsageDescription = "Der Standort wird zum Erstellen von GPX und zum Hinzufügen von PDIs benötigt";
+NSLocationWhenInUseUsageDescription = "Der Standort wird verwendet um auf der Karte zu zeigen wo Du bist und wo Du warst.";
diff --git a/OpenGpxTracker-Watch Extension/en.lproj/InfoPlist.strings b/OpenGpxTracker-Watch Extension/en.lproj/InfoPlist.strings
new file mode 100644
index 00000000..5a2429b2
--- /dev/null
+++ b/OpenGpxTracker-Watch Extension/en.lproj/InfoPlist.strings	
@@ -0,0 +1,12 @@
+/* 
+  InfoPlist.strings
+  OpenGpxTracker
+
+  Created by Kurt Höblinger on 21.08.19.
+  Copyright © 2019 TransitBox. All rights reserved.
+*/
+
+NSLocationAlwaysAndWhenInUseUsageDescription = "Location is only used to display your position in the map. Select always allow to track your position even if the app is in background.";
+NSLocationAlwaysUsageDescription = "Location is needed to create the GPX track even when the app is in background.";
+NSLocationUsageDescription = "Location is needed to create GPX and add PDIs";
+NSLocationWhenInUseUsageDescription = "Location is used to display in the map where are you and where you have been.";
diff --git a/OpenGpxTracker-Watch/Interface.storyboard b/OpenGpxTracker-Watch/Base.lproj/Interface.storyboard
similarity index 100%
rename from OpenGpxTracker-Watch/Interface.storyboard
rename to OpenGpxTracker-Watch/Base.lproj/Interface.storyboard
diff --git a/OpenGpxTracker-Watch/Info.plist b/OpenGpxTracker-Watch/Info.plist
index 8f05fc63..6b806441 100644
--- a/OpenGpxTracker-Watch/Info.plist
+++ b/OpenGpxTracker-Watch/Info.plist
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.6.1</string>
+	<string>1.6.2</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>UISupportedInterfaceOrientations</key>
diff --git a/OpenGpxTracker-Watch/de.lproj/Interface.strings b/OpenGpxTracker-Watch/de.lproj/Interface.strings
new file mode 100644
index 00000000..e59935f1
--- /dev/null
+++ b/OpenGpxTracker-Watch/de.lproj/Interface.strings
@@ -0,0 +1,54 @@
+
+/* Class = "WKInterfaceLabel"; text = "signal"; ObjectID = "036-aK-7ie"; */
+"036-aK-7ie.text" = "Signal";
+
+/* Class = "WKInterfaceLabel"; text = "File Name"; ObjectID = "19n-8m-sC5"; */
+"19n-8m-sC5.text" = "Dateiname";
+
+/* Class = "WKInterfaceButton"; title = "View files"; ObjectID = "3QZ-4r-aMl"; */
+"3QZ-4r-aMl.title" = "Zeige Dateien";
+
+/* Class = "WKInterfaceLabel"; text = "speedValue"; ObjectID = "9yb-Lf-Jav"; */
+"9yb-Lf-Jav.text" = "speedValue";
+
+/* Class = "WKInterfaceLabel"; text = "Sending"; ObjectID = "AyV-DL-c4U"; */
+"AyV-DL-c4U.text" = "Sende";
+
+/* Class = "WKInterfaceLabel"; text = "SPEED"; ObjectID = "HGe-sg-FuE"; */
+"HGe-sg-FuE.text" = "GESCHWINDIGKEIT";
+
+/* Class = "WKInterfaceLabel"; text = "Alert Label"; ObjectID = "IdU-wH-bcW"; */
+"IdU-wH-bcW.text" = "Alert Label";
+
+/* Class = "WKInterfaceLabel"; text = "File Name"; ObjectID = "Nkq-Do-muS"; */
+"Nkq-Do-muS.text" = "Dateiname";
+
+/* Class = "WKInterfaceButton"; title = "Save"; ObjectID = "Yfa-t6-5Sk"; */
+"Yfa-t6-5Sk.title" = "Sichern";
+
+/* Class = "WKInterfaceButton"; title = "Start Tracking"; ObjectID = "exN-t4-It8"; */
+"exN-t4-It8.title" = "Tracking starten";
+
+/* Class = "WKInterfaceLabel"; text = "altValue"; ObjectID = "fLq-xh-Kj9"; */
+"fLq-xh-Kj9.text" = "altValue";
+
+/* Class = "WKInterfaceButton"; title = "Reset"; ObjectID = "g9l-MU-PFV"; */
+"g9l-MU-PFV.title" = "Löschen";
+
+/* Class = "WKInterfaceLabel"; text = "0m"; ObjectID = "jLD-vq-fZz"; */
+"jLD-vq-fZz.text" = "0m";
+
+/* Class = "WKInterfaceButton"; title = "📍"; ObjectID = "juk-oB-3vV"; */
+"juk-oB-3vV.title" = "📍";
+
+/* Class = "WKInterfaceLabel"; text = "COORDINATES"; ObjectID = "oWE-ge-dGJ"; */
+"oWE-ge-dGJ.text" = "KOORDINATEN";
+
+/* Class = "WKInterfaceLabel"; text = "ALTITUDE"; ObjectID = "sOz-Cs-O7T"; */
+"sOz-Cs-O7T.text" = "HÖHE";
+
+/* Class = "WKInterfaceLabel"; text = "coordinatesLabel"; ObjectID = "wh4-cP-QA2"; */
+"wh4-cP-QA2.text" = "coordinatesLabel";
+
+/* Class = "WKInterfaceLabel"; text = "00:00"; ObjectID = "xum-IR-If6"; */
+"xum-IR-If6.text" = "00:00";
diff --git a/OpenGpxTracker-Watch/en.lproj/Interface.strings b/OpenGpxTracker-Watch/en.lproj/Interface.strings
new file mode 100644
index 00000000..4c6df169
--- /dev/null
+++ b/OpenGpxTracker-Watch/en.lproj/Interface.strings
@@ -0,0 +1,54 @@
+
+/* Class = "WKInterfaceLabel"; text = "signal"; ObjectID = "036-aK-7ie"; */
+"036-aK-7ie.text" = "signal";
+
+/* Class = "WKInterfaceLabel"; text = "File Name"; ObjectID = "19n-8m-sC5"; */
+"19n-8m-sC5.text" = "File Name";
+
+/* Class = "WKInterfaceButton"; title = "View files"; ObjectID = "3QZ-4r-aMl"; */
+"3QZ-4r-aMl.title" = "View files";
+
+/* Class = "WKInterfaceLabel"; text = "speedValue"; ObjectID = "9yb-Lf-Jav"; */
+"9yb-Lf-Jav.text" = "speedValue";
+
+/* Class = "WKInterfaceLabel"; text = "Sending"; ObjectID = "AyV-DL-c4U"; */
+"AyV-DL-c4U.text" = "Sending";
+
+/* Class = "WKInterfaceLabel"; text = "SPEED"; ObjectID = "HGe-sg-FuE"; */
+"HGe-sg-FuE.text" = "SPEED";
+
+/* Class = "WKInterfaceLabel"; text = "Alert Label"; ObjectID = "IdU-wH-bcW"; */
+"IdU-wH-bcW.text" = "Alert Label";
+
+/* Class = "WKInterfaceLabel"; text = "File Name"; ObjectID = "Nkq-Do-muS"; */
+"Nkq-Do-muS.text" = "File Name";
+
+/* Class = "WKInterfaceButton"; title = "Save"; ObjectID = "Yfa-t6-5Sk"; */
+"Yfa-t6-5Sk.title" = "Save";
+
+/* Class = "WKInterfaceButton"; title = "Start Tracking"; ObjectID = "exN-t4-It8"; */
+"exN-t4-It8.title" = "Start Tracking";
+
+/* Class = "WKInterfaceLabel"; text = "altValue"; ObjectID = "fLq-xh-Kj9"; */
+"fLq-xh-Kj9.text" = "altValue";
+
+/* Class = "WKInterfaceButton"; title = "Reset"; ObjectID = "g9l-MU-PFV"; */
+"g9l-MU-PFV.title" = "Reset";
+
+/* Class = "WKInterfaceLabel"; text = "0m"; ObjectID = "jLD-vq-fZz"; */
+"jLD-vq-fZz.text" = "0m";
+
+/* Class = "WKInterfaceButton"; title = "📍"; ObjectID = "juk-oB-3vV"; */
+"juk-oB-3vV.title" = "📍";
+
+/* Class = "WKInterfaceLabel"; text = "COORDINATES"; ObjectID = "oWE-ge-dGJ"; */
+"oWE-ge-dGJ.text" = "COORDINATES";
+
+/* Class = "WKInterfaceLabel"; text = "ALTITUDE"; ObjectID = "sOz-Cs-O7T"; */
+"sOz-Cs-O7T.text" = "ALTITUDE";
+
+/* Class = "WKInterfaceLabel"; text = "coordinatesLabel"; ObjectID = "wh4-cP-QA2"; */
+"wh4-cP-QA2.text" = "coordinatesLabel";
+
+/* Class = "WKInterfaceLabel"; text = "00:00"; ObjectID = "xum-IR-If6"; */
+"xum-IR-If6.text" = "00:00";
diff --git a/OpenGpxTracker.xcodeproj/project.pbxproj b/OpenGpxTracker.xcodeproj/project.pbxproj
index 086c8877..17f70993 100644
--- a/OpenGpxTracker.xcodeproj/project.pbxproj
+++ b/OpenGpxTracker.xcodeproj/project.pbxproj
@@ -9,12 +9,14 @@
 /* Begin PBXBuildFile section */
 		120DE71022B274910055C4CB /* GPXSession.swift in Sources */ = {isa = PBXBuildFile; fileRef = 120DE70F22B274910055C4CB /* GPXSession.swift */; };
 		120DE71122B274A40055C4CB /* GPXSession.swift in Sources */ = {isa = PBXBuildFile; fileRef = 120DE70F22B274910055C4CB /* GPXSession.swift */; };
+		4588D66B230D23AA009AA75F /* InfoPlist.strings in Resources */ = {isa = PBXBuildFile; fileRef = 4588D66D230D23AA009AA75F /* InfoPlist.strings */; };
+		4588D66F230D2476009AA75F /* InfoPlist.strings in Resources */ = {isa = PBXBuildFile; fileRef = 4588D671230D2476009AA75F /* InfoPlist.strings */; };
+		45A430E9230D61240003C2F2 /* Interface.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 45A430EB230D61240003C2F2 /* Interface.storyboard */; };
 		8913CD8C1BDB0A9A009EC729 /* PreferencesTableViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8913CD8B1BDB0A9A009EC729 /* PreferencesTableViewController.swift */; };
 		8913CD8E1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8913CD8D1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift */; };
 		893BEC8619C7E62200C77354 /* GPXWaypoint+MKAnnotation.swift in Sources */ = {isa = PBXBuildFile; fileRef = 893BEC8519C7E62200C77354 /* GPXWaypoint+MKAnnotation.swift */; };
 		8945A90E1DC7AD040029AF93 /* CachedTileOverlay.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8945A90D1DC7AD040029AF93 /* CachedTileOverlay.swift */; };
 		895F04B41A7143CE004BEE8A /* GPXExtentCoordinates.swift in Sources */ = {isa = PBXBuildFile; fileRef = 895F04B31A7143CE004BEE8A /* GPXExtentCoordinates.swift */; };
-		89686CC52218793A0050D1E3 /* Interface.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = BF692F7C2209AB8000F8A0F1 /* Interface.storyboard */; };
 		898EECDB19C49B5800B4B207 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 898EECDA19C49B5800B4B207 /* AppDelegate.swift */; };
 		898EECDE19C49B5800B4B207 /* OpenGpxTracker.xcdatamodeld in Sources */ = {isa = PBXBuildFile; fileRef = 898EECDC19C49B5800B4B207 /* OpenGpxTracker.xcdatamodeld */; };
 		898EECE019C49B5800B4B207 /* ViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 898EECDF19C49B5800B4B207 /* ViewController.swift */; };
@@ -49,6 +51,7 @@
 		89F6E107227F7C5500CE3E3F /* Double+Measures.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F6E0F4227D091900CE3E3F /* Double+Measures.swift */; };
 		89F8749B1BBB7362004EF41A /* GPXTrack+length.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F8749A1BBB7362004EF41A /* GPXTrack+length.swift */; };
 		89F8749F1BBCB97F004EF41A /* GPXRoot+length.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89F8749E1BBCB97F004EF41A /* GPXRoot+length.swift */; };
+		BF1FE1A8230EAB9B00404B59 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = BF1FE1A6230EAB9B00404B59 /* Localizable.strings */; };
 		BF214B5E22E60B64000E96AA /* CLActivityType+Info.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */; };
 		BF36B6CF220C6F1A00077B46 /* GPXFileTableInterfaceController.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF36B6CE220C6F1A00077B46 /* GPXFileTableInterfaceController.swift */; };
 		BF5C0D2222D771D1001C2575 /* DistanceLabel.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF5C0D2122D771D0001C2575 /* DistanceLabel.swift */; };
@@ -134,6 +137,15 @@
 		120DE70F22B274910055C4CB /* GPXSession.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = GPXSession.swift; sourceTree = "<group>"; };
 		3C3088B74A230AB1FD234D6F /* Pods_OpenGpxTracker_Watch.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_OpenGpxTracker_Watch.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		44E786790A477838358F948C /* Pods-OpenGpxTracker.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-OpenGpxTracker.debug.xcconfig"; path = "Pods/Target Support Files/Pods-OpenGpxTracker/Pods-OpenGpxTracker.debug.xcconfig"; sourceTree = "<group>"; };
+		455F78BD230AA9D6003705F3 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/LaunchScreen.strings; sourceTree = "<group>"; };
+		455F78BE230AA9D6003705F3 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/Main.strings; sourceTree = "<group>"; };
+		4588D66C230D23AA009AA75F /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		4588D66E230D23AD009AA75F /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		4588D670230D2476009AA75F /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		4588D672230D247C009AA75F /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/InfoPlist.strings; sourceTree = "<group>"; };
+		45A430EA230D61240003C2F2 /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/Interface.storyboard; sourceTree = "<group>"; };
+		45A430ED230D612B0003C2F2 /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/Interface.strings; sourceTree = "<group>"; };
+		45A430EF230D612C0003C2F2 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/Interface.strings; sourceTree = "<group>"; };
 		554D2ACAA71E64B3A7B1A167 /* Pods-OpenGpxTracker-Watch Extension.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-OpenGpxTracker-Watch Extension.release.xcconfig"; path = "Pods/Target Support Files/Pods-OpenGpxTracker-Watch Extension/Pods-OpenGpxTracker-Watch Extension.release.xcconfig"; sourceTree = "<group>"; };
 		8913CD8B1BDB0A9A009EC729 /* PreferencesTableViewController.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = PreferencesTableViewController.swift; sourceTree = "<group>"; };
 		8913CD8D1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = PreferencesTableViewControllerDelegate.swift; sourceTree = "<group>"; };
@@ -181,6 +193,8 @@
 		93BD71BED3CF101C2B01ABDA /* Pods_OpenGpxTracker_Watch_Extension.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_OpenGpxTracker_Watch_Extension.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		95D28CBDF2535138FA16F38F /* Pods_OpenGpxTracker.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_OpenGpxTracker.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		9A591D96EF2E41D00AD6BB3A /* Pods-OpenGpxTracker-Watch Extension.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-OpenGpxTracker-Watch Extension.debug.xcconfig"; path = "Pods/Target Support Files/Pods-OpenGpxTracker-Watch Extension/Pods-OpenGpxTracker-Watch Extension.debug.xcconfig"; sourceTree = "<group>"; };
+		BF1FE1A7230EAB9B00404B59 /* en */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = en; path = en.lproj/Localizable.strings; sourceTree = "<group>"; };
+		BF1FE1A9230EABA500404B59 /* de */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = de; path = de.lproj/Localizable.strings; sourceTree = "<group>"; };
 		BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "CLActivityType+Info.swift"; sourceTree = "<group>"; };
 		BF36B6CE220C6F1A00077B46 /* GPXFileTableInterfaceController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = GPXFileTableInterfaceController.swift; sourceTree = "<group>"; };
 		BF5C0D2122D771D0001C2575 /* DistanceLabel.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = DistanceLabel.swift; sourceTree = "<group>"; };
@@ -196,7 +210,6 @@
 		BF692F6C2209AAFB00F8A0F1 /* ComplicationController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ComplicationController.swift; sourceTree = "<group>"; };
 		BF692F6E2209AAFC00F8A0F1 /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
 		BF692F702209AAFC00F8A0F1 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
-		BF692F7C2209AB8000F8A0F1 /* Interface.storyboard */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = file.storyboard; path = Interface.storyboard; sourceTree = "<group>"; };
 		BFA3CE8122C23B2300A8B965 /* CoreDataAlertView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CoreDataAlertView.swift; sourceTree = "<group>"; };
 		DC4908A01C4A136F009484AE /* TrackerButton.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = TrackerButton.swift; sourceTree = "<group>"; };
 		DC4908A21C4A332E009484AE /* MapViewDelegate.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = MapViewDelegate.swift; sourceTree = "<group>"; };
@@ -309,6 +322,7 @@
 			isa = PBXGroup;
 			children = (
 				898EECD919C49B5800B4B207 /* Info.plist */,
+				4588D66D230D23AA009AA75F /* InfoPlist.strings */,
 			);
 			name = "Supporting Files";
 			sourceTree = "<group>";
@@ -377,6 +391,7 @@
 		899D678919D3551300C88A0A /* Resources */ = {
 			isa = PBXGroup;
 			children = (
+				BF1FE1A6230EAB9B00404B59 /* Localizable.strings */,
 				898EECE619C49B5800B4B207 /* LaunchScreen.xib */,
 				898EECE119C49B5800B4B207 /* Main.storyboard */,
 				898EECDA19C49B5800B4B207 /* AppDelegate.swift */,
@@ -411,7 +426,7 @@
 		BF692F582209AAF900F8A0F1 /* OpenGpxTracker-Watch */ = {
 			isa = PBXGroup;
 			children = (
-				BF692F7C2209AB8000F8A0F1 /* Interface.storyboard */,
+				45A430EB230D61240003C2F2 /* Interface.storyboard */,
 				BF692F5C2209AAFB00F8A0F1 /* Assets.xcassets */,
 				BF692F5E2209AAFB00F8A0F1 /* Info.plist */,
 			);
@@ -428,6 +443,7 @@
 				BF692F6C2209AAFB00F8A0F1 /* ComplicationController.swift */,
 				BF692F6E2209AAFC00F8A0F1 /* Assets.xcassets */,
 				BF692F702209AAFC00F8A0F1 /* Info.plist */,
+				4588D671230D2476009AA75F /* InfoPlist.strings */,
 			);
 			path = "OpenGpxTracker-Watch Extension";
 			sourceTree = "<group>";
@@ -525,7 +541,7 @@
 				TargetAttributes = {
 					898EECD419C49B5800B4B207 = {
 						CreatedOnToolsVersion = 6.0;
-						DevelopmentTeam = SU64RGPDF2;
+						DevelopmentTeam = PMCWJ9GD9L;
 						LastSwiftMigration = 1020;
 						SystemCapabilities = {
 							com.apple.BackgroundModes = {
@@ -535,16 +551,18 @@
 					};
 					898EECEC19C49B5800B4B207 = {
 						CreatedOnToolsVersion = 6.0;
+						DevelopmentTeam = PMCWJ9GD9L;
 						LastSwiftMigration = 1020;
 						TestTargetID = 898EECD419C49B5800B4B207;
 					};
 					BF692F562209AAF900F8A0F1 = {
 						CreatedOnToolsVersion = 10.1;
+						DevelopmentTeam = PMCWJ9GD9L;
 						ProvisioningStyle = Automatic;
 					};
 					BF692F622209AAFB00F8A0F1 = {
 						CreatedOnToolsVersion = 10.1;
-						DevelopmentTeam = SU64RGPDF2;
+						DevelopmentTeam = PMCWJ9GD9L;
 						LastSwiftMigration = 1020;
 						ProvisioningStyle = Automatic;
 						SystemCapabilities = {
@@ -562,6 +580,7 @@
 			knownRegions = (
 				en,
 				Base,
+				de,
 			);
 			mainGroup = 898EECCC19C49B5800B4B207;
 			productRefGroup = 898EECD619C49B5800B4B207 /* Products */;
@@ -581,6 +600,8 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				4588D66B230D23AA009AA75F /* InfoPlist.strings in Resources */,
+				BF1FE1A8230EAB9B00404B59 /* Localizable.strings in Resources */,
 				899D678719D354EB00C88A0A /* about.html in Resources */,
 				898EECE319C49B5800B4B207 /* Main.storyboard in Resources */,
 				898EECE819C49B5800B4B207 /* LaunchScreen.xib in Resources */,
@@ -599,7 +620,7 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				89686CC52218793A0050D1E3 /* Interface.storyboard in Resources */,
+				45A430E9230D61240003C2F2 /* Interface.storyboard in Resources */,
 				BF692F5D2209AAFB00F8A0F1 /* Assets.xcassets in Resources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -609,6 +630,7 @@
 			buildActionMask = 2147483647;
 			files = (
 				BF692F6F2209AAFC00F8A0F1 /* Assets.xcassets in Resources */,
+				4588D66F230D2476009AA75F /* InfoPlist.strings in Resources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
@@ -806,10 +828,39 @@
 /* End PBXTargetDependency section */
 
 /* Begin PBXVariantGroup section */
+		4588D66D230D23AA009AA75F /* InfoPlist.strings */ = {
+			isa = PBXVariantGroup;
+			children = (
+				4588D66C230D23AA009AA75F /* de */,
+				4588D66E230D23AD009AA75F /* en */,
+			);
+			name = InfoPlist.strings;
+			sourceTree = "<group>";
+		};
+		4588D671230D2476009AA75F /* InfoPlist.strings */ = {
+			isa = PBXVariantGroup;
+			children = (
+				4588D670230D2476009AA75F /* en */,
+				4588D672230D247C009AA75F /* de */,
+			);
+			name = InfoPlist.strings;
+			sourceTree = "<group>";
+		};
+		45A430EB230D61240003C2F2 /* Interface.storyboard */ = {
+			isa = PBXVariantGroup;
+			children = (
+				45A430EA230D61240003C2F2 /* Base */,
+				45A430ED230D612B0003C2F2 /* en */,
+				45A430EF230D612C0003C2F2 /* de */,
+			);
+			name = Interface.storyboard;
+			sourceTree = "<group>";
+		};
 		898EECE119C49B5800B4B207 /* Main.storyboard */ = {
 			isa = PBXVariantGroup;
 			children = (
 				898EECE219C49B5800B4B207 /* Base */,
+				455F78BE230AA9D6003705F3 /* de */,
 			);
 			name = Main.storyboard;
 			sourceTree = "<group>";
@@ -818,10 +869,20 @@
 			isa = PBXVariantGroup;
 			children = (
 				898EECE719C49B5800B4B207 /* Base */,
+				455F78BD230AA9D6003705F3 /* de */,
 			);
 			name = LaunchScreen.xib;
 			sourceTree = "<group>";
 		};
+		BF1FE1A6230EAB9B00404B59 /* Localizable.strings */ = {
+			isa = PBXVariantGroup;
+			children = (
+				BF1FE1A7230EAB9B00404B59 /* en */,
+				BF1FE1A9230EABA500404B59 /* de */,
+			);
+			name = Localizable.strings;
+			sourceTree = "<group>";
+		};
 /* End PBXVariantGroup section */
 
 /* Begin XCBuildConfiguration section */
@@ -938,7 +999,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
-				DEVELOPMENT_TEAM = SU64RGPDF2;
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				INFOPLIST_FILE = OpenGpxTracker/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
@@ -958,7 +1019,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
-				DEVELOPMENT_TEAM = SU64RGPDF2;
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				INFOPLIST_FILE = OpenGpxTracker/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
@@ -974,7 +1035,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(SDKROOT)/Developer/Library/Frameworks",
 					"$(inherited)",
@@ -996,7 +1057,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(SDKROOT)/Developer/Library/Frameworks",
 					"$(inherited)",
@@ -1024,7 +1085,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
 				DEBUG_INFORMATION_FORMAT = dwarf;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				IBSC_MODULE = OpenGpxTracker_Watch_Extension;
 				INFOPLIST_FILE = "OpenGpxTracker-Watch/Info.plist";
@@ -1056,7 +1117,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				COPY_PHASE_STRIP = NO;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				IBSC_MODULE = OpenGpxTracker_Watch_Extension;
 				INFOPLIST_FILE = "OpenGpxTracker-Watch/Info.plist";
@@ -1084,7 +1145,7 @@
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
 				CODE_SIGN_STYLE = Automatic;
 				DEBUG_INFORMATION_FORMAT = dwarf;
-				DEVELOPMENT_TEAM = SU64RGPDF2;
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				INFOPLIST_FILE = "OpenGpxTracker-Watch Extension/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @executable_path/../../Frameworks";
@@ -1115,7 +1176,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				COPY_PHASE_STRIP = NO;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
-				DEVELOPMENT_TEAM = SU64RGPDF2;
+				DEVELOPMENT_TEAM = PMCWJ9GD9L;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				INFOPLIST_FILE = "OpenGpxTracker-Watch Extension/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @executable_path/../../Frameworks";
diff --git a/OpenGpxTracker/AboutViewController.swift b/OpenGpxTracker/AboutViewController.swift
index b3f63457..e63c0c40 100644
--- a/OpenGpxTracker/AboutViewController.swift
+++ b/OpenGpxTracker/AboutViewController.swift
@@ -4,6 +4,8 @@
 //
 //  Created by merlos on 24/09/14.
 //
+//  Localized by nitricware on 19/08/19.
+//
 
 import UIKit
 import WebKit
@@ -38,10 +40,10 @@ class AboutViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.title = "About"
+        self.title = NSLocalizedString("ABOUT", comment: "no comment")
         
         //Add the done button
-        let shareItem = UIBarButtonItem(title: "Done", style: UIBarButtonItem.Style.plain, target: self, action: #selector(AboutViewController.closeViewController))
+        let shareItem = UIBarButtonItem(title: NSLocalizedString("DONE", comment: "no comment"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(AboutViewController.closeViewController))
         self.navigationItem.rightBarButtonItems = [shareItem]
   
         //Add the Webview
diff --git a/OpenGpxTracker/CLActivityType+Info.swift b/OpenGpxTracker/CLActivityType+Info.swift
index aa6ad8db..72666714 100644
--- a/OpenGpxTracker/CLActivityType+Info.swift
+++ b/OpenGpxTracker/CLActivityType+Info.swift
@@ -13,11 +13,11 @@ extension CLActivityType {
     /// Returns the string name of this activty type
     var name: String {
         switch self {
-        case .other:                return "Automatic"
-        case .automotiveNavigation: return "Automotive navigation"
-        case .fitness:              return "Fitness"
-        case .otherNavigation:      return "Other navigation"
-        case .airborne:             return "Flight"
+        case .other:                return NSLocalizedString("AT_AUTOMATIC", comment: "no comment")
+        case .automotiveNavigation: return NSLocalizedString("AT_AUTOMOTIVE", comment: "no comment")
+        case .fitness:              return NSLocalizedString("AT_FITNESS", comment: "no comment")
+        case .otherNavigation:      return NSLocalizedString("AT_OTHER", comment: "no comment")
+        case .airborne:             return NSLocalizedString("AT_FLIGHT", comment: "no comment")
         @unknown default:
             return "Unknown Type: \(self.rawValue)"
         }
@@ -28,11 +28,11 @@ extension CLActivityType {
     /// "System default. Automatically selects the mode"
     var description: String {
         switch self {
-        case .other:                return "System default. Automatically selects the mode"
-        case .automotiveNavigation: return "Car, motorbike, trucks..."
-        case .fitness:              return "Running, hiking, cycling..."
-        case .otherNavigation:      return "Other than automotive navigation"
-        case .airborne:             return "Airborne activities"
+        case .other:                return NSLocalizedString("AT_AUTOMATIC_DESC", comment: "no comment")
+        case .automotiveNavigation: return NSLocalizedString("AT_AUTOMOTIVE_DESC", comment: "no comment")
+        case .fitness:              return NSLocalizedString("AT_FITNESS_DESC", comment: "no comment")
+        case .otherNavigation:      return NSLocalizedString("AT_OTHER_DESC", comment: "no comment")
+        case .airborne:             return NSLocalizedString("AT_FLIGHT_DESC", comment: "no comment")
         @unknown default:
             return "Unknown Type: \(self.rawValue)"
         }
diff --git a/OpenGpxTracker/CoreDataHelper.swift b/OpenGpxTracker/CoreDataHelper.swift
index c345bf5d..d1de1f96 100644
--- a/OpenGpxTracker/CoreDataHelper.swift
+++ b/OpenGpxTracker/CoreDataHelper.swift
@@ -592,19 +592,19 @@ class CoreDataHelper {
                 // asks user on what to do with recovered data
                 DispatchQueue.main.sync {
                     // main action sheet setup
-                    let alertController = UIAlertController(title: "Continue last session?", message: "What would you like to do with the recovered content from last session?", preferredStyle: .actionSheet)
+                    let alertController = UIAlertController(title: NSLocalizedString("CONTINUE_SESSION_TITLE", comment: "no comment"), message: NSLocalizedString("CONTINUE_SESSION_MESSAGE", comment: "no comment"), preferredStyle: .actionSheet)
                     
                     // option to cancel
-                    let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (action) in
+                    let cancelAction = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { (action) in
                         self.clearAll()
                     }
                     // option to continue previous session, which will load it, but not save
-                    let continueAction = UIAlertAction(title: "Continue Session", style: .default) { (action) in
+                    let continueAction = UIAlertAction(title: NSLocalizedString("CONTINUE_SESSION", comment: "no comment"), style: .default) { (action) in
                         NotificationCenter.default.post(name: .loadRecoveredFile, object: nil, userInfo: ["recoveredRoot" : root, "fileName" : self.lastFileName])
                     }
                     
                     // option to save silently as file, session remains new
-                    let saveAction = UIAlertAction(title: "Save and Start New", style: .default) { (action) in
+                    let saveAction = UIAlertAction(title: NSLocalizedString("SAVE_START_NEW", comment: "no comment"), style: .default) { (action) in
                         self.saveFile(from: root, andIfAvailable: self.lastFileName)
                     }
                     
diff --git a/OpenGpxTracker/Date+timeAgo.swift b/OpenGpxTracker/Date+timeAgo.swift
index 4d2afc42..c2c484e2 100644
--- a/OpenGpxTracker/Date+timeAgo.swift
+++ b/OpenGpxTracker/Date+timeAgo.swift
@@ -4,6 +4,7 @@
 //
 //  Created by merlos on 23/09/2018.
 //
+//  Localized by nitricware on 19/08/19.
 //
 // Based on this discussion: https://gist.github.com/minorbug/468790060810e0d29545
 
@@ -32,51 +33,51 @@ extension Date {
         //print(components)
         if let year = components.year {
             if (year >= 2) {
-                return "\(year) years ago"
+                return String(format: NSLocalizedString("T_YEARS_AGO", comment: "no comment"), year)
             } else if (year >= 1) {
-                return numericDates ?  "1 year ago" : "Last year"
+                return numericDates ? NSLocalizedString("T_YEAR_AGO", comment: "no comment") : NSLocalizedString("T_LAST_YEAR", comment: "no comment")
             }
         }
         if let month = components.month {
             if (month >= 2) {
-                return "\(month) months ago"
+                return String(format: NSLocalizedString("T_MONTHS_AGO", comment: "no comment"), month)
             } else if (month >= 1) {
-                return numericDates ? "1 month ago" : "Last month"
+                return numericDates ? NSLocalizedString("T_MONTH_AGO", comment: "no comment") : NSLocalizedString("T_LAST_MONTH", comment: "no comment")
             }
         }
         if let weekOfMonth = components.weekOfMonth {
             if (weekOfMonth >= 2) {
-                return "\(weekOfMonth) weeks ago"
+                return String(format: NSLocalizedString("T_WEEKS_AGO", comment: "no comment"), weekOfMonth)
             } else if (weekOfMonth >= 1) {
-                return numericDates ? "1 week ago" : "Last week"
+                return numericDates ? NSLocalizedString("T_WEEK_AGO", comment: "no comment") : NSLocalizedString("T_LAST_WEEK", comment: "no comment")
             }
         }
         if let day = components.day {
             if (day >= 2) {
-                return "\(day) days ago"
+                return String(format: NSLocalizedString("T_DAYS_AGO", comment: "no comment"), day)
             } else if (day >= 1) {
-                return numericDates ? "1 day ago" : "Yesterday"
+                return numericDates ? NSLocalizedString("T_DAY_AGO", comment: "no comment") : NSLocalizedString("T_YESTERDAY", comment: "no comment")
             }
         }
         if let hour = components.hour {
             if (hour >= 2) {
-                return "\(hour) hours ago"
+                return String(format: NSLocalizedString("T_HOURS_AGO", comment: "no comment"), hour)
             } else if (hour >= 1) {
-                return numericDates ? "1 hour ago" : "An hour ago"
+                return numericDates ? NSLocalizedString("T_HOUR_AGO", comment: "no comment") : NSLocalizedString("T_LAST_HOUR", comment: "no comment")
             }
         }
         if let minute = components.minute {
             if (minute >= 2) {
-                return "\(minute) minutes ago"
+                return String(format: NSLocalizedString("T_MINUTES_AGO", comment: "no comment"), minute)
             } else if (minute >= 1) {
-                return numericDates ? "1 minute ago" : "A minute ago"
+                return numericDates ? NSLocalizedString("T_MINUTE_AGO", comment: "no comment") : NSLocalizedString("T_LAST_MINUTE", comment: "no comment")
             }
         }
         if let second = components.second {
             if (second >= 3) {
-                return "\(second) seconds ago"
+                return String(format: NSLocalizedString("T_SECONDS_AGO", comment: "no comment"), second)
             }
         }
-        return "Just now"
+        return NSLocalizedString("T_JUST_NOW", comment: "no comment")
     }
 }
diff --git a/OpenGpxTracker/GPXFilesTableViewController.swift b/OpenGpxTracker/GPXFilesTableViewController.swift
index 8d769801..5b782e6c 100644
--- a/OpenGpxTracker/GPXFilesTableViewController.swift
+++ b/OpenGpxTracker/GPXFilesTableViewController.swift
@@ -4,6 +4,8 @@
 //
 //  Created by merlos on 14/09/14.
 //
+//  Localized by nitricware on 19/08/19.
+//
 
 import Foundation
 import UIKit
@@ -12,7 +14,7 @@ import MessageUI
 import WatchConnectivity
 
 /// Text displayed when there are no GPX files in the folder.
-let kNoFiles = "No gpx files"
+let kNoFiles = NSLocalizedString("NO_FILES", comment: "no comment")
 
 ///
 /// TableViewController that displays the list of files that have been saved in previous sessions.
@@ -54,13 +56,13 @@ class GPXFilesTableViewController: UITableViewController, UINavigationBarDelegat
         self.tableView.frame = CGRect(x: navBarFrame.width + 1, y: 0, width: self.view.frame.width, height:
             self.view.frame.height - navBarFrame.height)
         
-        self.title = "Your GPX Files"
+        self.title = NSLocalizedString("YOUR_FILES", comment: "no comment")
         
         // add notification observer for reloading table when file is added.
         addNotificationObservers()
         
         // Button to return to the map
-        let shareItem = UIBarButtonItem(title: "Done", style: UIBarButtonItem.Style.plain, target: self, action: #selector(GPXFilesTableViewController.closeGPXFilesTableViewController))
+        let shareItem = UIBarButtonItem(title: NSLocalizedString("DONE", comment: "no comment"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(GPXFilesTableViewController.closeGPXFilesTableViewController))
         
         self.navigationItem.rightBarButtonItems = [shareItem]
         
@@ -138,8 +140,7 @@ class GPXFilesTableViewController: UITableViewController, UINavigationBarDelegat
             //cell.accessoryView = [[ UIImageView alloc ] initWithImage:[UIImage imageNamed:@"Something" ]];
             let gpxFileInfo = fileList.object(at: (indexPath as NSIndexPath).row) as! GPXFileInfo
             cell.textLabel?.text = gpxFileInfo.fileName
-            cell.detailTextLabel?.text =
-                "last saved \(gpxFileInfo.modifiedDatetimeAgo) (\(gpxFileInfo.fileSizeHumanised))"
+            cell.detailTextLabel?.text = String(format: NSLocalizedString("LAST_SAVED", comment: "no comment"), gpxFileInfo.modifiedDatetimeAgo, gpxFileInfo.fileSizeHumanised)
             cell.detailTextLabel?.textColor = UIColor.darkGray
             return cell
         } else {
@@ -153,19 +154,19 @@ class GPXFilesTableViewController: UITableViewController, UINavigationBarDelegat
     /// Displays an action sheet with the actions for that file (Send it by email, Load in map and Delete).
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
 
-        let sheet = UIAlertController(title: nil, message: "Select option", preferredStyle: .actionSheet)
-        let mapOption = UIAlertAction(title: "Load in Map", style: .default) { action in
+        let sheet = UIAlertController(title: nil, message: NSLocalizedString("SELECT_OPTION", comment: "no comment"), preferredStyle: .actionSheet)
+        let mapOption = UIAlertAction(title: NSLocalizedString("LOAD_IN_MAP", comment: "no comment"), style: .default) { action in
             self.actionLoadFileAtIndex(indexPath.row)
         }
-        let shareOption = UIAlertAction(title: "Share", style: .default) { action in
+        let shareOption = UIAlertAction(title: NSLocalizedString("SHARE", comment: "no comment"), style: .default) { action in
             self.actionShareFileAtIndex(indexPath.row, tableView: tableView, indexPath: indexPath)
         }
         
-        let cancelOption = UIAlertAction(title: "Cancel", style: .cancel) { action in
+        let cancelOption = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { action in
             self.actionSheetCancel(sheet)
         }
         
-        let deleteOption = UIAlertAction(title: "Delete", style: .destructive) { action in
+        let deleteOption = UIAlertAction(title: NSLocalizedString("DELETE", comment: "no comment"), style: .destructive) { action in
             self.actionDeleteFileAtIndex(indexPath.row)
         }
         
@@ -260,7 +261,7 @@ class GPXFilesTableViewController: UITableViewController, UINavigationBarDelegat
     /// Displays an alert with a activity indicator view to indicate loading of gpx file to map
     func displayLoadingFileAlert(_ loading: Bool, completion: (() -> Void)? = nil) {
         // setup of controllers and views
-        let alertController = UIAlertController(title: "Loading GPX File...", message: nil, preferredStyle: .alert)
+        let alertController = UIAlertController(title: NSLocalizedString("LOADING_FILE", comment: "no comment"), message: nil, preferredStyle: .alert)
         let activityIndicatorView = UIActivityIndicatorView(frame: CGRect(x: 35, y: 30, width: 32, height: 32))
         activityIndicatorView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
         activityIndicatorView.style = .whiteLarge
diff --git a/OpenGpxTracker/Info.plist b/OpenGpxTracker/Info.plist
index 1844b49c..4d1aae24 100644
--- a/OpenGpxTracker/Info.plist
+++ b/OpenGpxTracker/Info.plist
@@ -34,7 +34,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.6.1</string>
+	<string>1.6.2</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
diff --git a/OpenGpxTracker/MapViewDelegate.swift b/OpenGpxTracker/MapViewDelegate.swift
index f6e1e9fb..bdb0b19d 100644
--- a/OpenGpxTracker/MapViewDelegate.swift
+++ b/OpenGpxTracker/MapViewDelegate.swift
@@ -1,3 +1,7 @@
+//
+//  Localized by nitricware on 19/08/19.
+//
+
 import MapKit
 import CoreGPX
 
@@ -72,17 +76,17 @@ class MapViewDelegate: NSObject, MKMapViewDelegate, UIAlertViewDelegate {
             
             let indexofEditedWaypoint = map.session.waypoints.firstIndex(of: waypoint)
             
-            let alertController = UIAlertController(title: "Edit waypoint name", message: "Hint: To change the waypoint location drag and drop the pin", preferredStyle: .alert)
+            let alertController = UIAlertController(title: NSLocalizedString("EDIT_WAYPOINT_NAME_TITLE", comment: "no comment"), message: NSLocalizedString("EDIT_WAYPOINT_NAME_MESSAGE", comment: "no comment"), preferredStyle: .alert)
             alertController.addTextField { (textField) in
                 textField.text = waypoint.title
                 textField.clearButtonMode = .always
             }
-            let saveAction = UIAlertAction(title: "Save", style: .default) { (action) in
+            let saveAction = UIAlertAction(title: NSLocalizedString("SAVE", comment: "no comment"), style: .default) { (action) in
                 print("Edit waypoint alert view")
                 self.waypointBeingEdited.title = alertController.textFields?[0].text
                 map.coreDataHelper.update(toCoreData: self.waypointBeingEdited, from: indexofEditedWaypoint!)
             }
-            let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (action) in }
+            let cancelAction = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { (action) in }
             
             alertController.addAction(saveAction)
             alertController.addAction(cancelAction)
diff --git a/OpenGpxTracker/PreferencesTableViewController.swift b/OpenGpxTracker/PreferencesTableViewController.swift
index 9e78783a..1f93c65e 100644
--- a/OpenGpxTracker/PreferencesTableViewController.swift
+++ b/OpenGpxTracker/PreferencesTableViewController.swift
@@ -4,6 +4,8 @@
 //
 //  Created by merlos on 24/10/15.
 //
+//  Localized by nitricware on 19/08/19.
+//
 
 import Foundation
 import UIKit
@@ -59,8 +61,8 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
         self.tableView.frame = CGRect(x: navBarFrame.width + 1, y: 0, width: self.view.frame.width, height:
             self.view.frame.height - navBarFrame.height)
         
-        self.title = "Preferences"
-        let shareItem = UIBarButtonItem(title: "Done", style: UIBarButtonItem.Style.plain, target: self, action: #selector(PreferencesTableViewController.closePreferencesTableViewController))
+        self.title = NSLocalizedString("PREFERENCES", comment: "no comment")
+        let shareItem = UIBarButtonItem(title: NSLocalizedString("DONE", comment: "no comment"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(PreferencesTableViewController.closePreferencesTableViewController))
         self.navigationItem.rightBarButtonItems = [shareItem]
     }
     
@@ -95,10 +97,10 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
     /// for deciding which is the section title
     override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
         switch(section) {
-        case kUnitsSection: return "Units"
-        case kCacheSection: return "Cache"
-        case kMapSourceSection: return "Map source"
-        case kActivityTypeSection: return "Activity Type"
+        case kUnitsSection: return NSLocalizedString("UNITS", comment: "no comment")
+        case kCacheSection: return NSLocalizedString("CACHE", comment: "no comment")
+        case kMapSourceSection: return NSLocalizedString("MAP_SOURCE", comment: "no comment")
+        case kActivityTypeSection: return NSLocalizedString("ACTIVITY_TYPE", comment: "no comment")
         default: fatalError("Unknown section")
         }
     }
@@ -136,7 +138,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
              switch (indexPath.row) {
              case kUseImperialUnitsCell:
                 cell = UITableViewCell(style: .value1, reuseIdentifier: "CacheCell")
-                cell.textLabel?.text = "Use imperial units?"
+                cell.textLabel?.text = NSLocalizedString("USE_IMPERIAL_UNITS", comment: "no comment")
                 if preferences.useImperial {
                     cell.accessoryType = .checkmark
                 }
@@ -149,13 +151,13 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             switch (indexPath.row) {
             case kUseOfflineCacheCell:
                 cell = UITableViewCell(style: .value1, reuseIdentifier: "CacheCell")
-                cell.textLabel?.text = "Offline cache"
+                cell.textLabel?.text = NSLocalizedString("OFFLINE_CACHE", comment: "no comment")
                 if preferences.useCache {
                     cell.accessoryType = .checkmark
                 }
             case kClearCacheCell:
                 cell = UITableViewCell(style: .value1, reuseIdentifier: "CacheCell")
-                cell.textLabel?.text = "Clear cache"
+                cell.textLabel?.text = NSLocalizedString("CLEAR_CACHE", comment: "no comment")
                 cell.textLabel?.textColor = UIColor.red
             default: fatalError("Unknown section")
             }
diff --git a/OpenGpxTracker/ViewController.swift b/OpenGpxTracker/ViewController.swift
index 8df643df..7acbd3ec 100644
--- a/OpenGpxTracker/ViewController.swift
+++ b/OpenGpxTracker/ViewController.swift
@@ -4,6 +4,8 @@
 //
 //  Created by merlos on 13/09/14.
 //
+//  Localized by nitricware on 19/08/19.
+//
 
 import UIKit
 import CoreLocation
@@ -39,7 +41,7 @@ let kDeleteWaypointAccesoryButtonTag = 666
 let kEditWaypointAccesoryButtonTag = 333
 
 /// Text to display when the system is not providing coordinates.
-let kNotGettingLocationText = "Not getting location"
+let kNotGettingLocationText = NSLocalizedString("NO_LOCATION", comment: "no comment")
 
 /// Text to display unknown accuracy
 let kUnknownAccuracyText = "±···"
@@ -165,7 +167,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
             case .notStarted:
                 print("switched to non started")
                 // set Tracker button to allow Start 
-                trackerButton.setTitle("Start Tracking", for: UIControl.State())
+                trackerButton.setTitle(NSLocalizedString("START_TRACKING", comment: "no comment"), for: UIControl.State())
                 trackerButton.backgroundColor = kGreenButtonBackgroundColor
                 //save & reset button to transparent.
                 saveButton.backgroundColor = kDisabledBlueButtonBackgroundColor
@@ -196,7 +198,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
             case .tracking:
                 print("switched to tracking mode")
                 // set tracerkButton to allow Pause
-                trackerButton.setTitle("Pause", for: UIControl.State())
+                trackerButton.setTitle(NSLocalizedString("PAUSE", comment: "no comment"), for: UIControl.State())
                 trackerButton.backgroundColor = kPurpleButtonBackgroundColor
                 //activate save & reset buttons
                 saveButton.backgroundColor = kBlueButtonBackgroundColor
@@ -207,7 +209,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
             case .paused:
                 print("switched to paused mode")
                 // set trackerButton to allow Resume
-                self.trackerButton.setTitle("Resume", for: UIControl.State())
+                self.trackerButton.setTitle(NSLocalizedString("RESUME", comment: "no comment"), for: UIControl.State())
                 self.trackerButton.backgroundColor = kGreenButtonBackgroundColor
                 // activate save & reset (just in case switched from .NotStarted)
                 saveButton.backgroundColor = kBlueButtonBackgroundColor
@@ -563,7 +565,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         
         // Start/Pause button
         trackerButton.layer.cornerRadius = kButtonLargeSize/2
-        trackerButton.setTitle("Start Tracking", for: UIControl.State())
+        trackerButton.setTitle(NSLocalizedString("START_TRACKING", comment: "no comment"), for: UIControl.State())
         trackerButton.backgroundColor = kGreenButtonBackgroundColor
         trackerButton.addTarget(self, action: #selector(ViewController.trackerButtonTapped), for: .touchUpInside)
         trackerButton.isHidden = false
@@ -593,20 +595,22 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         
         // Save button
         saveButton.layer.cornerRadius = kButtonSmallSize/2
-        saveButton.setTitle("Save", for: UIControl.State())
+        saveButton.setTitle(NSLocalizedString("SAVE", comment: "no comment"), for: UIControl.State())
         saveButton.backgroundColor = kDisabledBlueButtonBackgroundColor
         saveButton.addTarget(self, action: #selector(ViewController.saveButtonTapped), for: .touchUpInside)
         saveButton.isHidden = false
         saveButton.titleLabel?.textAlignment = .center
+        saveButton.titleLabel?.adjustsFontSizeToFitWidth = true
         map.addSubview(saveButton)
         
         // Reset button
         resetButton.layer.cornerRadius = kButtonSmallSize/2
-        resetButton.setTitle("Reset", for: UIControl.State())
+        resetButton.setTitle(NSLocalizedString("RESET", comment: "no comment"), for: UIControl.State())
         resetButton.backgroundColor = kDisabledRedButtonBackgroundColor
         resetButton.addTarget(self, action: #selector(ViewController.resetButtonTapped), for: .touchUpInside)
         resetButton.isHidden = false
         resetButton.titleLabel?.textAlignment = .center
+        resetButton.titleLabel?.adjustsFontSizeToFitWidth = true
         map.addSubview(resetButton)
         
         addConstraints(isIPhoneX)
@@ -760,10 +764,11 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
     @objc func presentReceivedFile(_ notification: Notification) {
         
         guard let fileName = notification.userInfo?["fileName"] as? String? else { return }
-        
         // alert to display to notify user that file has been received.
-        let controller = UIAlertController(title: "File Received from Apple Watch", message: "Received file: \"\(fileName ?? "")\"", preferredStyle: .alert)
-        let action = UIAlertAction(title: "Done", style: .default) {
+        let alertTitle = NSLocalizedString("WATCH_FILE_RECEIVED_TITLE", comment: "no comment")
+        let alertMessage = NSLocalizedString("WATCH_FILE_RECEIVED_MESSAGE", comment: "no comment")
+        let controller = UIAlertController(title: alertTitle, message: String(format: alertMessage, fileName ?? ""), preferredStyle: .alert)
+        let action = UIAlertAction(title: NSLocalizedString("DONE", comment: "no comment"), style: .default) {
             (action) in
             print("ViewController:: Presented file received message from WatchConnectivity Session")
         }
@@ -1047,14 +1052,14 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         }
         
         // save alert configuration and presentation
-        let alertController = UIAlertController(title: "Save as", message: "Enter GPX session name", preferredStyle: .alert)
+        let alertController = UIAlertController(title: NSLocalizedString("SAVE_AS", comment: "no comment"), message: NSLocalizedString("ENTER_SESSION_NAME", comment: "no comment"), preferredStyle: .alert)
         
         alertController.addTextField(configurationHandler: { (textField) in
             textField.clearButtonMode = .always
             textField.text = self.lastGpxFilename.isEmpty ? self.defaultFilename() : self.lastGpxFilename
         })
         
-        let saveAction = UIAlertAction(title: "Save", style: .default) { (action) in
+        let saveAction = UIAlertAction(title: NSLocalizedString("SAVE", comment: "no comment"), style: .default) { (action) in
             let filename = (alertController.textFields?[0].text!.utf16.count == 0) ? self.defaultFilename() : alertController.textFields?[0].text
             print("Save File \(String(describing: filename))")
             //export to a file
@@ -1064,7 +1069,7 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
             self.map.coreDataHelper.deleteLastFileNameFromCoreData()
             self.map.coreDataHelper.add(toCoreData: filename!)
         }
-        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (action) in }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { (action) in }
         
         alertController.addAction(saveAction)
         alertController.addAction(cancelAction)
@@ -1108,13 +1113,13 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
     ///
     func displayLocationServicesDisabledAlert() {
         
-        let alertController = UIAlertController(title: "Location services disabled", message: "Go to settings and enable location.", preferredStyle: .alert)
-        let settingsAction = UIAlertAction(title: "Settings", style: .default) { (action) in
+        let alertController = UIAlertController(title: NSLocalizedString("LOCATION_SERVICES_DISABLED", comment: "no comment"), message: NSLocalizedString("ENABLE_LOCATION_SERVICES", comment: "no comment"), preferredStyle: .alert)
+        let settingsAction = UIAlertAction(title: NSLocalizedString("SETTINGS", comment: "no comment"), style: .default) { (action) in
             if let url = URL(string: UIApplication.openSettingsURLString) {
                 UIApplication.shared.openURL(url)
             }
         }
-        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (action) in }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { (action) in }
         
         alertController.addAction(settingsAction)
         alertController.addAction(cancelAction)
@@ -1132,13 +1137,13 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
         if isDisplayingLocationServicesDenied {
             return // display it only once.
         }
-        let alertController = UIAlertController(title: "Access to location denied", message: "On Location settings, allow always access to location for GPX Tracker ", preferredStyle: .alert)
-        let settingsAction = UIAlertAction(title: "Settings", style: .default) { (action) in
+        let alertController = UIAlertController(title: NSLocalizedString("ACCESS_TO_LOCATION_DENIED", comment: "no comment"), message: NSLocalizedString("ALLOW_LOCATION", comment: "no comment"), preferredStyle: .alert)
+        let settingsAction = UIAlertAction(title: NSLocalizedString("SETTINGS", comment: "no comment"), style: .default) { (action) in
             if let url = URL(string: UIApplication.openSettingsURLString) {
                 UIApplication.shared.openURL(url)
             }
         }
-        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (action) in }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("CANCEL", comment: "no comment"), style: .cancel) { (action) in }
         
         alertController.addAction(settingsAction)
         alertController.addAction(cancelAction)
@@ -1307,7 +1312,8 @@ extension ViewController: CLLocationManagerDelegate {
         //Update coordsLabel
         let latFormat = String(format: "%.6f", newLocation.coordinate.latitude)
         let lonFormat = String(format: "%.6f", newLocation.coordinate.longitude)
-        coordsLabel.text = "(\(latFormat),\(lonFormat)) · altitude: \(newLocation.altitude.toAltitude(useImperial: useImperial))"
+        let altitude = newLocation.altitude.toAltitude(useImperial: useImperial)
+        coordsLabel.text = String(format: NSLocalizedString("COORDS_LABEL", comment: "no comment"), latFormat, lonFormat, altitude)
         
         //Update speed
         speedLabel.text = (newLocation.speed < 0) ? kUnknownSpeedText : newLocation.speed.toSpeed(useImperial: useImperial)
diff --git a/OpenGpxTracker/de.lproj/InfoPlist.strings b/OpenGpxTracker/de.lproj/InfoPlist.strings
new file mode 100644
index 00000000..2334689e
--- /dev/null
+++ b/OpenGpxTracker/de.lproj/InfoPlist.strings
@@ -0,0 +1,12 @@
+/* 
+  InfoPlist.strings
+  OpenGpxTracker
+
+  Created by Kurt Höblinger on 21.08.19.
+  Copyright © 2019 TransitBox. All rights reserved.
+*/
+
+NSLocationAlwaysAndWhenInUseUsageDescription = "Der Standort wird nur verwendet um Deinen Standort auf der Karte darzustellen. Wähle \"immer\" um den Standort auch zur Verfügung zu stellen, wenn die App im Hintergrund arbeitet.";
+NSLocationAlwaysUsageDescription = "Dein Standort wird benötigt um den GPX Track auch dann zu erstellen, wenn die App im Hintergrund läuft.";
+NSLocationUsageDescription = "Der Standort wird zum Erstellen von GPX und zum Hinzufügen von PDIs benötigt";
+NSLocationWhenInUseUsageDescription = "Der Standort wird verwendet um auf der Karte zu zeigen wo Du bist und wo Du warst.";
diff --git a/OpenGpxTracker/de.lproj/LaunchScreen.strings b/OpenGpxTracker/de.lproj/LaunchScreen.strings
new file mode 100644
index 00000000..f8672872
--- /dev/null
+++ b/OpenGpxTracker/de.lproj/LaunchScreen.strings
@@ -0,0 +1,6 @@
+
+/* Class = "UILabel"; text = "  Copyright (c) Juan M. Merlos"; ObjectID = "8ie-xW-0ye"; */
+"8ie-xW-0ye.text" = "  Copyright (c) Juan M. Merlos";
+
+/* Class = "UILabel"; text = "OPEN GPX TRACKER"; ObjectID = "kId-c2-rCX"; */
+"kId-c2-rCX.text" = "OPEN GPX TRACKER";
diff --git a/OpenGpxTracker/de.lproj/Localizable.strings b/OpenGpxTracker/de.lproj/Localizable.strings
new file mode 100644
index 00000000..cc8f30d7
--- /dev/null
+++ b/OpenGpxTracker/de.lproj/Localizable.strings
@@ -0,0 +1,106 @@
+/*
+ Localizable.strings
+ OpenGpxTracker
+ 
+ Created by Kurt Höblinger on 19.08.19.
+ */
+
+"NO_LOCATION" = "GPS nicht verfügbar";
+"START_TRACKING" = "Tracking starten";
+"PAUSE" = "Pause";
+"RESUME" = "Fortsetzen";
+"SAVE" = "Sichern";
+"RESET" = "Löschen";
+"DONE" = "Fertig";
+"CANCEL" = "Abbrechen";
+"WATCH_FILE_RECEIVED_TITLE" = "Datei von der Apple Watch erhalten.";
+"WATCH_FILE_RECEIVED_MESSAGE" = "Datei erhalten: \"%@\"";
+"SAVE_AS" = "Speichern unter";
+"ENTER_SESSION_NAME" = "GPX Sitzungs-Name eingeben";
+"LOCATION_SERVICES_DISABLED" = "Ortungsdienste ausgeschalten";
+"ENABLE_LOCATION_SERVICES" = "Gehe in die Einstellungen und schalte die Ortungsdienste ein.";
+"SETTINGS" = "Einstellungen";
+"ACCESS_TO_LOCATION_DENIED" = "Zugriff auf Standort verweigert.";
+"ALLOW_LOCATION" = "Erlaube GPX Tracker immer den Zugriff auf den Standort in den Einstellungen. ";
+"COORDS_LABEL" = "(%@, %@) · Höhe: %@";
+"NO_FILES" = "Keine GPX Dateien";
+"YOUR_FILES" = "Deine GPX Dateien";
+"LAST_SAVED" = "%@ (%@) zuletzt gespeichert";
+"DELETE" = "Löschen";
+"SELECT_OPTION" = "Auswählen";
+"LOAD_IN_MAP" = "In Karte laden";
+"SHARE" = "Teilen";
+"LOADING_FILE" = "Lade GPX Datei...";
+"ABOUT" = "Über";
+"PREFERENCES" = "Einstellungen";
+"UNITS" = "Einheiten";
+"CACHE" = "Cache";
+"MAP_SOURCE" = "Kartentyp";
+"ACTIVITY_TYPE" = "Aktivitätstyp";
+"USE_IMPERIAL_UNITS" = "Meilen verwenden";
+"OFFLINE_CACHE" = "Offline cache";
+"CLEAR_CACHE" = "Cache leeren";
+"EDIT_WAYPOINT_NAME_TITLE" = "Wegpunkt ändern";
+"EDIT_WAYPOINT_NAME_MESSAGE" = "Hinweis: Um einen Wegpunkt zu verändern, verschiebe den Pin.";
+
+// Watch
+"SENDING" = "Sende:";
+"X_FILES" = "%d Dateien";
+"SUCCESSFULLY_SENT" = "Erfolgreich gesendet:";
+"FAILED_TO_SEND" = "Senden fehlgeschlagen:";
+"SEND_TO_IOS" = "Sende an iOS App";
+"FILE_SELECTED_TITLE" = "GPX Datei ausgewählt";
+"FILE_SELECTED_MESSAGE" = "Was möchtest Du tun?";
+"ERROR_OCCURED_TITLE" = "Fehler aufgetreten";
+"ERROR_OCCURED_MESSAGE" = "GPX Datei wurde nicht an iOS App gesendet: %s ";
+"GPX_TRACKER" = "GPX Tracker";
+"FILE_SAVED_TITLE" = "GPX Datei gespeichert";
+"FILE_SAVED_MESSAGE" = "Aktuelle Sitzung als %s.gpx gespeichert ";
+
+"AT_AUTOMATIC" = "Automatisch";
+"AT_AUTOMOTIVE" = "Fahrzeugnavigation";
+"AT_FITNESS" = "Fitness";
+"AT_OTHER" = "Andere Navigation";
+"AT_FLIGHT" = "Flug";
+
+"AT_AUTOMATIC_DESC" = "Systemstandard. Modus wird automatisch erkannt";
+"AT_AUTOMOTIVE_DESC" = "Auto, Motorrad, LKW...";
+"AT_FITNESS_DESC" = "Laufen, Wandern, Radfahren...";
+"AT_OTHER_DESC" = "Andere Fahrzeugnavigation";
+"AT_FLIGHT_DESC" = "Flugaktivitäten";
+
+"T_YEARS_AGO" = "vor %d Jahren";
+"T_YEAR_AGO" = "vor 1 Jahr";
+"T_LAST_YEAR" = "Letztes Jahr";
+
+"T_MONTHS_AGO" = "vor %d Monaten";
+"T_MONTH_AGO" = "vor 1 Monat";
+"T_LAST_MONTH" = "Letztes Monat";
+
+"T_WEEKS_AGO" = "vor %d Wochen";
+"T_WEEK_AGO" = "vor 1 Woche";
+"T_LAST_WEEK" = "Letzte Woche";
+
+"T_DAYS_AGO" = "vor %d Tagen";
+"T_DAY_AGO" = "vor 1 Tag";
+"T_YESTERDAY" = "Gestern";
+
+"T_HOURS_AGO" = "vor %d Stunden";
+"T_HOUR_AGO" = "vor 1 Stunde";
+"T_LAST_HOUR" = "Letzte Stunde";
+
+"T_MINUTES_AGO" = "vor %d Minuten";
+"T_MINUTE_AGO" = "vor 1 Minute";
+"T_MINUTE_HOUR" = "Vorige Minute";
+
+"T_SECONDS_AGO" = "vor %d Sekunden";
+
+"T_JUST_NOW" = "Gerade eben";
+
+// CoreDataHelper
+
+"CONTINUE_SESSION_TITLE" = "Letzte Sitzung fortsetzen?";
+"CONTINUE_SESSION_MESSAGE" = "Was möchtest Du mit dem Wiederhergestellten Track der letzten Sitzung machen?";
+
+"CONTINUE_SESSION_TITLE" = "Sitzung fortsetzen";
+"SAVE_START_NEW" = "Speichern und Neustart";
diff --git a/OpenGpxTracker/de.lproj/Main.strings b/OpenGpxTracker/de.lproj/Main.strings
new file mode 100644
index 00000000..8b137891
--- /dev/null
+++ b/OpenGpxTracker/de.lproj/Main.strings
@@ -0,0 +1 @@
+
diff --git a/OpenGpxTracker/en.lproj/InfoPlist.strings b/OpenGpxTracker/en.lproj/InfoPlist.strings
new file mode 100644
index 00000000..5a2429b2
--- /dev/null
+++ b/OpenGpxTracker/en.lproj/InfoPlist.strings
@@ -0,0 +1,12 @@
+/* 
+  InfoPlist.strings
+  OpenGpxTracker
+
+  Created by Kurt Höblinger on 21.08.19.
+  Copyright © 2019 TransitBox. All rights reserved.
+*/
+
+NSLocationAlwaysAndWhenInUseUsageDescription = "Location is only used to display your position in the map. Select always allow to track your position even if the app is in background.";
+NSLocationAlwaysUsageDescription = "Location is needed to create the GPX track even when the app is in background.";
+NSLocationUsageDescription = "Location is needed to create GPX and add PDIs";
+NSLocationWhenInUseUsageDescription = "Location is used to display in the map where are you and where you have been.";
diff --git a/OpenGpxTracker/en.lproj/Localizable.strings b/OpenGpxTracker/en.lproj/Localizable.strings
new file mode 100644
index 00000000..cad2e58f
--- /dev/null
+++ b/OpenGpxTracker/en.lproj/Localizable.strings
@@ -0,0 +1,109 @@
+/*
+ Localizable.strings
+ OpenGpxTracker
+ 
+ Created by Kurt Höblinger on 19.08.19.
+ */
+
+// App
+"NO_LOCATION" = "Not getting location";
+"START_TRACKING" = "Start Tracking";
+"PAUSE" = "Pause";
+"RESUME" = "Resume";
+"SAVE" = "Save";
+"RESET" = "Reset";
+"DONE" = "Done";
+"CANCEL" = "Cancel";
+"WATCH_FILE_RECEIVED_TITLE" = "File Received from Apple Watch";
+"WATCH_FILE_RECEIVED_MESSAGE" = "Received file: \"%@\"";
+"SAVE_AS" = "Save as";
+"ENTER_SESSION_NAME" = "Enter GPX session name";
+"LOCATION_SERVICES_DISABLED" = "Location services disabled";
+"ENABLE_LOCATION_SERVICES" = "Go to settings and enable location.";
+"SETTINGS" = "Settings";
+"ACCESS_TO_LOCATION_DENIED" = "Access to location denied";
+"ALLOW_LOCATION" = "On Location settings, allow always access to location for GPX Tracker ";
+"COORDS_LABEL" = "(%@, %@) · altitude: %@";
+"NO_FILES" = "No gpx files";
+"YOUR_FILES" = "Your gpx files";
+"LAST_SAVED" = "last saved %@ (%@)";
+"DELETE" = "Delete";
+"SELECT_OPTION" = "Select option";
+"LOAD_IN_MAP" = "Load in map";
+"SHARE" = "Share";
+"LOADING_FILE" = "Loading GPX File...";
+"ABOUT" = "About";
+"PREFERENCES" = "Preferences";
+"UNITS" = "Units";
+"CACHE" = "Cache";
+"MAP_SOURCE" = "Map Source";
+"ACTIVITY_TYPE" = "Activity Type";
+"USE_IMPERIAL_UNITS" = "Use imperial units?";
+"OFFLINE_CACHE" = "Offline cache";
+"CLEAR_CACHE" = "Clear cache";
+"EDIT_WAYPOINT_NAME_TITLE" = "Edit waypoint name";
+"EDIT_WAYPOINT_NAME_MESSAGE" = "Hint: To change the waypoint location drag and drop the pin";
+
+// Watch
+"SENDING" = "Sending:";
+"X_FILES" = "%d files";
+"SUCCESSFULLY_SENT" = "Successfully sent:";
+"FAILED_TO_SEND" = "Failed to send:";
+"SEND_TO_IOS" = "Send to iOS app";
+"FILE_SELECTED_TITLE" = "GPX file selected";
+"FILE_SELECTED_MESSAGE" = "What would you like to do?";
+"ERROR_OCCURED_TITLE" = "Error occured";
+"ERROR_OCCURED_MESSAGE" = "GPX file was unsuccessfully sent to iOS app, due to %s ";
+"GPX_TRACKER" = "GPX Tracker";
+"FILE_SAVED_TITLE" = "GPX file saved";
+"FILE_SAVED_MESSAGE" = "Current session saved as %s.gpx ";
+
+// Activities
+"AT_AUTOMATIC" = "Automatic";
+"AT_AUTOMOTIVE" = "Automotive navigation";
+"AT_FITNESS" = "Fitness";
+"AT_OTHER" = "Other navigation";
+"AT_FLIGHT" = "Flight";
+
+"AT_AUTOMATIC_DESC" = "System default. Automatically selects the mode";
+"AT_AUTOMOTIVE_DESC" = "Car, motorbike, trucks...";
+"AT_FITNESS_DESC" = "Running, hiking, cycling...";
+"AT_OTHER_DESC" = "Other than automotive navigation";
+"AT_FLIGHT_DESC" = "Airborne activities";
+
+// Dates
+"T_YEARS_AGO" = "%d years ago";
+"T_YEAR_AGO" = "1 year ago";
+"T_LAST_YEAR" = "Last year";
+
+"T_MONTHS_AGO" = "%d months ago";
+"T_MONTH_AGO" = "1 month ago";
+"T_LAST_MONTH" = "Last month";
+
+"T_WEEKS_AGO" = "%d weeks ago";
+"T_WEEK_AGO" = "1 week ago";
+"T_LAST_WEEK" = "Last week";
+
+"T_DAYS_AGO" = "%d days ago";
+"T_DAY_AGO" = "1 day ago";
+"T_YESTERDAY" = "Yesterday";
+
+"T_HOURS_AGO" = "%d hours ago";
+"T_HOUR_AGO" = "1 hour ago";
+"T_LAST_HOUR" = "Last hour";
+
+"T_MINUTES_AGO" = "%d minutes ago";
+"T_MINUTE_AGO" = "1 minute ago";
+"T_MINUTE_HOUR" = "A minute ago";
+
+"T_SECONDS_AGO" = "%d seconds ago";
+
+"T_JUST_NOW" = "Just now";
+
+// CoreDataHelper
+
+"CONTINUE_SESSION_TITLE" = "Continue last session?";
+"CONTINUE_SESSION_MESSAGE" = "What would you like to do with the recovered content from last session?";
+
+"CONTINUE_SESSION_TITLE" = "Continue session";
+"SAVE_START_NEW" = "Save and Start New";
diff --git a/Pods/Pods.xcodeproj/project.pbxproj b/Pods/Pods.xcodeproj/project.pbxproj
index 0aefd0b8..bafa228e 100644
--- a/Pods/Pods.xcodeproj/project.pbxproj
+++ b/Pods/Pods.xcodeproj/project.pbxproj
@@ -1071,6 +1071,7 @@
 			hasScannedForEncodings = 0;
 			knownRegions = (
 				en,
+				Base,
 			);
 			mainGroup = CF1408CF629C7361332E53B88F7BD30C;
 			productRefGroup = 63AADCA50B3A50E4D1D7F22B9D3EA226 /* Products */;
