diff --git a/OpenGpxTracker.xcodeproj/project.pbxproj b/OpenGpxTracker.xcodeproj/project.pbxproj
index f4900fd2..229ba477 100644
--- a/OpenGpxTracker.xcodeproj/project.pbxproj
+++ b/OpenGpxTracker.xcodeproj/project.pbxproj
@@ -57,6 +57,7 @@
 		BF6408A3225C63E700BB5242 /* CoreDataHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF6408A2225C63E700BB5242 /* CoreDataHelper.swift */; };
 		BF6428B8220EC1A90015447E /* GPXFileTableRowController.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF6428B7220EC1A90015447E /* GPXFileTableRowController.swift */; };
 		BF64602F220AE9CF00957FF6 /* GPXTrackSegment+MapKit.swift in Sources */ = {isa = PBXBuildFile; fileRef = 899D676B19CD9E9E00C88A0A /* GPXTrackSegment+MapKit.swift */; };
+		BF68ECC62448B887005C779F /* DateFieldTypeView.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF68ECC52448B887005C779F /* DateFieldTypeView.swift */; };
 		BF692F5D2209AAFB00F8A0F1 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = BF692F5C2209AAFB00F8A0F1 /* Assets.xcassets */; };
 		BF692F642209AAFB00F8A0F1 /* OpenGpxTracker-Watch Extension.appex in Embed App Extensions */ = {isa = PBXBuildFile; fileRef = BF692F632209AAFB00F8A0F1 /* OpenGpxTracker-Watch Extension.appex */; settings = {ATTRIBUTES = (RemoveHeadersOnCopy, ); }; };
 		BF692F692209AAFB00F8A0F1 /* InterfaceController.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF692F682209AAFB00F8A0F1 /* InterfaceController.swift */; };
@@ -75,7 +76,14 @@
 		BF692F872209B81A00F8A0F1 /* Int+asFilesize.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89EA86642157C444002E03A7 /* Int+asFilesize.swift */; };
 		BF692F882209B81E00F8A0F1 /* GPXFileInfo.swift in Sources */ = {isa = PBXBuildFile; fileRef = 89EA86622157B42C002E03A7 /* GPXFileInfo.swift */; };
 		BF692F892209B82500F8A0F1 /* StopWatch.swift in Sources */ = {isa = PBXBuildFile; fileRef = 899D677119CE8B3600C88A0A /* StopWatch.swift */; };
+		BF9AE57C2410011D00B103A0 /* DefaultNameSetupViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF9AE57B2410011D00B103A0 /* DefaultNameSetupViewController.swift */; };
+		BF9AE57E2410012700B103A0 /* DefaultDateFormat.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF9AE57D2410012700B103A0 /* DefaultDateFormat.swift */; };
 		BFA3CE8222C23B2300A8B965 /* CoreDataAlertView.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFA3CE8122C23B2300A8B965 /* CoreDataAlertView.swift */; };
+		BFB06F02244A037300AC0CDC /* UIColor+Keyboard.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFB06F01244A037300AC0CDC /* UIColor+Keyboard.swift */; };
+		BFE026C0244A0160005423E6 /* String+CountInstances.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFE026BF244A0160005423E6 /* String+CountInstances.swift */; };
+		BFE026C2244A01DB005423E6 /* UIInsetLabel.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFE026C1244A01DB005423E6 /* UIInsetLabel.swift */; };
+		BFE026C4244A0281005423E6 /* DateField.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFE026C3244A0281005423E6 /* DateField.swift */; };
+		BFE026C6244A02A4005423E6 /* DatePatternButton.swift in Sources */ = {isa = PBXBuildFile; fileRef = BFE026C5244A02A4005423E6 /* DatePatternButton.swift */; };
 		BFF3047F2312DC1D0012D263 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = BF1FE1A6230EAB9B00404B59 /* Localizable.strings */; };
 		BFF304802312DC1F0012D263 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = BF1FE1A6230EAB9B00404B59 /* Localizable.strings */; };
 		BFF304812312DC700012D263 /* Interface.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 45A430EB230D61240003C2F2 /* Interface.storyboard */; };
@@ -210,6 +218,7 @@
 		BF637D122216C5FA0003EC04 /* WatchConnectivity.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = WatchConnectivity.framework; path = System/Library/Frameworks/WatchConnectivity.framework; sourceTree = SDKROOT; };
 		BF6408A2225C63E700BB5242 /* CoreDataHelper.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CoreDataHelper.swift; sourceTree = "<group>"; };
 		BF6428B7220EC1A90015447E /* GPXFileTableRowController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = GPXFileTableRowController.swift; sourceTree = "<group>"; };
+		BF68ECC52448B887005C779F /* DateFieldTypeView.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = DateFieldTypeView.swift; sourceTree = "<group>"; };
 		BF692F572209AAF900F8A0F1 /* OpenGpxTracker-Watch.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = "OpenGpxTracker-Watch.app"; sourceTree = BUILT_PRODUCTS_DIR; };
 		BF692F5C2209AAFB00F8A0F1 /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
 		BF692F5E2209AAFB00F8A0F1 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
@@ -219,7 +228,14 @@
 		BF692F6C2209AAFB00F8A0F1 /* ComplicationController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ComplicationController.swift; sourceTree = "<group>"; };
 		BF692F6E2209AAFC00F8A0F1 /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
 		BF692F702209AAFC00F8A0F1 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+		BF9AE57B2410011D00B103A0 /* DefaultNameSetupViewController.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = DefaultNameSetupViewController.swift; sourceTree = "<group>"; };
+		BF9AE57D2410012700B103A0 /* DefaultDateFormat.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = DefaultDateFormat.swift; sourceTree = "<group>"; };
 		BFA3CE8122C23B2300A8B965 /* CoreDataAlertView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CoreDataAlertView.swift; sourceTree = "<group>"; };
+		BFB06F01244A037300AC0CDC /* UIColor+Keyboard.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "UIColor+Keyboard.swift"; sourceTree = "<group>"; };
+		BFE026BF244A0160005423E6 /* String+CountInstances.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "String+CountInstances.swift"; sourceTree = "<group>"; };
+		BFE026C1244A01DB005423E6 /* UIInsetLabel.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = UIInsetLabel.swift; sourceTree = "<group>"; };
+		BFE026C3244A0281005423E6 /* DateField.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DateField.swift; sourceTree = "<group>"; };
+		BFE026C5244A02A4005423E6 /* DatePatternButton.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DatePatternButton.swift; sourceTree = "<group>"; };
 		BFF3047A2312D8A00012D263 /* zh-Hans */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = "zh-Hans"; path = "zh-Hans.lproj/Main.strings"; sourceTree = "<group>"; };
 		BFF3047B2312D8A00012D263 /* zh-Hans */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = "zh-Hans"; path = "zh-Hans.lproj/Interface.strings"; sourceTree = "<group>"; };
 		BFF3047C2312D8A00012D263 /* zh-Hans */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = "zh-Hans"; path = "zh-Hans.lproj/Localizable.strings"; sourceTree = "<group>"; };
@@ -292,6 +308,11 @@
 				BF6408A2225C63E700BB5242 /* CoreDataHelper.swift */,
 				BF214B5D22E60B64000E96AA /* CLActivityType+Info.swift */,
 				BF15092123916FC100F51F1B /* UIColor+DarkMode.swift */,
+				BFB06F01244A037300AC0CDC /* UIColor+Keyboard.swift */,
+				BFE026BF244A0160005423E6 /* String+CountInstances.swift */,
+				BFE026C1244A01DB005423E6 /* UIInsetLabel.swift */,
+				BFE026C3244A0281005423E6 /* DateField.swift */,
+				BF9AE57D2410012700B103A0 /* DefaultDateFormat.swift */,
 			);
 			name = Models;
 			sourceTree = "<group>";
@@ -389,6 +410,7 @@
 				8913CD8B1BDB0A9A009EC729 /* PreferencesTableViewController.swift */,
 				8913CD8D1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift */,
 				DC4908A21C4A332E009484AE /* MapViewDelegate.swift */,
+				BF9AE57B2410011D00B103A0 /* DefaultNameSetupViewController.swift */,
 			);
 			name = Controllers;
 			sourceTree = "<group>";
@@ -400,6 +422,8 @@
 				BF5C0D2122D771D0001C2575 /* DistanceLabel.swift */,
 				BFA3CE8122C23B2300A8B965 /* CoreDataAlertView.swift */,
 				DC4908A01C4A136F009484AE /* TrackerButton.swift */,
+				BFE026C5244A02A4005423E6 /* DatePatternButton.swift */,
+				BF68ECC52448B887005C779F /* DateFieldTypeView.swift */,
 			);
 			name = Views;
 			sourceTree = "<group>";
@@ -751,12 +775,15 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				BFE026C6244A02A4005423E6 /* DatePatternButton.swift in Sources */,
 				BF15092223916FC100F51F1B /* UIColor+DarkMode.swift in Sources */,
 				899D677F19D2366100C88A0A /* GPXMapView.swift in Sources */,
 				BF6408A3225C63E700BB5242 /* CoreDataHelper.swift in Sources */,
 				89E327BE1A7481EA00FC559E /* GPXTileServer.swift in Sources */,
 				898EECE019C49B5800B4B207 /* ViewController.swift in Sources */,
 				120DE71022B274910055C4CB /* GPXSession.swift in Sources */,
+				BF68ECC62448B887005C779F /* DateFieldTypeView.swift in Sources */,
+				BF9AE57E2410012700B103A0 /* DefaultDateFormat.swift in Sources */,
 				899D677019CE024600C88A0A /* GPXFileManager.swift in Sources */,
 				898EECDE19C49B5800B4B207 /* OpenGpxTracker.xcdatamodeld in Sources */,
 				89EA86652157C444002E03A7 /* Int+asFilesize.swift in Sources */,
@@ -766,6 +793,7 @@
 				8913CD8E1BDB2477009EC729 /* PreferencesTableViewControllerDelegate.swift in Sources */,
 				899D676E19CD9FAF00C88A0A /* GPXTrackPoint+MapKit.swift in Sources */,
 				89F8749B1BBB7362004EF41A /* GPXTrack+length.swift in Sources */,
+				BFB06F02244A037300AC0CDC /* UIColor+Keyboard.swift in Sources */,
 				BFA3CE8222C23B2300A8B965 /* CoreDataAlertView.swift in Sources */,
 				899D678219D351DA00C88A0A /* AboutViewController.swift in Sources */,
 				89EA86612157A08E002E03A7 /* Date+timeAgo.swift in Sources */,
@@ -778,6 +806,10 @@
 				899D676C19CD9E9E00C88A0A /* GPXTrackSegment+MapKit.swift in Sources */,
 				899D677219CE8B3600C88A0A /* StopWatch.swift in Sources */,
 				89F6E0F7227E107C00CE3E3F /* Preferences.swift in Sources */,
+				BFE026C0244A0160005423E6 /* String+CountInstances.swift in Sources */,
+				BFE026C4244A0281005423E6 /* DateField.swift in Sources */,
+				BFE026C2244A01DB005423E6 /* UIInsetLabel.swift in Sources */,
+				BF9AE57C2410011D00B103A0 /* DefaultNameSetupViewController.swift in Sources */,
 				893BEC8619C7E62200C77354 /* GPXWaypoint+MKAnnotation.swift in Sources */,
 				DC4908A31C4A332E009484AE /* MapViewDelegate.swift in Sources */,
 				898EED3619C6402900B4B207 /* GPXFilesTableViewController.swift in Sources */,
diff --git a/OpenGpxTracker/DateField.swift b/OpenGpxTracker/DateField.swift
new file mode 100644
index 00000000..a0d78b49
--- /dev/null
+++ b/OpenGpxTracker/DateField.swift
@@ -0,0 +1,23 @@
+//
+//  DateField.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 17/4/20.
+//
+
+import Foundation
+
+/// To hold each date pattern type for `DateFieldTypeView`
+struct DateField {
+    
+    /// Title/type of the pattern (e.g. Year / Second)
+    var type: String
+    
+    /// Patterns of that falls under said type (e.g `YYYY` / `ss`)
+    var patterns: [String]
+    
+    /// To facilitate explanation of said pattern, that falls under same type, if needed.
+    ///
+    /// Key of subtitle should be accessible in `patterns`
+    var subtitles: [String : String]?
+}
diff --git a/OpenGpxTracker/DateFieldTypeView.swift b/OpenGpxTracker/DateFieldTypeView.swift
new file mode 100644
index 00000000..4e43592c
--- /dev/null
+++ b/OpenGpxTracker/DateFieldTypeView.swift
@@ -0,0 +1,240 @@
+//
+//  DateFieldTypeView.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 15/4/20.
+//
+
+import UIKit
+
+/// View that is meant to be attached above keyboard, supplementing default name inputs.
+@available(iOS 9.0, *)
+class DateFieldTypeView: UIScrollView {
+    
+    // MARK:- Localization Strings
+    private let kSingleDigit = NSLocalizedString("SINGLE_DIGIT", comment: "")
+    private let kFull = NSLocalizedString("FULL_TEXT", comment: "")
+    private let kText = NSLocalizedString("TEXT", comment: "")
+    
+    private let kOfMonth = NSLocalizedString("OF_MONTH", comment: "")
+    private let kOfYear = NSLocalizedString("OF_YEAR", comment: "")
+    
+    private let kAbbr = NSLocalizedString("ABBR_GMT", comment: "")
+    private let kUTCoffset = NSLocalizedString("UTC_OFFSET", comment: "")
+    private let kGMTshort = NSLocalizedString("GMT_SHORT", comment: "")
+    private let kGMTfull = NSLocalizedString("GMT_FULL", comment: "")
+    private let kLocation = NSLocalizedString("LOCATION", comment: "")
+    private let kLocationTime = NSLocalizedString("LOCATION_TIME", comment: "")
+    
+    /// date formatter to display current time example
+    private let dateFormatter = DateFormatter()
+    
+    /// valid date fields that are to be displayed.
+    private var dateFields: [DateField] {
+        get {
+            var fields = [DateField]()
+            // some subtitles are an attempt to clarify, in case of different Locales, causing some example of patterns to look the same.
+            fields.append(DateField(type: NSLocalizedString("YEAR", comment: ""),
+                                    patterns: ["YY", "YYYY"]))
+            fields.append(DateField(type: NSLocalizedString("MONTH", comment: ""),
+                                    patterns: ["M", "MM", "MMMMM", "MMM", "MMMM"],
+                                    subtitles: ["M" : kSingleDigit,
+                                                "MMMMM" : "•",
+                                                "MMM" : "• • •",
+                                                "MMMM" : kFull]))
+            fields.append(DateField(type: NSLocalizedString("DAY", comment: ""),
+                                    patterns: ["d", "dd", "D"],
+                                    subtitles: ["d" : kSingleDigit,
+                                                "dd" : kOfMonth,
+                                                "D" : kOfYear]))
+            fields.append(DateField(type: NSLocalizedString("HOUR", comment: ""),
+                                    patterns: ["h", "hh", "H", "HH", "K", "KK", "k", "kk"],
+                                    subtitles: ["h" : kSingleDigit, "hh" : "12hr",
+                                                "H" : kSingleDigit, "HH" : "24hr",
+                                                "K" : kSingleDigit, "KK" : "0-11",
+                                                "k" : kSingleDigit, "kk" : "1-24"]))
+            fields.append(DateField(type: NSLocalizedString("MINUTE", comment: ""),
+                                    patterns: ["m", "mm"],
+                                    subtitles: ["m" : kSingleDigit]))
+            fields.append(DateField(type: NSLocalizedString("SECOND", comment: ""),
+                                    patterns: ["s", "ss"],
+                                    subtitles: ["s" : kSingleDigit]))
+            fields.append(DateField(type: NSLocalizedString("DAY_OF_THE_WEEK", comment: ""),
+                                    patterns: ["e", "ee", "EEEEE", "EEEEEE", "E", "EEEE"],
+                                    subtitles: ["e" : kSingleDigit,
+                                                "EEEEE" : "•",
+                                                "EEEEEE" : "• •",
+                                                "E" : "• • •",
+                                                "EEEE" : kFull]))
+            fields.append(DateField(type: NSLocalizedString("TIME_OF_DAY", comment: ""),
+                                    patterns: ["aaaaa", "a", "B"],
+                                    subtitles: ["aaaaa" : kSingleDigit,
+                                                "B" : "Text"]))
+            fields.append(DateField(type: NSLocalizedString("WEEK", comment: ""),
+                                    patterns: ["w", "ww", "W"],
+                                    subtitles: ["w" : kSingleDigit,
+                                                "ww" : kOfYear,
+                                                "W" : kOfMonth]))
+            fields.append(DateField(type: NSLocalizedString("QUARTER", comment: ""),
+                                    patterns: ["Q", "QQ", "QQQ", "QQQQ"]))
+            fields.append(DateField(type: NSLocalizedString("ERA", comment: ""),
+                                    patterns: ["GGGGG", "G", "GGGG"]))
+            fields.append(DateField(type: NSLocalizedString("TIME_ZONE", comment: ""),
+                                    patterns: ["X", "Z", "ZZZZZ", "z", "O", "ZZZZ", "zzzz", "VVV", "VVVV"],
+                                    subtitles: ["z" : kAbbr,
+                                                "zzzz" : kFull,
+                                                "X" : kUTCoffset,
+                                                "O" : kGMTshort,
+                                                "ZZZZ" : kGMTfull,
+                                                "VVV" : kLocation,
+                                                "VVVV" : kLocationTime]))
+        
+            return fields
+        }
+    }
+    /// Default initializer
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        viewDidInit()
+    }
+    
+    /// Default initializer
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        viewDidInit()
+    }
+    
+    
+    /// Things to do, when view successfully inits.
+    func viewDidInit() {
+        if #available(iOS 13.0, *) {
+            self.backgroundColor = .keyboardColor
+        } else {
+            self.backgroundColor = .lightKeyboard
+        }
+        
+        // holds everything
+        let scrollStack = UIStackView()
+        scrollStack.axis = .horizontal
+        scrollStack.distribution = .fill
+        scrollStack.alignment = .leading
+        scrollStack.spacing = 25
+        scrollStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        for field in dateFields {
+            let vStack = genVStack(field: field)
+            scrollStack.addArrangedSubview(vStack)
+        }
+        
+        self.addSubview(scrollStack)
+        self.addConstraints( NSLayoutConstraint.constraints(withVisualFormat: "H:|-20-[sStack]-20-|", options: .alignAllLeft, metrics: nil, views: ["sStack": scrollStack]) )
+
+    }
+    
+    /// Generates vertical stack that encapsulates text title, with all date patterns of same type.
+    ///
+    ///     |TYPE|
+    ///     |genHStack(field:)|
+    ///
+    func genVStack(field: DateField) -> UIStackView {
+        let vStack = UIStackView()
+        vStack.axis = .vertical
+        vStack.distribution = .fill
+        vStack.alignment = .leading
+        vStack.spacing = 0
+        vStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        let textTitle = UIInsetLabel()
+        textTitle.text = field.type.uppercased()
+        textTitle.textColor = .gray
+        textTitle.font = .boldSystemFont(ofSize: 14)
+        textTitle.insets = UIEdgeInsets(top: 5, left: 5, bottom: 0, right: 0)
+        
+        vStack.addArrangedSubview(textTitle)
+        vStack.addArrangedSubview(genHStack(field: field))
+        
+        return vStack
+        
+    }
+    
+    /// Generates horizontal stack that encapsulates all date patterns of same type.
+    ///
+    ///     |pattern|pattern|pattern|
+    ///     |SUBTITLE|SUBTITLE|
+    ///
+    func genHStack(field: DateField) -> UIStackView {
+        
+        let hStack = UIStackView()
+        hStack.axis = .horizontal
+        hStack.distribution = .fill
+        hStack.alignment = .leading
+        hStack.spacing = 10
+        hStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        for pattern in field.patterns {
+            let button = DatePatternButton(type: .custom)
+            button.layer.cornerRadius = 10
+
+            dateFormatter.dateFormat = pattern
+            let currentDateExample = dateFormatter.string(from: Date())
+            button.setTitle(currentDateExample, for: .normal)
+            button.titleLabel?.text = currentDateExample
+            button.contentEdgeInsets = UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5)
+            
+            // color of button font
+            if #available(iOS 13, *) {
+                button.setTitleColor(.blackAndWhite, for: .normal)
+            }
+            else {
+                button.setTitleColor(.black, for: .normal)
+            }
+            
+            // for passing pattern to textField when needed
+            button.pattern = pattern
+            
+            button.addTarget(self, action: #selector(buttonTapped(sender:)), for: .touchUpInside)
+
+            
+            // Subtitle implementation (optional)
+            if let subtitle = field.subtitles?[pattern] {
+                let subtitleLabel = UIInsetLabel()
+                let subVStack = UIStackView()
+                subVStack.axis = .vertical
+                subVStack.distribution = .fill
+                subVStack.alignment = .leading
+                subVStack.spacing = -2.5
+                subVStack.translatesAutoresizingMaskIntoConstraints = false
+                
+                subtitleLabel.insets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+                subtitleLabel.text = subtitle.uppercased()
+                subtitleLabel.font = .boldSystemFont(ofSize: 8)
+                subtitleLabel.textAlignment = .center
+                
+                subVStack.addArrangedSubview(button)
+                subVStack.addArrangedSubview(subtitleLabel)
+                NSLayoutConstraint(item: subtitleLabel, attribute: .width, relatedBy: .equal, toItem: button, attribute: .width, multiplier: 1, constant: 0).isActive = true
+                hStack.addArrangedSubview(subVStack)
+            }
+            else {
+                hStack.addArrangedSubview(button)
+            }
+
+        }
+        
+        return hStack
+    }
+    
+    /// Called when any pattern button is tapped.
+    @objc func buttonTapped(sender: DatePatternButton) {
+        NotificationCenter.default.post(name: .dateFieldTapped, object: nil, userInfo: ["sender" : sender.pattern])
+    }
+
+    
+}
+/// Notifications name of for date pattern sending
+extension Notification.Name {
+    
+    /// When date field type is tapped from view.
+    static let dateFieldTapped = Notification.Name("dateFieldTapped")
+    
+}
diff --git a/OpenGpxTracker/DatePatternButton.swift b/OpenGpxTracker/DatePatternButton.swift
new file mode 100644
index 00000000..e5b06b6f
--- /dev/null
+++ b/OpenGpxTracker/DatePatternButton.swift
@@ -0,0 +1,40 @@
+//
+//  DateFieldButton.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 17/4/20.
+//
+
+import UIKit
+
+/// Each individual button that represents a date pattern
+class DatePatternButton: UIButton {
+    
+    /// DateFormatter-friendly pattern of that the button holds.
+    ///
+    /// such as `YYYY` or `s`
+    var pattern = String()
+    
+    /// For dealing with button tap highlight
+    override var isSelected: Bool {
+        didSet {
+            if #available(iOS 13.0, *) {
+                backgroundColor = isSelected ?  .highlightKeyboardColor : .keyboardColor
+            } else {
+                backgroundColor = isSelected ?  .highlightLightKeyboard : .lightKeyboard
+            }
+        }
+    }
+    
+    /// For dealing with button tap highlight
+    override var isHighlighted: Bool {
+        didSet {
+            if #available(iOS 13.0, *) {
+                backgroundColor = isHighlighted ?  .highlightKeyboardColor : .keyboardColor
+            } else {
+                backgroundColor = isHighlighted ?  .highlightLightKeyboard : .darkKeyboard
+            }
+        }
+    }
+    
+}
diff --git a/OpenGpxTracker/DefaultDateFormat.swift b/OpenGpxTracker/DefaultDateFormat.swift
new file mode 100644
index 00000000..a976c5d2
--- /dev/null
+++ b/OpenGpxTracker/DefaultDateFormat.swift
@@ -0,0 +1,64 @@
+//
+//  DefaultDateFormat.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent on 4/3/20.
+//
+
+import Foundation
+
+/// Handles processing of 'unprocessed' user input date format, processing of sample date format, etc
+class DefaultDateFormat {
+    
+    /// DateFormatter for use in each instance.
+    let dateFormatter = DateFormatter()
+    
+    /// returns a 'processed', `DateFormatter`-friendly date format.
+    func getDateFormat(unprocessed: String) -> String {
+        var newText = ""
+        
+        // prevents acknowledging unterminated date formats as valid
+        if (unprocessed.countInstances(of: "{") != unprocessed.countInstances(of: "}"))
+        || unprocessed.countInstances(of: "{}") > 0 {
+            newText = "'invalid'"
+        }
+        else {
+            let arr = unprocessed.components(separatedBy: CharacterSet(charactersIn: "{}"))
+            var lastField: String?
+            let arrCount = arr.count
+            for i in 0...arrCount - 1 {
+                if let lastField = lastField, lastField.countInstances(of: String(arr[i].last ?? Character(" "))) > 0 {
+                    newText = "'invalid: { ... } must not consecutively repeat'"
+                    break
+                }
+                if arr.count == 1  {
+                    newText += "'invalid'"
+                }
+                else if arrCount > 1 && !arr[i].isEmpty {
+                    newText += (i % 2 == 0) ? "'\(arr[i])'" : arr[i]
+                    lastField = (i % 2 != 0) ? arr[i] : nil
+                }
+            }
+        }
+        return newText
+    }
+    
+    /// Returns sample date time based on user input.
+    func getDate(processedFormat dateFormat: String, useUTC: Bool = false, useENLocale: Bool = false) -> String {
+        //processedDateFormat = DefaultDateFormat.getDateFormat(unprocessed: self.cellTextField.text!)
+        dateFormatter.dateFormat = dateFormat
+        dateFormatter.timeZone = useUTC ? TimeZone(secondsFromGMT: 0) : TimeZone.current
+        dateFormatter.locale = useENLocale ? Locale(identifier: "en_US_POSIX") : Locale.current
+        return dateFormatter.string(from: Date())
+    }
+    
+    /// Returns Preference stored date format and its settings.
+    func getDateFromPrefs() -> String {
+        let dateFormat = Preferences.shared.dateFormat
+        let useUTC = Preferences.shared.dateFormatUseUTC
+        let useEN = Preferences.shared.dateFormatUseEN
+        return getDate(processedFormat: dateFormat, useUTC: useUTC, useENLocale: useEN)
+        
+    }
+
+}
diff --git a/OpenGpxTracker/DefaultNameSetupViewController.swift b/OpenGpxTracker/DefaultNameSetupViewController.swift
new file mode 100644
index 00000000..9f2166ba
--- /dev/null
+++ b/OpenGpxTracker/DefaultNameSetupViewController.swift
@@ -0,0 +1,348 @@
+//
+//  DefaultNameSetupViewController.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent on 3/3/20.
+//
+
+import UIKit
+
+/// It is an editor for the default file name.
+class DefaultNameSetupViewController: UITableViewController, UITextFieldDelegate {
+    
+    /// text field for user to key in file name format.
+    var cellTextField = UITextField()
+    
+    /// sample text to depict possible date time format.
+    var cellSampleLabel = UILabel()
+    
+    /// proccessed date format text, such as 'TIME' HH:MM:ss, instead of TIME {HH}:{MM}:{ss}
+    ///
+    /// also used for final date formatting for use in default name callup when saving.
+    var processedDateFormat = String()
+    
+    ///
+    let defaultDateFormat = DefaultDateFormat()
+    
+    /// to use UTC time instead of local.
+    var useUTC = false
+    
+    /// to use en_US_POSIX instead of usual locale. (Force English name scheme)
+    var useEN = false
+    
+    /// Global Preferences
+    var preferences : Preferences = Preferences.shared
+
+    /// Built in presets. Should be made addable customs next time.
+    let presets =  [("Defaults", "dd-MMM-yyyy-HHmm", "{dd}-{MMM}-{yyyy}-{HH}{mm}"),
+                    ("ISO8601 (UTC)", "yyyy-MM-dd'T'HH:mm:ss'Z'", "{yyyy}-{MM}-{dd}T{HH}:{mm}:{ss}Z"),
+                    ("ISO8601 (UTC offset)", "yyyy-MM-dd'T'HH:mm:ssZ", "{yyyy}-{MM}-{dd}T{HH}:{mm}:{ss}{Z}"),
+                    ("Day, Date at time (12 hr)", "EEEE, MMM d, yyyy 'at' h:mm a", "{EEEE}, {MMM} {d}, {yyyy} at {h}:{mm} {a}"),
+                    ("Day, Date at time (24 hr)", "EEEE, MMM d, yyyy 'at' HH:mm", "{EEEE}, {MMM} {d}, {yyyy} at {HH}:{mm}")]
+    
+    /// Sections of table view
+    private enum kSections: Int, CaseIterable {
+        case input, settings, presets
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        useUTC = preferences.dateFormatUseUTC
+        useEN = preferences.dateFormatUseEN
+        
+        addNotificationObservers()
+    }
+    
+    func addNotificationObservers() {
+        let notificationCenter = NotificationCenter.default
+        
+        notificationCenter.addObserver(self, selector: #selector(dateButtonTapped(_:)), name: .dateFieldTapped, object: nil)
+    }
+    
+    // MARK:- Text Field Related
+    
+    @objc func dateButtonTapped(_ sender: Notification) {
+        if cellTextField.text != nil {
+            guard let notificationValues = sender.userInfo else { return }
+            let patternDict = notificationValues as! [String : String]
+            guard let pattern = patternDict["sender"] else { return }
+
+            cellTextField.insertText("{\(pattern)}")
+
+        }
+    }
+    
+    /// Handles text insertion as per keyboard bar button pressed.
+    @objc func buttonTapped(_ sender: UIBarButtonItem, for event: UIEvent) {
+        if cellTextField.text != nil {
+            switch sender.tag {
+            case 0: cellTextField.insertText("{dd}")
+            case 1: cellTextField.insertText("{MM}")
+            case 2: cellTextField.insertText("{yyyy}")
+            case 3: cellTextField.insertText("{HH}")
+            case 4: cellTextField.insertText("{mm}")
+            case 5: cellTextField.insertText("{ss}")
+            case 6:
+                cellTextField.insertText("{}")
+                
+                guard let currRange = cellTextField.selectedTextRange,
+                      let wantedRange = cellTextField.position(from: currRange.start, offset: -1) else { return }
+                cellTextField.selectedTextRange = cellTextField.textRange(from: wantedRange, to: wantedRange)
+            default: return
+            }
+            updateSampleTextField()
+        }
+    }
+
+    /// Call when text field is currently editing, and an update to sample label is required.
+    @objc func updateSampleTextField() {
+        processedDateFormat = defaultDateFormat.getDateFormat(unprocessed: self.cellTextField.text!)
+        //dateFormatter.dateFormat = processedDateFormat
+        cellSampleLabel.text = defaultDateFormat.getDate(processedFormat: processedDateFormat, useUTC: useUTC, useENLocale: useEN)
+    }
+    
+    // MARK:- UITextField Delegate
+    
+    /// Enables keyboard 'done' action to resign text field
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+        return true
+    
+    }
+    
+    /// handling of textfield when editing commence.
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        //remove checkmark from selected date format preset, as textfield edited == not preset anymore
+        let selectedIndexPath = IndexPath(row: preferences.dateFormatPreset, section: kSections.presets.rawValue)
+        tableView.cellForRow(at: selectedIndexPath)?.accessoryType = .none
+        useUTC = false // clear UTC value, unlock UTC cell, as format may now be custom.
+        lockUTCCell(useUTC)
+    }
+    
+    /// handling of textfield when editing is done.
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if cellTextField.text != preferences.dateFormatInput {
+            saveDateFormat(processedDateFormat, input: cellTextField.text)
+            // save if user input is custom / derivative of preset.
+        }
+        else {
+            // to get back preset set, and its rules (such as UTC for ISO8601 preset)
+            let selectedIndexPath = IndexPath(row: preferences.dateFormatPreset, section: kSections.presets.rawValue)
+            tableView.cellForRow(at: selectedIndexPath)?.accessoryType = .checkmark
+            
+            if preferences.dateFormatPreset == 1 {
+                useUTC = true
+                lockUTCCell(useUTC)
+            }
+        }
+    }
+
+    /// allows clear button to operate.
+    func textFieldShouldClear(_ textField: UITextField) -> Bool {
+        return true
+    }
+    
+    // MARK:- Preference Setting
+    
+    /// Saves date format to Preferences/UserDefaults
+    func saveDateFormat(_ dateFormat: String, input: String?, index: Int = -1) {
+        guard let input = input else { return }
+        print(dateFormat)
+        if dateFormat == "invalid" || dateFormat == "'invalid'" || dateFormat == "'invalid: { ... } must not consecutively repeat'" || input.isEmpty || dateFormat.isEmpty { return } // ensures no invalid date format (revert)
+        preferences.dateFormat = dateFormat
+        preferences.dateFormatInput = input
+        preferences.dateFormatPreset = index
+        preferences.dateFormatUseUTC = useUTC
+    }
+    
+    // MARK:- Table View
+    
+    /// Locks UTC cell such that it cannot be unchecked, for preset that require it.
+    func lockUTCCell(_ state: Bool) {
+        let indexPath = IndexPath(row: 0, section: kSections.settings.rawValue)
+        useUTC = state
+        
+        tableView.cellForRow(at: indexPath)?.accessoryType = state ? .checkmark : .none
+        tableView.cellForRow(at: indexPath)?.isUserInteractionEnabled = !state
+        tableView.cellForRow(at: indexPath)?.textLabel?.isEnabled = !state
+        updateSampleTextField()
+    }
+    
+    /// return number of sections based on `kSections`
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return kSections.allCases.count
+    }
+    
+    /// implement title of each section
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case kSections.input.rawValue: return NSLocalizedString("DEFAULT_NAME_DATE_FORMAT", comment: "no comment")
+        case kSections.settings.rawValue: return NSLocalizedString("DEFAULT_NAME_SETTINGS", comment: "no comment")
+        case kSections.presets.rawValue: return NSLocalizedString("DEFAULT_NAME_PRESET", comment: "no comment")
+        default: fatalError("Section out of range")
+        }
+    }
+    
+    /// implement footer for input section only
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section == kSections.input.rawValue {
+            return NSLocalizedString("DEFAULT_NAME_INPUT_FOOTER", comment: "no comment")
+        }
+        else { return nil }
+    }
+    
+    /// return number of rows in each section
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case kSections.input.rawValue: return 2
+        case kSections.settings.rawValue:
+            if Locale.current.languageCode == "en" {
+                return 1 // force locale to EN should only be shown if Locale is not EN.
+            }
+            else { return 2 }
+        case kSections.presets.rawValue: return presets.count
+        default: fatalError("Row out of range")
+        }
+    }
+    
+    /// setup each cell, according its requirements.
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        var cell = UITableViewCell(style: .default, reuseIdentifier: "inputCell")
+        
+        if indexPath.section == kSections.input.rawValue {
+            if indexPath.row == 0 {
+                
+                cellSampleLabel = UILabel(frame: CGRect(x: 87, y: 0, width: view.frame.width - 99, height: cell.frame.height))
+                cellSampleLabel.font = .boldSystemFont(ofSize: 17)
+                cell.addSubview(cellSampleLabel)
+                cellSampleLabel.text = ""
+                updateSampleTextField()
+                cellSampleLabel.adjustsFontSizeToFitWidth = true
+                cell.textLabel!.text = NSLocalizedString("DEFAULT_NAME_SAMPLE_OUTPUT_TITLE", comment: "no comment")
+                cell.textLabel?.font = .systemFont(ofSize: 17)
+                
+            }
+            else if indexPath.row == 1 {
+                cellTextField = UITextField(frame: CGRect(x: 22, y: 0, width: view.frame.width - 48, height: cell.frame.height))
+                //let textView = UITextView(frame: CGRect(x: 25, y: 5, width: view.frame.width - 50, height: cell.frame.height))
+                cellTextField.text = preferences.dateFormatInput
+                updateSampleTextField()
+                cellTextField.clearButtonMode = .whileEditing
+                cellTextField.delegate = self
+                cellTextField.returnKeyType = .done
+                let bar = UIToolbar()
+                let bracket = UIBarButtonItem(title: "{ ... }", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                bracket.tag = 6
+                let day = UIBarButtonItem(title: "Day", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                day.tag = 0
+                let month = UIBarButtonItem(title: "Month", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                month.tag = 1
+                let year = UIBarButtonItem(title: "Year", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                year.tag = 2
+                let hour = UIBarButtonItem(title: "Hr", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                hour.tag = 3
+                let min = UIBarButtonItem(title: "Min", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                min.tag = 4
+                let sec = UIBarButtonItem(title: "Sec", style: .plain, target: self, action: #selector(buttonTapped(_:for:)))
+                sec.tag = 5
+
+                bar.items = [bracket, day, month, year, hour, min, sec]
+                bar.sizeToFit()
+                
+                if #available(iOS 9, *) {
+                    let dateFieldSelector = DateFieldTypeView(frame: CGRect(x: 0, y: 0, width: cellTextField.frame.width, height: 75))
+                    cellTextField.inputAccessoryView = dateFieldSelector
+                }
+                else {
+                    cellTextField.inputAccessoryView = bar
+                }
+                cellTextField.addTarget(self, action: #selector(updateSampleTextField), for: UIControl.Event.editingChanged)
+
+
+                cell.contentView.addSubview(cellTextField)
+                if indexPath.section == kSections.input.rawValue {
+                    
+                }
+            }
+            cell.selectionStyle = .none
+            cellTextField.autocorrectionType = .no
+        }
+            
+        else if indexPath.section == kSections.settings.rawValue {
+            if indexPath.row == 0 {
+                cell.textLabel!.text = NSLocalizedString("DEFAULT_NAME_USE_UTC", comment: "no comment")//"Use UTC?"
+                cell.accessoryType = preferences.dateFormatUseUTC ? .checkmark : .none
+                
+                if preferences.dateFormatPreset == 1 {
+                    cell.isUserInteractionEnabled = !useUTC
+                    cell.textLabel?.isEnabled = !useUTC
+                }
+                updateSampleTextField()
+            }
+            else if indexPath.row == 1 {
+                cell.textLabel!.text = NSLocalizedString("DEFAULT_NAME_ENGLISH_LOCALE", comment: "no comment")//"Force English Locale?"
+                cell.accessoryType = preferences.dateFormatUseEN ? .checkmark : .none
+                updateSampleTextField()
+            }
+        }
+        
+        else if indexPath.section == kSections.presets.rawValue {
+            cell = UITableViewCell(style: .subtitle, reuseIdentifier: "presetCell")
+            cell.textLabel?.text = presets[indexPath.row].0
+            cell.detailTextLabel?.text = defaultDateFormat.getDate(processedFormat: presets[indexPath.row].1, useUTC: useUTC, useENLocale: useEN)//presets[indexPath.row].1
+            
+            if preferences.dateFormatPreset != -1 { // if not custom
+                cell.accessoryType = preferences.dateFormatPreset == indexPath.row ? .checkmark : .none
+            }
+        }
+
+        return cell
+    }
+    
+    /// handling of cell selection.
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if indexPath.section == kSections.settings.rawValue {
+            if indexPath.row == 0 {
+                //remove checkmark from selected utc setting
+                let newUseUTC = !preferences.dateFormatUseUTC
+                preferences.dateFormatUseUTC = newUseUTC
+                useUTC = newUseUTC
+                tableView.cellForRow(at: indexPath)?.accessoryType = newUseUTC ? .checkmark : .none
+            }
+            else if indexPath.row == 1 {
+                //remove checkmark from selected en locale setting
+                let newUseEN = !preferences.dateFormatUseEN
+                preferences.dateFormatUseEN = newUseEN
+                useEN = newUseEN
+                tableView.cellForRow(at: indexPath)?.accessoryType = newUseEN ? .checkmark : .none
+            }
+            updateSampleTextField()
+        }
+        if indexPath.section == kSections.presets.rawValue {
+            //cellSampleLabel.text = "{\(presets[indexPath.row].1)}"
+            cellTextField.text = presets[indexPath.row].2
+            
+            //update cell UI
+            //remove checkmark from selected date format preset
+            let selectedIndexPath = IndexPath(row: preferences.dateFormatPreset, section: indexPath.section)
+            tableView.cellForRow(at: selectedIndexPath)?.accessoryType = .none
+            
+            //add checkmark to new selected date format preset
+            tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
+            preferences.dateFormatPreset = indexPath.row
+            preferences.dateFormatInput = presets[indexPath.row].2
+            updateSampleTextField()
+            preferences.dateFormat = processedDateFormat
+            if preferences.dateFormatPreset == 1 {
+                lockUTCCell(true)
+            }
+            else {
+                lockUTCCell(false)
+            }
+            preferences.dateFormatUseUTC = useUTC
+        }
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+}
diff --git a/OpenGpxTracker/Preferences.swift b/OpenGpxTracker/Preferences.swift
index ffeb75f3..24d9f0c4 100644
--- a/OpenGpxTracker/Preferences.swift
+++ b/OpenGpxTracker/Preferences.swift
@@ -24,6 +24,20 @@ let kDefaultsKeyUseImperial: String = "UseImperial"
 /// Key on Defaults for the current selected activity type.
 let kDefaultsKeyActivityType: String = "ActivityType"
 
+/// Key on Defaults for the current date format..
+let kDefaultsKeyDateFormat: String = "DateFormat"
+
+/// Key on Defaults for the current input date format
+let kDefaultsKeyDateFormatInput: String = "DateFormatPresetInput"
+
+/// Key on Defaults for the current selected date format preset cell index.
+let kDefaultsKeyDateFormatPreset: String = "DateFormatPreset"
+
+/// Key on Defaults for the current selected date format, to use UTC time or not..
+let kDefaultsKeyDateFormatUseUTC: String = "DateFormatPresetUseUTC"
+
+/// Key on Defaults for the current selected date format, to use local Locale or `en_US_POSIX`
+let kDefaultsKeyDateFormatUseEN: String = "DateFormatPresetUseEN"
 
 /// A class to handle app preferences in one single place.
 /// When the app starts for the first time the following preferences are set:
@@ -53,6 +67,21 @@ class Preferences: NSObject {
     /// In memory value of the preference.
     private var _activityType: CLActivityType = .other
     
+    ///
+    private var _dateFormat = "dd-MMM-yyyy-HHmm"
+    
+    ///
+    private var _dateFormatInput = "{dd}-{MMM}-{yyyy}-{HH}{mm}"
+    
+    ///
+    private var _dateFormatPreset: Int = 0
+    
+    ///
+    private var _dateFormatUseUTC: Bool = false
+    
+    ///
+    private var _dateFormatUseEN: Bool = false
+    
     /// UserDefaults.standard shortcut
     private let defaults = UserDefaults.standard
     
@@ -89,6 +118,36 @@ class Preferences: NSObject {
             _activityType = CLActivityType(rawValue: activityTypeInt)!
             print("** Preferences:: loaded preference from defaults activityTypeInt \(activityTypeInt)")
         }
+        
+        // load previous date format
+        if let dateFormatStr = defaults.object(forKey: kDefaultsKeyDateFormat) as? String {
+            _dateFormat = dateFormatStr
+            print("** Preferences:: loaded preference from defaults dateFormatStr \(dateFormatStr)")
+        }
+        
+        // load previous date format (usr input)
+        if let dateFormatStrIn = defaults.object(forKey: kDefaultsKeyDateFormatInput) as? String {
+            _dateFormatInput = dateFormatStrIn
+            print("** Preferences:: loaded preference from defaults dateFormatStrIn \(dateFormatStrIn)")
+        }
+        
+        // load previous date format preset
+        if let dateFormatPresetInt = defaults.object(forKey: kDefaultsKeyDateFormatPreset) as? Int {
+            _dateFormatPreset = dateFormatPresetInt
+            print("** Preferences:: loaded preference from defaults dateFormatPresetInt \(dateFormatPresetInt)")
+        }
+        
+        // load previous date format, to use UTC time instead of local time
+        if let dateFormatUTCBool = defaults.object(forKey: kDefaultsKeyDateFormatUseUTC) as? Bool {
+            _dateFormatUseUTC = dateFormatUTCBool
+            print("** Preferences:: loaded preference from defaults dateFormatPresetUTCBool \(dateFormatUTCBool)")
+        }
+        
+        // load previous date format, to use EN locale instead of local locale
+        if let dateFormatENBool = defaults.object(forKey: kDefaultsKeyDateFormatUseEN) as? Bool {
+            _dateFormatUseEN = dateFormatENBool
+            print("** Preferences:: loaded preference from defaults dateFormatPresetENBool \(dateFormatENBool)")
+        }
     }
     
     /// If true, user prefers to display imperial units (miles, feets). Otherwise metric units
@@ -158,4 +217,69 @@ class Preferences: NSObject {
             defaults.set(newValue, forKey: kDefaultsKeyActivityType)
         }
     }
+    
+    /// Gets and sets the date formatter friendly date format
+    var dateFormat: String {
+        get {
+            return _dateFormat
+        }
+        
+        set {
+             _dateFormat = newValue
+             defaults.set(newValue, forKey: kDefaultsKeyDateFormat)
+        }
+    }
+    
+    /// Gets and sets the user friendly input date format
+    var dateFormatInput: String {
+        get {
+            return _dateFormatInput
+        }
+        
+        set {
+             _dateFormatInput = newValue
+             defaults.set(newValue, forKey: kDefaultsKeyDateFormatInput)
+        }
+    }
+    
+    /// Get and sets user preference of date format presets. (-1 if custom)
+    var dateFormatPreset: Int {
+        get {
+            return _dateFormatPreset
+        }
+        set {
+            _dateFormatPreset = newValue
+             defaults.set(newValue, forKey: kDefaultsKeyDateFormatPreset)
+        }
+    }
+    
+    /// Get date format preset name
+    var dateFormatPresetName: String {
+        get {
+            let presets =  ["Defaults", "ISO8601 (UTC)", "ISO8601 (UTC offset)", "Day, Date at time (12 hr)", "Day, Date at time (24 hr)"]
+            return _dateFormatPreset < presets.count ? presets[_dateFormatPreset] : "???"
+        }
+    }
+    
+    /// Get and sets whether to use UTC for date format
+    var dateFormatUseUTC: Bool {
+        get {
+            return _dateFormatUseUTC
+        }
+        set {
+            _dateFormatUseUTC = newValue
+             defaults.set(newValue, forKey: kDefaultsKeyDateFormatUseUTC)
+        }
+    }
+    
+    /// Get and sets whether to use local locale or EN
+    var dateFormatUseEN: Bool {
+        get {
+            return _dateFormatUseEN
+        }
+        set {
+            _dateFormatUseEN = newValue
+             defaults.set(newValue, forKey: kDefaultsKeyDateFormatUseEN)
+        }
+    }
 }
diff --git a/OpenGpxTracker/PreferencesTableViewController.swift b/OpenGpxTracker/PreferencesTableViewController.swift
index 080c097a..186734e9 100644
--- a/OpenGpxTracker/PreferencesTableViewController.swift
+++ b/OpenGpxTracker/PreferencesTableViewController.swift
@@ -24,6 +24,9 @@ let kMapSourceSection = 2
 /// Activity Type Section Id in PreferencesTableViewController
 let kActivityTypeSection = 3
 
+/// Default Name Section Id in PreferencesTableViewController
+let kDefaultNameSection = 4
+
 /// Cell Id of the Use Imperial units in UnitsSection
 let kUseImperialUnitsCell = 0
 
@@ -51,6 +54,10 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
     
     var cache : MapCache = MapCache(withConfig: MapCacheConfig(withUrlTemplate: ""))
     
+    // Compute once, better performance for scrolling table view (reuse)
+    /// Store cached size for reuse.
+    var cachedSize = String()
+    
     /// Does the following:
     /// 1. Defines the areas for navBar and the Table view
     /// 2. Sets the title
@@ -65,6 +72,9 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
         self.title = NSLocalizedString("PREFERENCES", comment: "no comment")
         let shareItem = UIBarButtonItem(title: NSLocalizedString("DONE", comment: "no comment"), style: UIBarButtonItem.Style.plain, target: self, action: #selector(PreferencesTableViewController.closePreferencesTableViewController))
         self.navigationItem.rightBarButtonItems = [shareItem]
+        
+        let fileSize = cache.diskCache.fileSize ?? 0
+        cachedSize = Int(fileSize).asFileSize()
     }
     
     /// Close this controller.
@@ -90,7 +100,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
     /// Returns 4 sections: Units, Cache, Map Source, Activity Type
     override func numberOfSections(in tableView: UITableView?) -> Int {
         // Return the number of sections.
-        return 4
+        return 5
     }
     
     /// Returns the title of the existing sections.
@@ -102,6 +112,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
         case kCacheSection: return NSLocalizedString("CACHE", comment: "no comment")
         case kMapSourceSection: return NSLocalizedString("MAP_SOURCE", comment: "no comment")
         case kActivityTypeSection: return NSLocalizedString("ACTIVITY_TYPE", comment: "no comment")
+        case kDefaultNameSection: return NSLocalizedString("DEFAULT_NAME_SECTION", comment: "no comment")
         default: fatalError("Unknown section")
         }
     }
@@ -115,6 +126,7 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
         case kUnitsSection: return 1
         case kMapSourceSection: return GPXTileServer.count
         case kActivityTypeSection: return CLActivityType.count
+        case kDefaultNameSection: return 1
         default: fatalError("Unknown section")
         }
     }
@@ -153,8 +165,8 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             case kUseOfflineCacheCell:
                 cell = UITableViewCell(style: .subtitle, reuseIdentifier: "CacheCell")
                 cell.textLabel?.text = NSLocalizedString("OFFLINE_CACHE", comment: "no comment")
-                let fileSize = cache.diskCache.fileSize ?? 0
-                cell.detailTextLabel?.text = Int(fileSize).asFileSize()
+                
+                cell.detailTextLabel?.text = cachedSize
                 if preferences.useCache {
                     cell.accessoryType = .checkmark
                 }
@@ -188,6 +200,16 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             }
         }
         
+        // Default Name section
+        if indexPath.section == kDefaultNameSection {
+            let dateFormatter = DefaultDateFormat()
+            cell = UITableViewCell(style: .subtitle, reuseIdentifier: "DefaultNameCell")
+            cell.textLabel?.text = preferences.dateFormatPreset == -1 ? preferences.dateFormatInput : preferences.dateFormatPresetName
+            let dateText = dateFormatter.getDate(processedFormat: preferences.dateFormat, useUTC: preferences.dateFormatUseUTC, useENLocale: preferences.dateFormatUseEN)
+            cell.detailTextLabel?.text = dateText
+            cell.accessoryType = .disclosureIndicator
+        }
+        
         return cell
     }
     
@@ -245,7 +267,8 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
                     cell.textLabel?.textColor = UIColor.gray
                     //Clear the size text
                     let cell2 = tableView.cellForRow(at: IndexPath(row: kUseOfflineCacheCell, section: kCacheSection))
-                    cell2?.detailTextLabel?.text = 0.asFileSize()
+                    self.cachedSize = 0.asFileSize()
+                    cell2?.detailTextLabel?.text = self.cachedSize
                 }
             default:
                 fatalError("didSelectRowAt: Unknown cell")
@@ -280,6 +303,11 @@ class PreferencesTableViewController: UITableViewController, UINavigationBarDele
             self.delegate?.didUpdateActivityType((indexPath as NSIndexPath).row + 1)
         }
         
+        if indexPath.section == kDefaultNameSection {
+            print("PreferencesTableView Default Name cell clicked")
+            self.navigationController?.pushViewController(DefaultNameSetupViewController(style: .grouped), animated: true)
+        }
+        
         //unselect row
         tableView.deselectRow(at: indexPath, animated: true)
     }
diff --git a/OpenGpxTracker/String+CountInstances.swift b/OpenGpxTracker/String+CountInstances.swift
new file mode 100644
index 00000000..0ed85409
--- /dev/null
+++ b/OpenGpxTracker/String+CountInstances.swift
@@ -0,0 +1,25 @@
+//
+//  String+CountInstances.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 17/4/20.
+//
+
+import Foundation
+
+/// from: https://stackoverflow.com/a/45073012/13292870
+/// count occurances
+extension String {
+    /// counts number of instances of stringToFind; must be of at least 1 character.
+    func countInstances(of stringToFind: String) -> Int {
+        assert(!stringToFind.isEmpty)
+        var count = 0
+        var searchRange: Range<String.Index>?
+        while let foundRange = range(of: stringToFind, options: [.literal], range: searchRange) {
+            count += 1
+            searchRange = Range(uncheckedBounds: (lower: foundRange.upperBound, upper: endIndex))
+        }
+        return count
+    }
+    
+}
diff --git a/OpenGpxTracker/UIColor+DarkMode.swift b/OpenGpxTracker/UIColor+DarkMode.swift
index 3c6533e3..a44f0d6c 100644
--- a/OpenGpxTracker/UIColor+DarkMode.swift
+++ b/OpenGpxTracker/UIColor+DarkMode.swift
@@ -28,5 +28,21 @@ extension UIColor {
             @unknown default:           return .systemGray
                 }
     }
+
+    static let keyboardColor = UIColor { (traitCollection: UITraitCollection) -> UIColor in
+        switch traitCollection.userInterfaceStyle {
+            case .unspecified, .light:  return .lightKeyboard
+            case .dark:                 return .darkKeyboard
+            @unknown default:           return .lightKeyboard
+                }
+    }
+    
+    static let highlightKeyboardColor = UIColor { (traitCollection: UITraitCollection) -> UIColor in
+        switch traitCollection.userInterfaceStyle {
+            case .unspecified, .light:  return .highlightLightKeyboard
+            case .dark:                 return .highlightDarkKeyboard
+            @unknown default:           return .highlightLightKeyboard
+                }
+    }
     
 }
diff --git a/OpenGpxTracker/UIColor+Keyboard.swift b/OpenGpxTracker/UIColor+Keyboard.swift
new file mode 100644
index 00000000..dda188e9
--- /dev/null
+++ b/OpenGpxTracker/UIColor+Keyboard.swift
@@ -0,0 +1,21 @@
+//
+//  UIColor+Keyboard.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 17/4/20.
+//
+
+import UIKit
+
+/// For `DateFieldTypeView`. Meant to match default keyboard colors.
+extension UIColor {
+    /// Default Light Appearance Keyboard
+    static let lightKeyboard = UIColor(red: 209/255, green: 213/255, blue: 219/255, alpha: 1.00)
+    /// Default Dark Appearance Keyboard
+    static let darkKeyboard = UIColor(red: 36/255, green: 36/255, blue: 36/255, alpha: 1.00)
+    
+    /// Button Highlight Light Appearance Keyboard
+    static let highlightLightKeyboard = UIColor(red: 229/255, green: 233/255, blue: 239/255, alpha: 1.00)
+    /// Button Highlight Dark Appearance Keyboard
+    static let highlightDarkKeyboard = UIColor(red: 56/255, green: 56/255, blue: 56/255, alpha: 1.00)
+}
diff --git a/OpenGpxTracker/UIInsetLabel.swift b/OpenGpxTracker/UIInsetLabel.swift
new file mode 100644
index 00000000..96ce6376
--- /dev/null
+++ b/OpenGpxTracker/UIInsetLabel.swift
@@ -0,0 +1,30 @@
+//
+//  UIInsetLabel.swift
+//  OpenGpxTracker
+//
+//  Created by Vincent Neo on 17/4/20.
+//
+
+import UIKit
+
+/// UILabel that accepts insets padding.
+class UIInsetLabel: UILabel {
+    /// Insert the insets you need here. Defaults to zero.
+    var insets = UIEdgeInsets.zero
+    
+    ///
+    override func drawText(in rect: CGRect) {
+        super.drawText(in: rect.inset(by: insets))
+    }
+    
+    ///
+    override var intrinsicContentSize: CGSize {
+        var size = super.intrinsicContentSize
+        
+        size.width += insets.left + insets.right
+        size.height += insets.top + insets.bottom
+
+        return size
+    }
+
+}
diff --git a/OpenGpxTracker/ViewController.swift b/OpenGpxTracker/ViewController.swift
index f815ad79..c19c5d7f 100644
--- a/OpenGpxTracker/ViewController.swift
+++ b/OpenGpxTracker/ViewController.swift
@@ -838,10 +838,12 @@ class ViewController: UIViewController, UIGestureRecognizerDelegate  {
     /// returns a string with the format of current date dd-MMM-yyyy-HHmm' (20-Jun-2018-1133)
     ///
     func defaultFilename() -> String {
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "dd-MMM-yyyy-HHmm"
-        print("fileName:" + dateFormatter.string(from: Date()))
-        return dateFormatter.string(from: Date())
+        let defaultDate = DefaultDateFormat()
+        //let dateFormatter = DateFormatter()
+        //dateFormatter.dateFormat = "dd-MMM-yyyy-HHmm"
+        let dateStr = defaultDate.getDateFromPrefs()
+        print("fileName:" + dateStr)//dateFormatter.string(from: Date()))
+        return dateStr//dateFormatter.string(from: Date())
     }
     
     @objc func loadRecoveredFile(_ notification: Notification) {
diff --git a/OpenGpxTracker/de.lproj/Localizable.strings b/OpenGpxTracker/de.lproj/Localizable.strings
index 1aa4192a..6a4a0c79 100644
--- a/OpenGpxTracker/de.lproj/Localizable.strings
+++ b/OpenGpxTracker/de.lproj/Localizable.strings
@@ -105,3 +105,19 @@
 
 "CONTINUE_SESSION" = "Sitzung fortsetzen";
 "SAVE_START_NEW" = "Speichern und Neustart";
+
+// Default File Name
+
+// in preferences section
+"DEFAULT_NAME_SECTION" = "Default file name setup";
+
+// in default name editor view
+"DEFAULT_NAME_DATE_FORMAT" = "Date Format";
+"DEFAULT_NAME_SETTINGS" = "Settings";
+"DEFAULT_NAME_PRESET" = "Presets";
+
+"DEFAULT_NAME_INPUT_FOOTER" = "Clicking done on the keyboard saves the date format for use, regardless if its a preset or custom. Date format should be encapsulated within { ... }. For full list of applicable date format syntax please refer to Unicode Technical Standard (UTS) #35.";
+
+"DEFAULT_NAME_SAMPLE_OUTPUT_TITLE" = "Sample: ";
+"DEFAULT_NAME_USE_UTC" = "Use UTC Time?";
+"DEFAULT_NAME_ENGLISH_LOCALE" = "Force English language based date formatting?";
diff --git a/OpenGpxTracker/en.lproj/Localizable.strings b/OpenGpxTracker/en.lproj/Localizable.strings
index cc73710b..2d1004ed 100644
--- a/OpenGpxTracker/en.lproj/Localizable.strings
+++ b/OpenGpxTracker/en.lproj/Localizable.strings
@@ -108,3 +108,48 @@
 
 "CONTINUE_SESSION" = "Continue session";
 "SAVE_START_NEW" = "Save and start new";
+
+// Default File Name
+
+// in preferences section
+"DEFAULT_NAME_SECTION" = "Default file name setup";
+
+// in default name editor view
+"DEFAULT_NAME_DATE_FORMAT" = "File Name Format";
+"DEFAULT_NAME_SETTINGS" = "Settings";
+"DEFAULT_NAME_PRESET" = "Presets";
+
+"DEFAULT_NAME_INPUT_FOOTER" = "Clicking done on the keyboard saves the date format for use, regardless if its a preset or custom. Date format should be encapsulated within { ... }. For full list of applicable date format syntax please refer to Unicode Technical Standard (UTS) #35.";
+
+"DEFAULT_NAME_SAMPLE_OUTPUT_TITLE" = "Sample: ";
+"DEFAULT_NAME_USE_UTC" = "Use UTC Time?";
+"DEFAULT_NAME_ENGLISH_LOCALE" = "Force English language based date formatting?";
+
+// DateFieldTypeView
+"YEAR" = "Year";
+"MONTH" = "Month";
+"DAY" = "Day";
+"HOUR" = "Hour";
+"MINUTE" = "Minute";
+"SECOND" = "Second";
+"DAY_OF_THE_WEEK" = "Day of the week";
+"TIME_OF_DAY" = "Time of day";
+"WEEK" = "Week";
+"QUARTER" = "Quarter";
+"ERA" = "Era";
+"TIME_ZONE" = "Time Zone";
+
+"SINGLE_DIGIT" = "Single";
+"FULL_TEXT" = "Full";
+"TEXT" = "Text";
+
+"OF_MONTH" = "Of Month";
+"OF_YEAR" = "Of Year";
+
+"ABBR_GMT" = "Abbr. / GMT";
+"UTC_OFFSET" = "UTC offset";
+"GMT_SHORT" = "GMT short";
+"GMT_FULL" = "GMT full";
+"LOCATION" = "Location";
+"LOCATION_TIME" = "Location's Time";
+
diff --git a/OpenGpxTracker/es.lproj/Localizable.strings b/OpenGpxTracker/es.lproj/Localizable.strings
index 1319d09f..3a4b6265 100644
--- a/OpenGpxTracker/es.lproj/Localizable.strings
+++ b/OpenGpxTracker/es.lproj/Localizable.strings
@@ -107,4 +107,48 @@
 "CONTINUE_SESSION_MESSAGE" = "¿Qué quieres hacer con la información recuperada de la última sesión?";
 
 "CONTINUE_SESSION" = "Continuar la sesión";
-"SAVE_START_NEW" = "Guardar y comernzar una nueva";
+"SAVE_START_NEW" = "Guardar y comenzar una nueva";
+
+// Default File Name
+
+// in preferences section
+"DEFAULT_NAME_SECTION" = "Ajuste de nombre por defecto";
+
+// in default name editor view
+"DEFAULT_NAME_DATE_FORMAT" = "Formato de fecha";
+"DEFAULT_NAME_SETTINGS" = "Ajustes";
+"DEFAULT_NAME_PRESET" = "Formatos predefinidos";
+
+"DEFAULT_NAME_INPUT_FOOTER" = "Al pulsar Aceptar en el teclado se guarda el formato independientemente de si es un formato predefinido o personalizado. Cada elemento de la de fecha debe encapsularse dentro de { ... }";
+
+"DEFAULT_NAME_SAMPLE_OUTPUT_TITLE" = "Ejemplo: ";
+"DEFAULT_NAME_USE_UTC" = "¿Usar hora UTC?";
+"DEFAULT_NAME_ENGLISH_LOCALE" = "¿Forzar texto de la fecha en inglés?";
+
+// DateFieldTypeView
+"YEAR" = "Año";
+"MONTH" = "Mes";
+"DAY" = "Día";
+"HOUR" = "Hora";
+"MINUTE" = "Minuto";
+"SECOND" = "Segundo";
+"DAY_OF_THE_WEEK" = "Día de la semana";
+"TIME_OF_DAY" = "Hora del día";
+"WEEK" = "Semana";
+"QUARTER" = "Trimestre";
+"ERA" = "Era";
+"TIME_ZONE" = "Huso horario";
+
+"SINGLE_DIGIT" = "Sencillo";
+"FULL_TEXT" = "Completo";
+"TEXT" = "Texto";
+
+"OF_MONTH" = "Del mes";
+"OF_YEAR" = "Del año";
+
+"ABBR_GMT" = "Abrev. / GMT";
+"UTC_OFFSET" = "Desplazaciento UTC";
+"GMT_SHORT" = "GMT corto";
+"GMT_FULL" = "GMT completo";
+"LOCATION" = "Ubicación";
+"LOCATION_TIME" = "Hora en la ubicación";
diff --git a/OpenGpxTracker/uk.lproj/Localizable.strings b/OpenGpxTracker/uk.lproj/Localizable.strings
index 8dd81ba9..5b3a8f83 100644
--- a/OpenGpxTracker/uk.lproj/Localizable.strings
+++ b/OpenGpxTracker/uk.lproj/Localizable.strings
@@ -108,3 +108,19 @@
 
 "CONTINUE_SESSION" = "Продовжити сеанс";
 "SAVE_START_NEW" = "Зберегти та розпочати новий";
+
+// Default File Name
+
+// in preferences section
+"DEFAULT_NAME_SECTION" = "Default file name setup";
+
+// in default name editor view
+"DEFAULT_NAME_DATE_FORMAT" = "Date Format";
+"DEFAULT_NAME_SETTINGS" = "Settings";
+"DEFAULT_NAME_PRESET" = "Presets";
+
+"DEFAULT_NAME_INPUT_FOOTER" = "Clicking done on the keyboard saves the date format for use, regardless if its a preset or custom. Date format should be encapsulated within { ... }. For full list of applicable date format syntax please refer to Unicode Technical Standard (UTS) #35.";
+
+"DEFAULT_NAME_SAMPLE_OUTPUT_TITLE" = "Sample: ";
+"DEFAULT_NAME_USE_UTC" = "Use UTC Time?";
+"DEFAULT_NAME_ENGLISH_LOCALE" = "Force English language based date formatting?";
diff --git a/OpenGpxTracker/zh-Hans.lproj/Localizable.strings b/OpenGpxTracker/zh-Hans.lproj/Localizable.strings
index 6e9bc6e8..75584bd7 100644
--- a/OpenGpxTracker/zh-Hans.lproj/Localizable.strings
+++ b/OpenGpxTracker/zh-Hans.lproj/Localizable.strings
@@ -112,3 +112,47 @@
 
 "CONTINUE_SESSION" = "继续记录";
 "SAVE_START_NEW" = "储存恢复好的记录，并从新开始";
+
+// Default File Name
+
+// in preferences section
+"DEFAULT_NAME_SECTION" = "预设文件名";
+
+// in default name editor view
+"DEFAULT_NAME_DATE_FORMAT" = "文件名格式（根据日期和时间）";
+"DEFAULT_NAME_SETTINGS" = "设置";
+"DEFAULT_NAME_PRESET" = "预定程序";
+
+"DEFAULT_NAME_INPUT_FOOTER" = "在键盘上，按下 “完成” 按钮，无论预定程序有被更改或新程序，就等于设定了文件名格式。关于日期或时间的句法设定，请使用 { ... } 来包围。全部日期或时间的句法可在 Unicode Technical Standard (UTS) #35 里了解。";
+
+"DEFAULT_NAME_SAMPLE_OUTPUT_TITLE" = "样品：";
+"DEFAULT_NAME_USE_UTC" = "要使用 协调世界时（UTC）吗？";
+"DEFAULT_NAME_ENGLISH_LOCALE" = "要使用英文日期和时间写法吗？";
+
+// DateFieldTypeView
+"YEAR" = "年";
+"MONTH" = "月";
+"DAY" = "日";
+"HOUR" = "小时";
+"MINUTE" = "分钟";
+"SECOND" = "秒";
+"DAY_OF_THE_WEEK" = "星期几";
+"TIME_OF_DAY" = "AM/PM";
+"WEEK" = "周";
+"QUARTER" = "季度";
+"ERA" = "时代";
+"TIME_ZONE" = "时区";
+
+"SINGLE_DIGIT" = "单数";
+"FULL_TEXT" = "整体";
+"TEXT" = "全文";
+
+"OF_MONTH" = "月";
+"OF_YEAR" = "年";
+
+"ABBR_GMT" = "简写时区 / GMT";
+"UTC_OFFSET" = "UTC 偏移量";
+"GMT_SHORT" = "GMT 简写";
+"GMT_FULL" = "GMT 整体";
+"LOCATION" = "地点";
+"LOCATION_TIME" = "地点时区";
